{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### MR files are nested in directories, make the structure flat"
   ]
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T18:34:42.140631Z",
     "start_time": "2025-01-06T18:34:42.138003Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import glob\n",
    "import os\n",
    "from os import write"
   ],
   "outputs": [],
   "execution_count": 2
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:40:25.573553Z",
     "start_time": "2025-01-06T22:40:25.570019Z"
    }
   },
   "cell_type": "code",
   "source": [
    "dataset_base_path = \"/media/jfallmann/T9/University/master_thesis/dataset\"\n",
    "\n",
    "mri_raw_path = f\"{dataset_base_path}/mri/raw\"\n",
    "mri_base_path = f\"{dataset_base_path}/mri\"\n",
    "snp_raw_path = f\"{dataset_base_path}/snp/raw\"\n",
    "mri_bids_path = f\"{dataset_base_path}/mri/bids\"\n",
    "mri_fastsurfer_out = f\"{dataset_base_path}/mri/processed\"\n",
    "tables_path = f\"{dataset_base_path}/tables\""
   ],
   "outputs": [],
   "execution_count": 106
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:40:28.272265Z",
     "start_time": "2025-01-06T22:40:26.934750Z"
    }
   },
   "source": [
    "# Get all the MR files\n",
    "mr_input_files = glob.glob(f'{mri_raw_path}/**/*.nii', recursive=True)\n",
    "print(f\"Found {len(mr_input_files)} MR files\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 2248 MR files\n"
     ]
    }
   ],
   "execution_count": 107
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:40:29.194491Z",
     "start_time": "2025-01-06T22:40:29.186277Z"
    }
   },
   "source": [
    "## are there any directories with multiple files?\n",
    "from collections import defaultdict\n",
    "dir_files = defaultdict(list)\n",
    "for file in mr_input_files:\n",
    "    dir_files[os.path.dirname(file)].append(file)\n",
    "\n",
    "for k, v in dir_files.items():\n",
    "    if len(v) > 1:\n",
    "        print(k, v)\n",
    "\n",
    "print(\"Done\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done\n"
     ]
    }
   ],
   "execution_count": 108
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:40:31.557319Z",
     "start_time": "2025-01-06T22:40:31.553603Z"
    }
   },
   "cell_type": "code",
   "source": [
    "def clean_id(id: str) -> str:\n",
    "    return id.replace(\"_\", \"\").replace(\".\", \"\").replace(\"-\", \"\").lower()"
   ],
   "outputs": [],
   "execution_count": 109
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:40:35.461430Z",
     "start_time": "2025-01-06T22:40:35.384042Z"
    }
   },
   "cell_type": "code",
   "source": [
    "import re\n",
    "# Get all subject ids where genome data is available\n",
    "genome_files = glob.glob(f'{snp_raw_path}/**/*.vcf', recursive=True)\n",
    "genome_split_regex = r'(.*)_SNP'\n",
    "genome_subjects = [re.search(genome_split_regex, os.path.basename(file)).group(1) for file in genome_files]\n",
    "genome_subjects = [clean_id(subject) for subject in genome_subjects]\n",
    "print(f\"Found {len(genome_subjects)} subjects with genome data\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 809 subjects with genome data\n"
     ]
    }
   ],
   "execution_count": 110
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:40:48.730449Z",
     "start_time": "2025-01-06T22:40:48.723759Z"
    }
   },
   "cell_type": "code",
   "source": [
    "banned_subjects = []\n",
    "\n",
    "with open(f'{dataset_base_path}/banned_subjects.txt') as subjects_file:\n",
    "    for line in subjects_file:\n",
    "        banned_subjects.append(line.strip())\n",
    "\n",
    "allowed_subjects = [subject for subject in genome_subjects if subject not in banned_subjects]\n",
    "print(f\"Found {len(allowed_subjects)} subjects after removing banned subjects\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 806 subjects after removing banned subjects\n"
     ]
    }
   ],
   "execution_count": 113
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:40:51.458251Z",
     "start_time": "2025-01-06T22:40:51.412560Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# files have the structure ADNI/<subject-id>/<preprocessing-name>/<session-id>/<some-id>/<filename>.nii\n",
    "# we want to group files by subject id, choose the folder with the longest name and use the most recent session\n",
    "# if there are multiple files in the same session, choose the one with the longest name\n",
    "# session name is a date string in the format YYYY-MM-DD_THH_MM.0\n",
    "\n",
    "files_per_subject = defaultdict(str)\n",
    "\n",
    "for file in mr_input_files:\n",
    "    file_splits = file.split('/')\n",
    "\n",
    "    adni_index = file_splits.index('ADNI')\n",
    "    subject_id = clean_id(file_splits[adni_index + 1])\n",
    "    preprocessing_name = file_splits[adni_index + 2]\n",
    "    session_id =  file_splits[adni_index + 3]\n",
    "\n",
    "    # skip if subject has no genome data\n",
    "    if subject_id not in allowed_subjects:\n",
    "        print(f\"Skipping {subject_id} as no genome data is available\")\n",
    "        continue\n",
    "\n",
    "    # get current file stored for subject\n",
    "    current_file = files_per_subject.get(subject_id, None)\n",
    "    if current_file is None:\n",
    "        files_per_subject[subject_id] = file\n",
    "    else:\n",
    "        current_file_splits = current_file.split('/')\n",
    "        current_session_id = current_file_splits[adni_index + 3]\n",
    "        current_preprocessing_name = current_file_splits[adni_index + 2]\n",
    "\n",
    "        if session_id > current_session_id:\n",
    "            files_per_subject[subject_id] = file\n",
    "        elif session_id == current_session_id:\n",
    "            if len(preprocessing_name) > len(current_preprocessing_name):\n",
    "                files_per_subject[subject_id] = file\n",
    "\n",
    "print(f\"Found {len(files_per_subject)} subjects with MR data\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Skipping 099s4124 as no genome data is available\n",
      "Skipping 153s4165 as no genome data is available\n",
      "Skipping 003s4142 as no genome data is available\n",
      "Skipping 137s4227 as no genome data is available\n",
      "Skipping 094s4295 as no genome data is available\n",
      "Skipping 109s4378 as no genome data is available\n",
      "Skipping 041s4375 as no genome data is available\n",
      "Skipping 941s4377 as no genome data is available\n",
      "Skipping 941s4377 as no genome data is available\n",
      "Skipping 109s4380 as no genome data is available\n",
      "Skipping 041s4495 as no genome data is available\n",
      "Skipping 109s4471 as no genome data is available\n",
      "Skipping 016s4601 as no genome data is available\n",
      "Skipping 137s4623 as no genome data is available\n",
      "Skipping 137s4587 as no genome data is available\n",
      "Skipping 109s4594 as no genome data is available\n",
      "Skipping 109s4594 as no genome data is available\n",
      "Skipping 021s4633 as no genome data is available\n",
      "Skipping 003s4644 as no genome data is available\n",
      "Skipping 024s4674 as no genome data is available\n",
      "Skipping 029s4652 as no genome data is available\n",
      "Skipping 135s4689 as no genome data is available\n",
      "Skipping 135s4689 as no genome data is available\n",
      "Skipping 137s4678 as no genome data is available\n",
      "Skipping 135s4710 as no genome data is available\n",
      "Skipping 016s4646 as no genome data is available\n",
      "Skipping 126s4712 as no genome data is available\n",
      "Skipping 135s4723 as no genome data is available\n",
      "Skipping 126s4743 as no genome data is available\n",
      "Skipping 073s4739 as no genome data is available\n",
      "Skipping 941s4764 as no genome data is available\n",
      "Skipping 941s4764 as no genome data is available\n",
      "Skipping 021s4744 as no genome data is available\n",
      "Skipping 128s4742 as no genome data is available\n",
      "Skipping 128s4742 as no genome data is available\n",
      "Skipping 128s4745 as no genome data is available\n",
      "Skipping 027s4757 as no genome data is available\n",
      "Skipping 073s4777 as no genome data is available\n",
      "Skipping 127s4765 as no genome data is available\n",
      "Skipping 072s4694 as no genome data is available\n",
      "Skipping 123s4780 as no genome data is available\n",
      "Skipping 123s4780 as no genome data is available\n",
      "Skipping 128s4774 as no genome data is available\n",
      "Skipping 035s4784 as no genome data is available\n",
      "Skipping 070s4798 as no genome data is available\n",
      "Skipping 032s4755 as no genome data is available\n",
      "Skipping 011s4827 as no genome data is available\n",
      "Skipping 011s4827 as no genome data is available\n",
      "Skipping 073s4825 as no genome data is available\n",
      "Skipping 073s4825 as no genome data is available\n",
      "Skipping 003s4839 as no genome data is available\n",
      "Skipping 021s4857 as no genome data is available\n",
      "Skipping 021s4857 as no genome data is available\n",
      "Skipping 027s4802 as no genome data is available\n",
      "Skipping 041s4874 as no genome data is available\n",
      "Skipping 116s4855 as no genome data is available\n",
      "Skipping 057s4888 as no genome data is available\n",
      "Skipping 052s4807 as no genome data is available\n",
      "Skipping 070s4856 as no genome data is available\n",
      "Skipping 070s4856 as no genome data is available\n",
      "Skipping 072s4769 as no genome data is available\n",
      "Skipping 127s4843 as no genome data is available\n",
      "Skipping 128s4832 as no genome data is available\n",
      "Skipping 128s4832 as no genome data is available\n",
      "Skipping 141s4819 as no genome data is available\n",
      "Skipping 072s4871 as no genome data is available\n",
      "Skipping 116s4898 as no genome data is available\n",
      "Skipping 116s4898 as no genome data is available\n",
      "Skipping 041s4876 as no genome data is available\n",
      "Skipping 024s4905 as no genome data is available\n",
      "Skipping 024s4905 as no genome data is available\n",
      "Skipping 016s4887 as no genome data is available\n",
      "Skipping 016s4887 as no genome data is available\n",
      "Skipping 052s4885 as no genome data is available\n",
      "Skipping 057s4909 as no genome data is available\n",
      "Skipping 027s4919 as no genome data is available\n",
      "Skipping 068s4914 as no genome data is available\n",
      "Skipping 037s4879 as no genome data is available\n",
      "Skipping 037s4879 as no genome data is available\n",
      "Skipping 032s4921 as no genome data is available\n",
      "Skipping 127s4940 as no genome data is available\n",
      "Skipping 020s4920 as no genome data is available\n",
      "Skipping 020s4920 as no genome data is available\n",
      "Skipping 135s4954 as no genome data is available\n",
      "Skipping 067s4918 as no genome data is available\n",
      "Skipping 067s4918 as no genome data is available\n",
      "Skipping 027s4926 as no genome data is available\n",
      "Skipping 016s4951 as no genome data is available\n",
      "Skipping 016s4951 as no genome data is available\n",
      "Skipping 009s0751 as no genome data is available\n",
      "Skipping 027s4962 as no genome data is available\n",
      "Skipping 016s4963 as no genome data is available\n",
      "Skipping 041s4989 as no genome data is available\n",
      "Skipping 023s4035 as no genome data is available\n",
      "Skipping 009s5000 as no genome data is available\n",
      "Skipping 009s5000 as no genome data is available\n",
      "Skipping 082s5014 as no genome data is available\n",
      "Skipping 082s5014 as no genome data is available\n",
      "Skipping 022s5004 as no genome data is available\n",
      "Skipping 033s5013 as no genome data is available\n",
      "Skipping 127s5028 as no genome data is available\n",
      "Skipping 073s4995 as no genome data is available\n",
      "Skipping 016s5007 as no genome data is available\n",
      "Skipping 052s4944 as no genome data is available\n",
      "Skipping 114s5047 as no genome data is available\n",
      "Skipping 009s5027 as no genome data is available\n",
      "Skipping 009s5027 as no genome data is available\n",
      "Skipping 009s5037 as no genome data is available\n",
      "Skipping 128s5066 as no genome data is available\n",
      "Skipping 073s5090 as no genome data is available\n",
      "Skipping 033s5087 as no genome data is available\n",
      "Skipping 041s5078 as no genome data is available\n",
      "Skipping 068s4968 as no genome data is available\n",
      "Skipping 941s5124 as no genome data is available\n",
      "Skipping 021s5099 as no genome data is available\n",
      "Skipping 036s5149 as no genome data is available\n",
      "Skipping 041s5097 as no genome data is available\n",
      "Skipping 041s5097 as no genome data is available\n",
      "Skipping 041s5131 as no genome data is available\n",
      "Skipping 036s5112 as no genome data is available\n",
      "Skipping 041s5173 as no genome data is available\n",
      "Skipping 021s4718 as no genome data is available\n",
      "Skipping 023s5120 as no genome data is available\n",
      "Skipping 067s5160 as no genome data is available\n",
      "Skipping 094s4649 as no genome data is available\n",
      "Skipping 036s4538 as no genome data is available\n",
      "Skipping 036s5210 as no genome data is available\n",
      "Skipping 033s5198 as no genome data is available\n",
      "Skipping 072s5207 as no genome data is available\n",
      "Skipping 020s5203 as no genome data is available\n",
      "Skipping 067s4767 as no genome data is available\n",
      "Skipping 128s4671 as no genome data is available\n",
      "Skipping 021s5236 as no genome data is available\n",
      "Skipping 029s5219 as no genome data is available\n",
      "Skipping 067s5212 as no genome data is available\n",
      "Skipping 036s5248 as no genome data is available\n",
      "Skipping 135s5269 as no genome data is available\n",
      "Skipping 082s5279 as no genome data is available\n",
      "Skipping 082s5278 as no genome data is available\n",
      "Skipping 011s4845 as no genome data is available\n",
      "Skipping 032s5263 as no genome data is available\n",
      "Skipping 032s5263 as no genome data is available\n",
      "Skipping 036s5271 as no genome data is available\n",
      "Skipping 041s5253 as no genome data is available\n",
      "Skipping 135s5273 as no genome data is available\n",
      "Skipping 073s5227 as no genome data is available\n",
      "Skipping 036s4820 as no genome data is available\n",
      "Skipping 016s4902 as no genome data is available\n",
      "Skipping 016s4638 as no genome data is available\n",
      "Skipping 003s5150 as no genome data is available\n",
      "Skipping 003s5209 as no genome data is available\n",
      "Skipping 007s5265 as no genome data is available\n",
      "Skipping 022s4922 as no genome data is available\n",
      "Skipping 051s5285 as no genome data is available\n",
      "Skipping 009s4814 as no genome data is available\n",
      "Skipping 128s4842 as no genome data is available\n",
      "Skipping 011s4949 as no genome data is available\n",
      "Skipping 035s4783 as no genome data is available\n",
      "Skipping 153s4838 as no genome data is available\n",
      "Skipping 073s4795 as no genome data is available\n",
      "Skipping 009s5125 as no genome data is available\n",
      "Skipping 009s5125 as no genome data is available\n",
      "Skipping 009s5252 as no genome data is available\n",
      "Skipping 041s5082 as no genome data is available\n",
      "Skipping 003s4872 as no genome data is available\n",
      "Skipping 051s5005 as no genome data is available\n",
      "Skipping 036s5063 as no genome data is available\n",
      "Skipping 098s4002 as no genome data is available\n",
      "Skipping 070s5040 as no genome data is available\n",
      "Skipping 041s5141 as no genome data is available\n",
      "Skipping 021s5129 as no genome data is available\n",
      "Skipping 941s4066 as no genome data is available\n",
      "Skipping 041s5100 as no genome data is available\n",
      "Skipping 027s4729 as no genome data is available\n",
      "Skipping 024s5054 as no genome data is available\n",
      "Skipping 029s5135 as no genome data is available\n",
      "Skipping 082s5179 as no genome data is available\n",
      "Skipping 013s5137 as no genome data is available\n",
      "Skipping 037s5126 as no genome data is available\n",
      "Skipping 128s5123 as no genome data is available\n",
      "Skipping 094s4459 as no genome data is available\n",
      "Skipping 094s4486 as no genome data is available\n",
      "Skipping 073s4986 as no genome data is available\n",
      "Skipping 099s4124 as no genome data is available\n",
      "Skipping 024s4186 as no genome data is available\n",
      "Skipping 003s4142 as no genome data is available\n",
      "Skipping 073s4230 as no genome data is available\n",
      "Skipping 094s4295 as no genome data is available\n",
      "Skipping 109s4378 as no genome data is available\n",
      "Skipping 941s4377 as no genome data is available\n",
      "Skipping 941s4377 as no genome data is available\n",
      "Skipping 094s4460 as no genome data is available\n",
      "Skipping 116s4537 as no genome data is available\n",
      "Skipping 109s4471 as no genome data is available\n",
      "Skipping 003s4524 as no genome data is available\n",
      "Skipping 016s4583 as no genome data is available\n",
      "Skipping 016s4575 as no genome data is available\n",
      "Skipping 024s4622 as no genome data is available\n",
      "Skipping 137s4587 as no genome data is available\n",
      "Skipping 020s4619 as no genome data is available\n",
      "Skipping 021s4633 as no genome data is available\n",
      "Skipping 003s4644 as no genome data is available\n",
      "Skipping 003s4644 as no genome data is available\n",
      "Skipping 029s4652 as no genome data is available\n",
      "Skipping 029s4652 as no genome data is available\n",
      "Skipping 137s4631 as no genome data is available\n",
      "Skipping 041s4509 as no genome data is available\n",
      "Skipping 094s4649 as no genome data is available\n",
      "Skipping 016s4646 as no genome data is available\n",
      "Skipping 141s4711 as no genome data is available\n",
      "Skipping 141s4711 as no genome data is available\n",
      "Skipping 135s4723 as no genome data is available\n",
      "Skipping 016s4638 as no genome data is available\n",
      "Skipping 126s4743 as no genome data is available\n",
      "Skipping 073s4739 as no genome data is available\n",
      "Skipping 073s4739 as no genome data is available\n",
      "Skipping 127s4624 as no genome data is available\n",
      "Skipping 021s4744 as no genome data is available\n",
      "Skipping 137s4756 as no genome data is available\n",
      "Skipping 128s4745 as no genome data is available\n",
      "Skipping 067s4767 as no genome data is available\n",
      "Skipping 073s4777 as no genome data is available\n",
      "Skipping 073s4777 as no genome data is available\n",
      "Skipping 016s4688 as no genome data is available\n",
      "Skipping 072s4694 as no genome data is available\n",
      "Skipping 123s4780 as no genome data is available\n",
      "Skipping 141s4803 as no genome data is available\n",
      "Skipping 035s4784 as no genome data is available\n",
      "Skipping 070s4798 as no genome data is available\n",
      "Skipping 073s4762 as no genome data is available\n",
      "Skipping 073s4762 as no genome data is available\n",
      "Skipping 011s4827 as no genome data is available\n",
      "Skipping 073s4825 as no genome data is available\n",
      "Skipping 003s4839 as no genome data is available\n",
      "Skipping 022s4805 as no genome data is available\n",
      "Skipping 027s4804 as no genome data is available\n",
      "Skipping 003s4872 as no genome data is available\n",
      "Skipping 003s4872 as no genome data is available\n",
      "Skipping 137s4852 as no genome data is available\n",
      "Skipping 057s4888 as no genome data is available\n",
      "Skipping 072s4769 as no genome data is available\n",
      "Skipping 127s4843 as no genome data is available\n",
      "Skipping 128s4832 as no genome data is available\n",
      "Skipping 141s4819 as no genome data is available\n",
      "Skipping 153s4838 as no genome data is available\n",
      "Skipping 153s4838 as no genome data is available\n",
      "Skipping 072s4871 as no genome data is available\n",
      "Skipping 137s4815 as no genome data is available\n",
      "Skipping 137s4816 as no genome data is available\n",
      "Skipping 024s4905 as no genome data is available\n",
      "Skipping 024s4905 as no genome data is available\n",
      "Skipping 016s4902 as no genome data is available\n",
      "Skipping 016s4902 as no genome data is available\n",
      "Skipping 052s4885 as no genome data is available\n",
      "Skipping 003s4892 as no genome data is available\n",
      "Skipping 003s4892 as no genome data is available\n",
      "Skipping 011s4906 as no genome data is available\n",
      "Skipping 027s4919 as no genome data is available\n",
      "Skipping 141s4907 as no genome data is available\n",
      "Skipping 021s4924 as no genome data is available\n",
      "Skipping 037s4879 as no genome data is available\n",
      "Skipping 037s4879 as no genome data is available\n",
      "Skipping 051s4929 as no genome data is available\n",
      "Skipping 020s4920 as no genome data is available\n",
      "Skipping 020s4920 as no genome data is available\n",
      "Skipping 135s4954 as no genome data is available\n",
      "Skipping 027s4936 as no genome data is available\n",
      "Skipping 027s4938 as no genome data is available\n",
      "Skipping 009s0751 as no genome data is available\n",
      "Skipping 073s4986 as no genome data is available\n",
      "Skipping 073s4986 as no genome data is available\n",
      "Skipping 027s4966 as no genome data is available\n",
      "Skipping 127s4992 as no genome data is available\n",
      "Skipping 051s4980 as no genome data is available\n",
      "Skipping 022s5004 as no genome data is available\n",
      "Skipping 073s4995 as no genome data is available\n",
      "Skipping 021s2077 as no genome data is available\n",
      "Skipping 016s5032 as no genome data is available\n",
      "Skipping 005s5038 as no genome data is available\n",
      "Skipping 114s5047 as no genome data is available\n",
      "Skipping 024s5054 as no genome data is available\n",
      "Skipping 024s5054 as no genome data is available\n",
      "Skipping 127s5056 as no genome data is available\n",
      "Skipping 009s5037 as no genome data is available\n",
      "Skipping 128s5066 as no genome data is available\n",
      "Skipping 041s5082 as no genome data is available\n",
      "Skipping 041s5082 as no genome data is available\n",
      "Skipping 068s4968 as no genome data is available\n",
      "Skipping 016s5031 as no genome data is available\n",
      "Skipping 021s5129 as no genome data is available\n",
      "Skipping 021s5129 as no genome data is available\n",
      "Skipping 036s5149 as no genome data is available\n",
      "Skipping 041s5100 as no genome data is available\n",
      "Skipping 041s5100 as no genome data is available\n",
      "Skipping 041s5141 as no genome data is available\n",
      "Skipping 036s5112 as no genome data is available\n",
      "Skipping 021s4718 as no genome data is available\n",
      "Skipping 024s4674 as no genome data is available\n",
      "Skipping 067s5160 as no genome data is available\n",
      "Skipping 067s4728 as no genome data is available\n",
      "Skipping 135s4689 as no genome data is available\n",
      "Skipping 036s4714 as no genome data is available\n",
      "Skipping 036s5210 as no genome data is available\n",
      "Skipping 072s5207 as no genome data is available\n",
      "Skipping 007s5196 as no genome data is available\n",
      "Skipping 128s4671 as no genome data is available\n",
      "Skipping 941s4764 as no genome data is available\n",
      "Skipping 021s5237 as no genome data is available\n",
      "Skipping 029s5219 as no genome data is available\n",
      "Skipping 128s5123 as no genome data is available\n",
      "Skipping 128s5123 as no genome data is available\n",
      "Skipping 067s5212 as no genome data is available\n",
      "Skipping 036s5248 as no genome data is available\n",
      "Skipping 009s4741 as no genome data is available\n",
      "Skipping 082s5279 as no genome data is available\n",
      "Skipping 082s5278 as no genome data is available\n",
      "Skipping 033s5259 as no genome data is available\n",
      "Skipping 036s5283 as no genome data is available\n",
      "Skipping 041s5253 as no genome data is available\n",
      "Skipping 128s4742 as no genome data is available\n",
      "Skipping 135s5273 as no genome data is available\n",
      "Skipping 036s4878 as no genome data is available\n",
      "Skipping 024s5290 as no genome data is available\n",
      "Skipping 116s4898 as no genome data is available\n",
      "Skipping 003s5154 as no genome data is available\n",
      "Skipping 003s5165 as no genome data is available\n",
      "Skipping 003s5209 as no genome data is available\n",
      "Skipping 007s4911 as no genome data is available\n",
      "Skipping 007s5265 as no genome data is available\n",
      "Skipping 009s4958 as no genome data is available\n",
      "Skipping 051s5285 as no genome data is available\n",
      "Skipping 036s4899 as no genome data is available\n",
      "Skipping 009s4814 as no genome data is available\n",
      "Skipping 011s4949 as no genome data is available\n",
      "Skipping 041s4877 as no genome data is available\n",
      "Skipping 123s4904 as no genome data is available\n",
      "Skipping 094s4858 as no genome data is available\n",
      "Skipping 094s4858 as no genome data is available\n",
      "Skipping 033s5013 as no genome data is available\n",
      "Skipping 082s5014 as no genome data is available\n",
      "Skipping 016s5007 as no genome data is available\n",
      "Skipping 009s5125 as no genome data is available\n",
      "Skipping 032s4755 as no genome data is available\n",
      "Skipping 073s5023 as no genome data is available\n",
      "Skipping 073s5023 as no genome data is available\n",
      "Skipping 041s5078 as no genome data is available\n",
      "Skipping 041s5097 as no genome data is available\n",
      "Skipping 027s4729 as no genome data is available\n",
      "Skipping 070s5040 as no genome data is available\n",
      "Skipping 016s5057 as no genome data is available\n",
      "Skipping 029s4585 as no genome data is available\n",
      "Skipping 116s4732 as no genome data is available\n",
      "Skipping 009s5147 as no genome data is available\n",
      "Skipping 082s5179 as no genome data is available\n",
      "Skipping 013s5137 as no genome data is available\n",
      "Skipping 013s4985 as no genome data is available\n",
      "Skipping 072s4941 as no genome data is available\n",
      "Skipping 032s5263 as no genome data is available\n",
      "Skipping 070s4708 as no genome data is available\n",
      "Skipping 123s4806 as no genome data is available\n",
      "Skipping 941s4066 as no genome data is available\n",
      "Skipping 067s4113 as no genome data is available\n",
      "Skipping 024s4186 as no genome data is available\n",
      "Skipping 073s4230 as no genome data is available\n",
      "Skipping 032s4304 as no genome data is available\n",
      "Skipping 094s4460 as no genome data is available\n",
      "Skipping 116s4537 as no genome data is available\n",
      "Skipping 003s4524 as no genome data is available\n",
      "Skipping 016s4583 as no genome data is available\n",
      "Skipping 016s4575 as no genome data is available\n",
      "Skipping 024s4622 as no genome data is available\n",
      "Skipping 035s4085 as no genome data is available\n",
      "Skipping 057s4110 as no genome data is available\n",
      "Skipping 020s4619 as no genome data is available\n",
      "Skipping 029s4585 as no genome data is available\n",
      "Skipping 029s4585 as no genome data is available\n",
      "Skipping 137s4631 as no genome data is available\n",
      "Skipping 137s4632 as no genome data is available\n",
      "Skipping 094s4649 as no genome data is available\n",
      "Skipping 141s4711 as no genome data is available\n",
      "Skipping 141s4711 as no genome data is available\n",
      "Skipping 021s4718 as no genome data is available\n",
      "Skipping 016s4638 as no genome data is available\n",
      "Skipping 067s4728 as no genome data is available\n",
      "Skipping 067s4728 as no genome data is available\n",
      "Skipping 127s4624 as no genome data is available\n",
      "Skipping 116s4732 as no genome data is available\n",
      "Skipping 116s4732 as no genome data is available\n",
      "Skipping 137s4756 as no genome data is available\n",
      "Skipping 128s4586 as no genome data is available\n",
      "Skipping 067s4767 as no genome data is available\n",
      "Skipping 127s4765 as no genome data is available\n",
      "Skipping 016s4688 as no genome data is available\n",
      "Skipping 009s4741 as no genome data is available\n",
      "Skipping 009s4741 as no genome data is available\n",
      "Skipping 123s4806 as no genome data is available\n",
      "Skipping 123s4806 as no genome data is available\n",
      "Skipping 141s4803 as no genome data is available\n",
      "Skipping 070s4793 as no genome data is available\n",
      "Skipping 037s4750 as no genome data is available\n",
      "Skipping 073s4762 as no genome data is available\n",
      "Skipping 073s4762 as no genome data is available\n",
      "Skipping 035s4783 as no genome data is available\n",
      "Skipping 011s4845 as no genome data is available\n",
      "Skipping 003s4840 as no genome data is available\n",
      "Skipping 022s4805 as no genome data is available\n",
      "Skipping 027s4801 as no genome data is available\n",
      "Skipping 027s4804 as no genome data is available\n",
      "Skipping 037s4770 as no genome data is available\n",
      "Skipping 003s4872 as no genome data is available\n",
      "Skipping 137s4852 as no genome data is available\n",
      "Skipping 068s4859 as no genome data is available\n",
      "Skipping 073s4853 as no genome data is available\n",
      "Skipping 127s4844 as no genome data is available\n",
      "Skipping 128s4842 as no genome data is available\n",
      "Skipping 153s4838 as no genome data is available\n",
      "Skipping 126s4896 as no genome data is available\n",
      "Skipping 137s4815 as no genome data is available\n",
      "Skipping 137s4816 as no genome data is available\n",
      "Skipping 003s4900 as no genome data is available\n",
      "Skipping 016s4902 as no genome data is available\n",
      "Skipping 011s4912 as no genome data is available\n",
      "Skipping 003s4892 as no genome data is available\n",
      "Skipping 003s4892 as no genome data is available\n",
      "Skipping 011s4906 as no genome data is available\n",
      "Skipping 041s4877 as no genome data is available\n",
      "Skipping 041s4877 as no genome data is available\n",
      "Skipping 141s4907 as no genome data is available\n",
      "Skipping 021s4924 as no genome data is available\n",
      "Skipping 011s4893 as no genome data is available\n",
      "Skipping 011s4893 as no genome data is available\n",
      "Skipping 011s4949 as no genome data is available\n",
      "Skipping 051s4929 as no genome data is available\n",
      "Skipping 051s4929 as no genome data is available\n",
      "Skipping 022s4922 as no genome data is available\n",
      "Skipping 005s4910 as no genome data is available\n",
      "Skipping 016s4952 as no genome data is available\n",
      "Skipping 016s4952 as no genome data is available\n",
      "Skipping 009s4958 as no genome data is available\n",
      "Skipping 009s4958 as no genome data is available\n",
      "Skipping 027s4955 as no genome data is available\n",
      "Skipping 072s4941 as no genome data is available\n",
      "Skipping 072s4941 as no genome data is available\n",
      "Skipping 141s4976 as no genome data is available\n",
      "Skipping 073s5016 as no genome data is available\n",
      "Skipping 051s4980 as no genome data is available\n",
      "Skipping 051s5005 as no genome data is available\n",
      "Skipping 135s5015 as no genome data is available\n",
      "Skipping 135s5015 as no genome data is available\n",
      "Skipping 099s4994 as no genome data is available\n",
      "Skipping 021s2077 as no genome data is available\n",
      "Skipping 005s5038 as no genome data is available\n",
      "Skipping 024s5054 as no genome data is available\n",
      "Skipping 127s5056 as no genome data is available\n",
      "Skipping 009s5037 as no genome data is available\n",
      "Skipping 009s5037 as no genome data is available\n",
      "Skipping 041s5082 as no genome data is available\n",
      "Skipping 020s5140 as no genome data is available\n",
      "Skipping 016s5031 as no genome data is available\n",
      "Skipping 016s5031 as no genome data is available\n",
      "Skipping 021s5129 as no genome data is available\n",
      "Skipping 067s5159 as no genome data is available\n",
      "Skipping 041s5100 as no genome data is available\n",
      "Skipping 041s5141 as no genome data is available\n",
      "Skipping 109s4380 as no genome data is available\n",
      "Skipping 021s5177 as no genome data is available\n",
      "Skipping 024s4674 as no genome data is available\n",
      "Skipping 068s5146 as no genome data is available\n",
      "Skipping 073s4739 as no genome data is available\n",
      "Skipping 941s5193 as no genome data is available\n",
      "Skipping 041s5204 as no genome data is available\n",
      "Skipping 036s4714 as no genome data is available\n",
      "Skipping 073s5167 as no genome data is available\n",
      "Skipping 135s4723 as no genome data is available\n",
      "Skipping 067s5205 as no genome data is available\n",
      "Skipping 135s4722 as no genome data is available\n",
      "Skipping 072s4769 as no genome data is available\n",
      "Skipping 021s5237 as no genome data is available\n",
      "Skipping 033s5235 as no genome data is available\n",
      "Skipping 128s5123 as no genome data is available\n",
      "Skipping 041s5244 as no genome data is available\n",
      "Skipping 153s5267 as no genome data is available\n",
      "Skipping 041s4876 as no genome data is available\n",
      "Skipping 016s5251 as no genome data is available\n",
      "Skipping 032s4823 as no genome data is available\n",
      "Skipping 033s5259 as no genome data is available\n",
      "Skipping 036s4894 as no genome data is available\n",
      "Skipping 036s5283 as no genome data is available\n",
      "Skipping 067s4782 as no genome data is available\n",
      "Skipping 073s4777 as no genome data is available\n",
      "Skipping 128s4772 as no genome data is available\n",
      "Skipping 135s5275 as no genome data is available\n",
      "Skipping 135s4863 as no genome data is available\n",
      "Skipping 036s4878 as no genome data is available\n",
      "Skipping 082s5282 as no genome data is available\n",
      "Skipping 024s5290 as no genome data is available\n",
      "Skipping 003s5130 as no genome data is available\n",
      "Skipping 003s5154 as no genome data is available\n",
      "Skipping 003s5165 as no genome data is available\n",
      "Skipping 023s4796 as no genome data is available\n",
      "Skipping 041s4989 as no genome data is available\n",
      "Skipping 036s4899 as no genome data is available\n",
      "Skipping 021s4744 as no genome data is available\n",
      "Skipping 032s5289 as no genome data is available\n",
      "Skipping 123s4904 as no genome data is available\n",
      "Skipping 051s5293 as no genome data is available\n",
      "Skipping 033s5013 as no genome data is available\n",
      "Skipping 016s5007 as no genome data is available\n",
      "Skipping 009s5224 as no genome data is available\n",
      "Skipping 009s4903 as no genome data is available\n",
      "Skipping 032s4755 as no genome data is available\n",
      "Skipping 073s5023 as no genome data is available\n",
      "Skipping 073s5023 as no genome data is available\n",
      "Skipping 041s5078 as no genome data is available\n",
      "Skipping 128s5066 as no genome data is available\n",
      "Skipping 098s4002 as no genome data is available\n",
      "Skipping 016s5057 as no genome data is available\n",
      "Skipping 941s5124 as no genome data is available\n",
      "Skipping 094s4630 as no genome data is available\n",
      "Skipping 029s4652 as no genome data is available\n",
      "Skipping 127s4992 as no genome data is available\n",
      "Skipping 041s5131 as no genome data is available\n",
      "Skipping 009s5147 as no genome data is available\n",
      "Skipping 013s4985 as no genome data is available\n",
      "Skipping 003s4644 as no genome data is available\n",
      "Skipping 135s5113 as no genome data is available\n",
      "Skipping 022s5004 as no genome data is available\n",
      "Skipping 094s4858 as no genome data is available\n",
      "Skipping 094s4486 as no genome data is available\n",
      "Skipping 014s4068 as no genome data is available\n",
      "Skipping 041s4083 as no genome data is available\n",
      "Skipping 067s4113 as no genome data is available\n",
      "Skipping 023s4035 as no genome data is available\n",
      "Skipping 032s4304 as no genome data is available\n",
      "Skipping 123s4362 as no genome data is available\n",
      "Skipping 003s4373 as no genome data is available\n",
      "Skipping 072s4606 as no genome data is available\n",
      "Skipping 035s4085 as no genome data is available\n",
      "Skipping 057s4110 as no genome data is available\n",
      "Skipping 137s4623 as no genome data is available\n",
      "Skipping 126s4675 as no genome data is available\n",
      "Skipping 029s4585 as no genome data is available\n",
      "Skipping 128s4671 as no genome data is available\n",
      "Skipping 137s4632 as no genome data is available\n",
      "Skipping 037s4706 as no genome data is available\n",
      "Skipping 037s4706 as no genome data is available\n",
      "Skipping 041s4509 as no genome data is available\n",
      "Skipping 135s4722 as no genome data is available\n",
      "Skipping 135s4722 as no genome data is available\n",
      "Skipping 021s4718 as no genome data is available\n",
      "Skipping 070s4719 as no genome data is available\n",
      "Skipping 041s4720 as no genome data is available\n",
      "Skipping 041s4720 as no genome data is available\n",
      "Skipping 067s4728 as no genome data is available\n",
      "Skipping 127s4749 as no genome data is available\n",
      "Skipping 116s4732 as no genome data is available\n",
      "Skipping 073s4748 as no genome data is available\n",
      "Skipping 128s4586 as no genome data is available\n",
      "Skipping 128s4586 as no genome data is available\n",
      "Skipping 073s4795 as no genome data is available\n",
      "Skipping 127s4765 as no genome data is available\n",
      "Skipping 128s4772 as no genome data is available\n",
      "Skipping 128s4772 as no genome data is available\n",
      "Skipping 009s4741 as no genome data is available\n",
      "Skipping 128s4792 as no genome data is available\n",
      "Skipping 123s4806 as no genome data is available\n",
      "Skipping 070s4793 as no genome data is available\n",
      "Skipping 037s4750 as no genome data is available\n",
      "Skipping 037s4750 as no genome data is available\n",
      "Skipping 009s4814 as no genome data is available\n",
      "Skipping 035s4783 as no genome data is available\n",
      "Skipping 011s4845 as no genome data is available\n",
      "Skipping 003s4840 as no genome data is available\n",
      "Skipping 027s4801 as no genome data is available\n",
      "Skipping 032s4823 as no genome data is available\n",
      "Skipping 032s4823 as no genome data is available\n",
      "Skipping 037s4770 as no genome data is available\n",
      "Skipping 037s4770 as no genome data is available\n",
      "Skipping 126s4891 as no genome data is available\n",
      "Skipping 067s4782 as no genome data is available\n",
      "Skipping 067s4782 as no genome data is available\n",
      "Skipping 068s4859 as no genome data is available\n",
      "Skipping 073s4853 as no genome data is available\n",
      "Skipping 073s4853 as no genome data is available\n",
      "Skipping 127s4844 as no genome data is available\n",
      "Skipping 135s4863 as no genome data is available\n",
      "Skipping 135s4863 as no genome data is available\n",
      "Skipping 027s4873 as no genome data is available\n",
      "Skipping 126s4896 as no genome data is available\n",
      "Skipping 057s4897 as no genome data is available\n",
      "Skipping 027s4869 as no genome data is available\n",
      "Skipping 003s4900 as no genome data is available\n",
      "Skipping 003s4900 as no genome data is available\n",
      "Skipping 023s4796 as no genome data is available\n",
      "Skipping 023s4796 as no genome data is available\n",
      "Skipping 011s4912 as no genome data is available\n",
      "Skipping 011s4912 as no genome data is available\n",
      "Skipping 007s4911 as no genome data is available\n",
      "Skipping 127s4928 as no genome data is available\n",
      "Skipping 041s4877 as no genome data is available\n",
      "Skipping 011s4893 as no genome data is available\n",
      "Skipping 011s4893 as no genome data is available\n",
      "Skipping 009s4903 as no genome data is available\n",
      "Skipping 009s4903 as no genome data is available\n",
      "Skipping 123s4904 as no genome data is available\n",
      "Skipping 011s4949 as no genome data is available\n",
      "Skipping 027s4943 as no genome data is available\n",
      "Skipping 022s4922 as no genome data is available\n",
      "Skipping 005s4910 as no genome data is available\n",
      "Skipping 016s4952 as no genome data is available\n",
      "Skipping 016s4952 as no genome data is available\n",
      "Skipping 009s4958 as no genome data is available\n",
      "Skipping 072s4941 as no genome data is available\n",
      "Skipping 141s4976 as no genome data is available\n",
      "Skipping 041s4974 as no genome data is available\n",
      "Skipping 041s4974 as no genome data is available\n",
      "Skipping 073s5016 as no genome data is available\n",
      "Skipping 052s4945 as no genome data is available\n",
      "Skipping 051s5005 as no genome data is available\n",
      "Skipping 033s5017 as no genome data is available\n",
      "Skipping 033s5017 as no genome data is available\n",
      "Skipping 135s5015 as no genome data is available\n",
      "Skipping 135s5015 as no genome data is available\n",
      "Skipping 099s4994 as no genome data is available\n",
      "Skipping 070s5040 as no genome data is available\n",
      "Skipping 020s5140 as no genome data is available\n",
      "Skipping 016s5057 as no genome data is available\n",
      "Skipping 067s5159 as no genome data is available\n",
      "Skipping 041s5131 as no genome data is available\n",
      "Skipping 041s5131 as no genome data is available\n",
      "Skipping 109s4380 as no genome data is available\n",
      "Skipping 016s4646 as no genome data is available\n",
      "Skipping 021s5177 as no genome data is available\n",
      "Skipping 068s5146 as no genome data is available\n",
      "Skipping 082s5184 as no genome data is available\n",
      "Skipping 941s5193 as no genome data is available\n",
      "Skipping 021s5194 as no genome data is available\n",
      "Skipping 041s5204 as no genome data is available\n",
      "Skipping 036s4736 as no genome data is available\n",
      "Skipping 035s4785 as no genome data is available\n",
      "Skipping 073s5167 as no genome data is available\n",
      "Skipping 135s5113 as no genome data is available\n",
      "Skipping 135s5113 as no genome data is available\n",
      "Skipping 007s5196 as no genome data is available\n",
      "Skipping 067s5205 as no genome data is available\n",
      "Skipping 072s4769 as no genome data is available\n",
      "Skipping 029s5158 as no genome data is available\n",
      "Skipping 033s5235 as no genome data is available\n",
      "Skipping 068s5206 as no genome data is available\n",
      "Skipping 041s5244 as no genome data is available\n",
      "Skipping 153s5261 as no genome data is available\n",
      "Skipping 153s5267 as no genome data is available\n",
      "Skipping 128s4774 as no genome data is available\n",
      "Skipping 041s4876 as no genome data is available\n",
      "Skipping 016s5251 as no genome data is available\n",
      "Skipping 023s5241 as no genome data is available\n",
      "Skipping 035s4784 as no genome data is available\n",
      "Skipping 036s4894 as no genome data is available\n",
      "Skipping 116s4855 as no genome data is available\n",
      "Skipping 135s5275 as no genome data is available\n",
      "Skipping 114s5234 as no genome data is available\n",
      "Skipping 082s5282 as no genome data is available\n",
      "Skipping 003s5130 as no genome data is available\n",
      "Skipping 003s5130 as no genome data is available\n",
      "Skipping 029s5166 as no genome data is available\n",
      "Skipping 003s5187 as no genome data is available\n",
      "Skipping 051s5294 as no genome data is available\n",
      "Skipping 041s4989 as no genome data is available\n",
      "Skipping 021s4744 as no genome data is available\n",
      "Skipping 032s5289 as no genome data is available\n",
      "Skipping 094s4858 as no genome data is available\n",
      "Skipping 135s4954 as no genome data is available\n",
      "Skipping 051s5293 as no genome data is available\n",
      "Skipping 082s5029 as no genome data is available\n",
      "Skipping 009s5224 as no genome data is available\n",
      "Skipping 013s5071 as no genome data is available\n",
      "Skipping 041s5026 as no genome data is available\n",
      "Skipping 032s4921 as no genome data is available\n",
      "Skipping 051s4929 as no genome data is available\n",
      "Skipping 128s5066 as no genome data is available\n",
      "Skipping 016s5032 as no genome data is available\n",
      "Skipping 016s5031 as no genome data is available\n",
      "Skipping 033s5087 as no genome data is available\n",
      "Skipping 009s5147 as no genome data is available\n",
      "Skipping 941s5124 as no genome data is available\n",
      "Skipping 094s4630 as no genome data is available\n",
      "Skipping 114s5047 as no genome data is available\n",
      "Skipping 009s5176 as no genome data is available\n",
      "Skipping 072s5207 as no genome data is available\n",
      "Skipping 009s5037 as no genome data is available\n",
      "Skipping 072s4871 as no genome data is available\n",
      "Skipping 022s5004 as no genome data is available\n",
      "Skipping 070s4708 as no genome data is available\n",
      "Skipping 094s4737 as no genome data is available\n",
      "Skipping 094s4737 as no genome data is available\n",
      "Skipping 014s4068 as no genome data is available\n",
      "Skipping 041s4083 as no genome data is available\n",
      "Skipping 023s4035 as no genome data is available\n",
      "Skipping 023s4035 as no genome data is available\n",
      "Skipping 153s4165 as no genome data is available\n",
      "Skipping 137s4227 as no genome data is available\n",
      "Skipping 123s4362 as no genome data is available\n",
      "Skipping 041s4375 as no genome data is available\n",
      "Skipping 109s4380 as no genome data is available\n",
      "Skipping 003s4373 as no genome data is available\n",
      "Skipping 041s4495 as no genome data is available\n",
      "Skipping 072s4606 as no genome data is available\n",
      "Skipping 035s4085 as no genome data is available\n",
      "Skipping 016s4601 as no genome data is available\n",
      "Skipping 137s4623 as no genome data is available\n",
      "Skipping 109s4594 as no genome data is available\n",
      "Skipping 109s4594 as no genome data is available\n",
      "Skipping 126s4675 as no genome data is available\n",
      "Skipping 024s4674 as no genome data is available\n",
      "Skipping 135s4689 as no genome data is available\n",
      "Skipping 128s4671 as no genome data is available\n",
      "Skipping 137s4678 as no genome data is available\n",
      "Skipping 135s4710 as no genome data is available\n",
      "Skipping 037s4706 as no genome data is available\n",
      "Skipping 037s4706 as no genome data is available\n",
      "Skipping 126s4712 as no genome data is available\n",
      "Skipping 135s4722 as no genome data is available\n",
      "Skipping 070s4719 as no genome data is available\n",
      "Skipping 041s4720 as no genome data is available\n",
      "Skipping 041s4720 as no genome data is available\n",
      "Skipping 941s4764 as no genome data is available\n",
      "Skipping 127s4749 as no genome data is available\n",
      "Skipping 128s4742 as no genome data is available\n",
      "Skipping 073s4748 as no genome data is available\n",
      "Skipping 027s4757 as no genome data is available\n",
      "Skipping 073s4795 as no genome data is available\n",
      "Skipping 073s4795 as no genome data is available\n",
      "Skipping 127s4765 as no genome data is available\n",
      "Skipping 128s4772 as no genome data is available\n",
      "Skipping 009s4741 as no genome data is available\n",
      "Skipping 128s4792 as no genome data is available\n",
      "Skipping 128s4774 as no genome data is available\n",
      "Skipping 128s4774 as no genome data is available\n",
      "Skipping 070s4793 as no genome data is available\n",
      "Skipping 032s4755 as no genome data is available\n",
      "Skipping 009s4814 as no genome data is available\n",
      "Skipping 021s4857 as no genome data is available\n",
      "Skipping 021s4857 as no genome data is available\n",
      "Skipping 027s4802 as no genome data is available\n",
      "Skipping 032s4823 as no genome data is available\n",
      "Skipping 041s4874 as no genome data is available\n",
      "Skipping 041s4874 as no genome data is available\n",
      "Skipping 041s4874 as no genome data is available\n",
      "Skipping 116s4855 as no genome data is available\n",
      "Skipping 116s4855 as no genome data is available\n",
      "Skipping 126s4891 as no genome data is available\n",
      "Skipping 052s4807 as no genome data is available\n",
      "Skipping 067s4782 as no genome data is available\n",
      "Skipping 070s4856 as no genome data is available\n",
      "Skipping 070s4856 as no genome data is available\n",
      "Skipping 127s4844 as no genome data is available\n",
      "Skipping 135s4863 as no genome data is available\n",
      "Skipping 027s4873 as no genome data is available\n",
      "Skipping 116s4898 as no genome data is available\n",
      "Skipping 057s4897 as no genome data is available\n",
      "Skipping 041s4876 as no genome data is available\n",
      "Skipping 027s4869 as no genome data is available\n",
      "Skipping 016s4887 as no genome data is available\n",
      "Skipping 016s4887 as no genome data is available\n",
      "Skipping 023s4796 as no genome data is available\n",
      "Skipping 057s4909 as no genome data is available\n",
      "Skipping 007s4911 as no genome data is available\n",
      "Skipping 007s4911 as no genome data is available\n",
      "Skipping 127s4928 as no genome data is available\n",
      "Skipping 041s4877 as no genome data is available\n",
      "Skipping 068s4914 as no genome data is available\n",
      "Skipping 032s4921 as no genome data is available\n",
      "Skipping 032s4921 as no genome data is available\n",
      "Skipping 009s4903 as no genome data is available\n",
      "Skipping 123s4904 as no genome data is available\n",
      "Skipping 127s4940 as no genome data is available\n",
      "Skipping 027s4943 as no genome data is available\n",
      "Skipping 067s4918 as no genome data is available\n",
      "Skipping 067s4918 as no genome data is available\n",
      "Skipping 005s4910 as no genome data is available\n",
      "Skipping 016s4951 as no genome data is available\n",
      "Skipping 016s4951 as no genome data is available\n",
      "Skipping 016s4963 as no genome data is available\n",
      "Skipping 041s4989 as no genome data is available\n",
      "Skipping 041s4974 as no genome data is available\n",
      "Skipping 041s4974 as no genome data is available\n",
      "Skipping 009s5000 as no genome data is available\n",
      "Skipping 009s5000 as no genome data is available\n",
      "Skipping 082s5014 as no genome data is available\n",
      "Skipping 052s4945 as no genome data is available\n",
      "Skipping 051s5005 as no genome data is available\n",
      "Skipping 051s5005 as no genome data is available\n",
      "Skipping 033s5013 as no genome data is available\n",
      "Skipping 033s5017 as no genome data is available\n",
      "Skipping 033s5017 as no genome data is available\n",
      "Skipping 127s5028 as no genome data is available\n",
      "Skipping 016s5007 as no genome data is available\n",
      "Skipping 052s4944 as no genome data is available\n",
      "Skipping 016s5032 as no genome data is available\n",
      "Skipping 016s5032 as no genome data is available\n",
      "Skipping 070s5040 as no genome data is available\n",
      "Skipping 009s5027 as no genome data is available\n",
      "Skipping 009s5027 as no genome data is available\n",
      "Skipping 027s4964 as no genome data is available\n",
      "Skipping 073s5090 as no genome data is available\n",
      "Skipping 033s5087 as no genome data is available\n",
      "Skipping 033s5087 as no genome data is available\n",
      "Skipping 041s5078 as no genome data is available\n",
      "Skipping 941s5124 as no genome data is available\n",
      "Skipping 016s5057 as no genome data is available\n",
      "Skipping 021s5099 as no genome data is available\n",
      "Skipping 041s5097 as no genome data is available\n",
      "Skipping 041s5131 as no genome data is available\n",
      "Skipping 041s5173 as no genome data is available\n",
      "Skipping 016s4646 as no genome data is available\n",
      "Skipping 023s5120 as no genome data is available\n",
      "Skipping 082s5184 as no genome data is available\n",
      "Skipping 094s4649 as no genome data is available\n",
      "Skipping 021s5194 as no genome data is available\n",
      "Skipping 036s4538 as no genome data is available\n",
      "Skipping 036s4736 as no genome data is available\n",
      "Skipping 033s5198 as no genome data is available\n",
      "Skipping 035s4785 as no genome data is available\n",
      "Skipping 135s5113 as no genome data is available\n",
      "Skipping 020s5203 as no genome data is available\n",
      "Skipping 067s4767 as no genome data is available\n",
      "Skipping 135s4723 as no genome data is available\n",
      "Skipping 021s5236 as no genome data is available\n",
      "Skipping 029s5158 as no genome data is available\n",
      "Skipping 068s5206 as no genome data is available\n",
      "Skipping 011s4827 as no genome data is available\n",
      "Skipping 153s5261 as no genome data is available\n",
      "Skipping 135s5269 as no genome data is available\n",
      "Skipping 011s4845 as no genome data is available\n",
      "Skipping 023s5241 as no genome data is available\n",
      "Skipping 032s5263 as no genome data is available\n",
      "Skipping 035s4784 as no genome data is available\n",
      "Skipping 036s5271 as no genome data is available\n",
      "Skipping 037s4750 as no genome data is available\n",
      "Skipping 128s4586 as no genome data is available\n",
      "Skipping 128s4832 as no genome data is available\n",
      "Skipping 073s5227 as no genome data is available\n",
      "Skipping 114s5234 as no genome data is available\n",
      "Skipping 036s4820 as no genome data is available\n",
      "Skipping 011s4912 as no genome data is available\n",
      "Skipping 016s4638 as no genome data is available\n",
      "Skipping 073s4853 as no genome data is available\n",
      "Skipping 003s5150 as no genome data is available\n",
      "Skipping 029s5166 as no genome data is available\n",
      "Skipping 003s5187 as no genome data is available\n",
      "Skipping 022s4922 as no genome data is available\n",
      "Skipping 051s5294 as no genome data is available\n",
      "Skipping 128s4842 as no genome data is available\n",
      "Skipping 003s4900 as no genome data is available\n",
      "Skipping 035s4783 as no genome data is available\n",
      "Skipping 123s4780 as no genome data is available\n",
      "Skipping 135s4954 as no genome data is available\n",
      "Skipping 082s5029 as no genome data is available\n",
      "Skipping 009s5252 as no genome data is available\n",
      "Skipping 013s5071 as no genome data is available\n",
      "Skipping 041s5026 as no genome data is available\n",
      "Skipping 036s5063 as no genome data is available\n",
      "Skipping 009s5125 as no genome data is available\n",
      "Skipping 041s5141 as no genome data is available\n",
      "Skipping 003s5130 as no genome data is available\n",
      "Skipping 009s5147 as no genome data is available\n",
      "Skipping 037s4770 as no genome data is available\n",
      "Skipping 114s5047 as no genome data is available\n",
      "Skipping 029s5135 as no genome data is available\n",
      "Skipping 009s5176 as no genome data is available\n",
      "Skipping 072s5207 as no genome data is available\n",
      "Skipping 037s5126 as no genome data is available\n",
      "Skipping 072s4871 as no genome data is available\n",
      "Skipping 073s4825 as no genome data is available\n",
      "Skipping 094s4459 as no genome data is available\n",
      "Skipping 094s4630 as no genome data is available\n",
      "Skipping 094s4630 as no genome data is available\n",
      "Found 410 subjects with MR data\n"
     ]
    }
   ],
   "execution_count": 114
  },
  {
   "cell_type": "code",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:40:58.889505Z",
     "start_time": "2025-01-06T22:40:57.069083Z"
    }
   },
   "source": [
    "# create a bids dataset in data/mri\n",
    "# folder structure should be data/mri/adni_thesis/sub-<subject_id>/ses-<session_id>\n",
    "\n",
    "if not os.path.exists(mri_bids_path):\n",
    "    os.makedirs(mri_bids_path)\n",
    "\n",
    "for subject_id, filepath in files_per_subject.items():\n",
    "    file_splits = filepath.split('/')\n",
    "\n",
    "    adni_index = file_splits.index('ADNI')\n",
    "\n",
    "    subject_id_clean = subject_id\n",
    "    session_id =  clean_id(file_splits[adni_index + 3])\n",
    "\n",
    "    folder = f'{mri_bids_path}/sub-{subject_id}/ses-{session_id}/anat'\n",
    "    if not os.path.exists(folder):\n",
    "        os.makedirs(folder)\n",
    "\n",
    "    new_file = f'{folder}/sub-{subject_id}_ses-{session_id}_T1w.nii'\n",
    "    os.rename(filepath, new_file)\n",
    "\n",
    "print(\"Done, created BIDS dataset\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done, created BIDS dataset\n"
     ]
    }
   ],
   "execution_count": 115
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:41:00.405139Z",
     "start_time": "2025-01-06T22:41:00.398820Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# get all subject ids in the bids dataset\n",
    "bids_subjects = glob.glob(f'{mri_bids_path}/sub-*')\n",
    "bids_subjects = [os.path.basename(subject).replace(\"sub-\",\"\") for subject in bids_subjects]\n",
    "print(f\"Found {len(bids_subjects)} subjects in BIDS dataset\")"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 410 subjects in BIDS dataset\n"
     ]
    }
   ],
   "execution_count": 116
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:41:02.707210Z",
     "start_time": "2025-01-06T22:41:02.702909Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# write list of subjects to dataset folder\n",
    "with open(f\"{dataset_base_path}/subjects.txt\", \"w\") as f:\n",
    "    for subject in bids_subjects:\n",
    "        f.write(f\"{subject}\\n\")"
   ],
   "outputs": [],
   "execution_count": 117
  },
  {
   "metadata": {},
   "cell_type": "markdown",
   "source": [
    "## Create batch file for fastsurfer"
   ]
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:41:22.810329Z",
     "start_time": "2025-01-06T22:41:22.800279Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# find all subjects not already processed\n",
    "processed_folders = glob.glob(f\"{mri_fastsurfer_out}/*\")\n",
    "processed_subjects = [processed_folder.split(\"/\")[-1] for processed_folder in processed_folders]\n",
    "\n",
    "# move all unused processed files to tmp folder\n",
    "#for processed_file, processed_subject in zip(processed_folders, processed_subjects):\n",
    "#    if processed_subject not in genome_subjects:\n",
    "#        os.rename(processed_file, f\"{mri_fastsurfer_out}_temp/{processed_subject}\")\n"
   ],
   "outputs": [],
   "execution_count": 119
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:41:24.427334Z",
     "start_time": "2025-01-06T22:41:24.315075Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# fastsurfer batchfile format is as follows:\n",
    "# <subject-id>=<path-to-mr-file>\n",
    "\n",
    "mr_files = glob.glob(f'{mri_bids_path}/**/*.nii', recursive=True)\n",
    "docker_filepath = \"/data\"\n",
    "\n",
    "with open(f\"{mri_bids_path}/fastsurfer_batch.txt\", \"w\") as f:\n",
    "    for filepath in mr_files:\n",
    "        # replace current filesystem path with docker path\n",
    "        docker_file = filepath.replace(mri_bids_path, docker_filepath)\n",
    "        file_splits = docker_file.split('/')\n",
    "        subject_id = clean_id(file_splits[2].replace(\"sub-\", \"\"))\n",
    "        if subject_id in processed_subjects:\n",
    "            continue\n",
    "        f.write(f\"{subject_id}={docker_file}\\n\")"
   ],
   "outputs": [],
   "execution_count": 120
  },
  {
   "cell_type": "code",
   "metadata": {},
   "source": [
    "# Run preprocessing pipeline\n",
    "#! smriprep-docker ./data/mri/adni_thesis ./data_processed/mri participant --fs-license-file ./license.txt --ncpus 4 --skull-strip-mode force"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {},
   "cell_type": "code",
   "source": [
    "# create fastsurfer output directory\n",
    "if not os.path.exists(mri_fastsurfer_out):\n",
    "    os.makedirs(mri_fastsurfer_out)"
   ],
   "outputs": [],
   "execution_count": null
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T15:01:19.464272Z",
     "start_time": "2025-01-07T13:30:00.439169Z"
    }
   },
   "cell_type": "code",
   "source": [
    "!docker run --privileged --gpus all -v /media/jfallmann/T9/University/master_thesis/dataset/mri/bids:/data -v /media/jfallmann/T9/University/master_thesis/dataset/mri/processed:/output -v /media/jfallmann/T9/University/master_thesis/license:/fs_license --entrypoint \"/fastsurfer/brun_fastsurfer.sh\" --rm --user $(id -u):$(id -g) deepmi/fastsurfer:latest --fs_license /fs_license/license.txt --subject_list /data/fastsurfer_batch.txt --sd /output --parallel --3T --seg_only --no_cereb --no_hypothal"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/fastsurfer/brun_fastsurfer.sh --fs_license /fs_license/license.txt --subject_list /data/fastsurfer_batch.txt --sd /output --parallel --3T --seg_only --no_cereb --no_hypothal\r\n",
      "Tue, 07 Jan 2025 13:30:00 +0000\r\n",
      "\r\n",
      "INFO: run_fastsurfer not explicitly specified, using $FASTSURFER_HOME/run_fastsurfer.sh.\r\n",
      "098s4050=/data/sub-098s4050/ses-201105271516560/anat/sub-098s4050_ses-201105271516560_T1w.nii 011s4075=/data/sub-011s4075/ses-201206221215290/anat/sub-011s4075_ses-201206221215290_T1w.nii 041s4051=/data/sub-041s4051/ses-201207121118550/anat/sub-041s4051_ses-201207121118550_T1w.nii 068s4134=/data/sub-068s4134/ses-201210151304380/anat/sub-068s4134_ses-201210151304380_T1w.nii 123s4127=/data/sub-123s4127/ses-201208030941490/anat/sub-123s4127_ses-201208030941490_T1w.nii 005s4168=/data/sub-005s4168/ses-201209060947300/anat/sub-005s4168_ses-201209060947300_T1w.nii 023s4122=/data/sub-023s4122/ses-201303051114060/anat/sub-023s4122_ses-201303051114060_T1w.nii 023s4115=/data/sub-023s4115/ses-201210021043050/anat/sub-023s4115_ses-201210021043050_T1w.nii 037s4146=/data/sub-037s4146/ses-201209191400510/anat/sub-037s4146_ses-201209191400510_T1w.nii 127s4240=/data/sub-127s4240/ses-201209251419310/anat/sub-127s4240_ses-201209251419310_T1w.nii 073s4311=/data/sub-073s4311/ses-201210151150390/anat/sub-073s4311_ses-201210151150390_T1w.nii 011s4278=/data/sub-011s4278/ses-201210251149410/anat/sub-011s4278_ses-201210251149410_T1w.nii 003s4288=/data/sub-003s4288/ses-201211010924190/anat/sub-003s4288_ses-201211010924190_T1w.nii 153s4297=/data/sub-153s4297/ses-201110310925210/anat/sub-153s4297_ses-201110310925210_T1w.nii 029s4307=/data/sub-029s4307/ses-201302141348210/anat/sub-029s4307_ses-201302141348210_T1w.nii 073s4393=/data/sub-073s4393/ses-201211281030330/anat/sub-073s4393_ses-201211281030330_T1w.nii 007s4387=/data/sub-007s4387/ses-201301211307270/anat/sub-007s4387_ses-201301211307270_T1w.nii 009s4359=/data/sub-009s4359/ses-201302131057010/anat/sub-009s4359_ses-201302131057010_T1w.nii 141s4423=/data/sub-141s4423/ses-201112201416050/anat/sub-141s4423_ses-201112201416050_T1w.nii 068s4424=/data/sub-068s4424/ses-201112231355220/anat/sub-068s4424_ses-201112231355220_T1w.nii 141s4456=/data/sub-141s4456/ses-201201170945340/anat/sub-141s4456_ses-201201170945340_T1w.nii 116s4453=/data/sub-116s4453/ses-201301161110340/anat/sub-116s4453_ses-201301161110340_T1w.nii 137s4466=/data/sub-137s4466/ses-201303011445310/anat/sub-137s4466_ses-201303011445310_T1w.nii 126s4507=/data/sub-126s4507/ses-201202031005260/anat/sub-126s4507_ses-201202031005260_T1w.nii 007s4516=/data/sub-007s4516/ses-201303131230280/anat/sub-007s4516_ses-201303131230280_T1w.nii 029s4385=/data/sub-029s4385/ses-201303051514580/anat/sub-029s4385_ses-201303051514580_T1w.nii 128s4571=/data/sub-128s4571/ses-201306191711580/anat/sub-128s4571_ses-201306191711580_T1w.nii 109s4531=/data/sub-109s4531/ses-201303251311530/anat/sub-109s4531_ses-201303251311530_T1w.nii 007s4620=/data/sub-007s4620/ses-201304151230160/anat/sub-007s4620_ses-201304151230160_T1w.nii\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:30:01 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-098s4050/ses-201105271516560/anat/sub-098s4050_ses-201105271516560_T1w.nii --asegdkt_segfile /output/098s4050/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/098s4050/mri/orig.mgz --brainmask_name /output/098s4050/mri/mask.mgz --aseg_name /output/098s4050/mri/aseg.auto_noCCseg.mgz --sid 098s4050 --seg_log /output/098s4050/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-098s4050/ses-201105271516560/anat/sub-098s4050_ses-201105271516560_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-098s4050/ses-201105271516560/anat/sub-098s4050_ses-201105271516560_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/098s4050/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4050/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19e+00 -6.85e-05 1.40e-01] [-4.90e-02 1.00e+00 3.23e-01] [-1.51e-01 -3.26e-01 9.94e-01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 13125.15625\r\n",
      "rescale:  min: 0.0  max: 7599.46533203125  scale: 0.033554992207832264\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4050/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.88batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-098s4050/ses-201105271516560/anat/sub-098s4050_ses-201105271516560_T1w.nii finished in 6.1149 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.97batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-098s4050/ses-201105271516560/anat/sub-098s4050_ses-201105271516560_T1w.nii finished in 6.2500 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.98batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-098s4050/ses-201105271516560/anat/sub-098s4050_ses-201105271516560_T1w.nii finished in 5.9569 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4050/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20815\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4050/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4050/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.16\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/098s4050/mri/orig.mgz --rescale /output/098s4050/mri/orig_nu.mgz --aseg /output/098s4050/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/098s4050/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/098s4050/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/098s4050/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 43.13\r\n",
      "- m: 2.3647\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/098s4050/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/098s4050/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/098s4050/stats/aseg+DKT.stats --normfile /output/098s4050/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 098s4050 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/098s4050/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/098s4050/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.33 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:33:10 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-011s4075/ses-201206221215290/anat/sub-011s4075_ses-201206221215290_T1w.nii --asegdkt_segfile /output/011s4075/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/011s4075/mri/orig.mgz --brainmask_name /output/011s4075/mri/mask.mgz --aseg_name /output/011s4075/mri/aseg.auto_noCCseg.mgz --sid 011s4075 --seg_log /output/011s4075/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-011s4075/ses-201206221215290/anat/sub-011s4075_ses-201206221215290_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-011s4075/ses-201206221215290/anat/sub-011s4075_ses-201206221215290_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/011s4075/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4075/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0. -0.01] [-0. 1.05 0.12] [ 0.01 -0.12 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1477.1085205078125\r\n",
      "rescale:  min: 0.0  max: 943.872314453125  scale: 0.2701636610114428\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4075/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.87batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-011s4075/ses-201206221215290/anat/sub-011s4075_ses-201206221215290_T1w.nii finished in 6.1166 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.57batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-011s4075/ses-201206221215290/anat/sub-011s4075_ses-201206221215290_T1w.nii finished in 6.3106 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.85batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-011s4075/ses-201206221215290/anat/sub-011s4075_ses-201206221215290_T1w.nii finished in 5.9752 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4075/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20210\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4075/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4075/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.13\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/011s4075/mri/orig.mgz --rescale /output/011s4075/mri/orig_nu.mgz --aseg /output/011s4075/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/011s4075/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/011s4075/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/011s4075/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 58.05\r\n",
      "- m: 1.7572\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/011s4075/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/011s4075/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/011s4075/stats/aseg+DKT.stats --normfile /output/011s4075/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 011s4075 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/011s4075/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/011s4075/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.67 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:36:18 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-041s4051/ses-201207121118550/anat/sub-041s4051_ses-201207121118550_T1w.nii --asegdkt_segfile /output/041s4051/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/041s4051/mri/orig.mgz --brainmask_name /output/041s4051/mri/mask.mgz --aseg_name /output/041s4051/mri/aseg.auto_noCCseg.mgz --sid 041s4051 --seg_log /output/041s4051/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-041s4051/ses-201207121118550/anat/sub-041s4051_ses-201207121118550_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-041s4051/ses-201207121118550/anat/sub-041s4051_ses-201207121118550_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/041s4051/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4051/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 0.09 -0.09] [-0.1 1.05 0.04] [ 0.11 -0.03 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2566.7763671875\r\n",
      "rescale:  min: 0.0  max: 1824.97802734375  scale: 0.1397277096925664\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4051/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.02batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-041s4051/ses-201207121118550/anat/sub-041s4051_ses-201207121118550_T1w.nii finished in 6.0948 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.16batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-041s4051/ses-201207121118550/anat/sub-041s4051_ses-201207121118550_T1w.nii finished in 6.2206 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.66batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-041s4051/ses-201207121118550/anat/sub-041s4051_ses-201207121118550_T1w.nii finished in 6.0023 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4051/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19988\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4051/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4051/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.32\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/041s4051/mri/orig.mgz --rescale /output/041s4051/mri/orig_nu.mgz --aseg /output/041s4051/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/041s4051/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/041s4051/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/041s4051/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 40.10\r\n",
      "- m: 2.5436\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/041s4051/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/041s4051/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/041s4051/stats/aseg+DKT.stats --normfile /output/041s4051/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 041s4051 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/041s4051/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/041s4051/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.56 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:39:27 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-068s4134/ses-201210151304380/anat/sub-068s4134_ses-201210151304380_T1w.nii --asegdkt_segfile /output/068s4134/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/068s4134/mri/orig.mgz --brainmask_name /output/068s4134/mri/mask.mgz --aseg_name /output/068s4134/mri/aseg.auto_noCCseg.mgz --sid 068s4134 --seg_log /output/068s4134/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-068s4134/ses-201210151304380/anat/sub-068s4134_ses-201210151304380_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-068s4134/ses-201210151304380/anat/sub-068s4134_ses-201210151304380_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/068s4134/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4134/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.01 -0.03] [-0.01 1.05 0.05] [ 0.04 -0.05 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1968.4508056640625\r\n",
      "rescale:  min: 0.0  max: 952.7301635742188  scale: 0.267651859623457\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4134/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.02batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-068s4134/ses-201210151304380/anat/sub-068s4134_ses-201210151304380_T1w.nii finished in 6.0953 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.94batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-068s4134/ses-201210151304380/anat/sub-068s4134_ses-201210151304380_T1w.nii finished in 6.2537 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.85batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-068s4134/ses-201210151304380/anat/sub-068s4134_ses-201210151304380_T1w.nii finished in 5.9747 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4134/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19047\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4134/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4134/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.11\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/068s4134/mri/orig.mgz --rescale /output/068s4134/mri/orig_nu.mgz --aseg /output/068s4134/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/068s4134/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/068s4134/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/068s4134/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 73.47\r\n",
      "- m: 1.3882\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/068s4134/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/068s4134/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/068s4134/stats/aseg+DKT.stats --normfile /output/068s4134/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 068s4134 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/068s4134/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/068s4134/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.42 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:42:36 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-123s4127/ses-201208030941490/anat/sub-123s4127_ses-201208030941490_T1w.nii --asegdkt_segfile /output/123s4127/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/123s4127/mri/orig.mgz --brainmask_name /output/123s4127/mri/mask.mgz --aseg_name /output/123s4127/mri/aseg.auto_noCCseg.mgz --sid 123s4127 --seg_log /output/123s4127/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-123s4127/ses-201208030941490/anat/sub-123s4127_ses-201208030941490_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-123s4127/ses-201208030941490/anat/sub-123s4127_ses-201208030941490_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/123s4127/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s4127/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 0.06] [-0.04 1.04 0.17] [-0.06 -0.17 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2549.858154296875\r\n",
      "rescale:  min: 0.0  max: 1315.726806640625  scale: 0.19380923054313828\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s4127/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.00batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-123s4127/ses-201208030941490/anat/sub-123s4127_ses-201208030941490_T1w.nii finished in 6.0974 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.08batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-123s4127/ses-201208030941490/anat/sub-123s4127_ses-201208030941490_T1w.nii finished in 6.2326 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-123s4127/ses-201208030941490/anat/sub-123s4127_ses-201208030941490_T1w.nii finished in 5.9661 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s4127/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19138\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s4127/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s4127/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.02\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/123s4127/mri/orig.mgz --rescale /output/123s4127/mri/orig_nu.mgz --aseg /output/123s4127/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/123s4127/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/123s4127/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/123s4127/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 57.30\r\n",
      "- m: 1.7802\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/123s4127/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/123s4127/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/123s4127/stats/aseg+DKT.stats --normfile /output/123s4127/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 123s4127 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/123s4127/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/123s4127/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.19 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:45:44 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-005s4168/ses-201209060947300/anat/sub-005s4168_ses-201209060947300_T1w.nii --asegdkt_segfile /output/005s4168/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/005s4168/mri/orig.mgz --brainmask_name /output/005s4168/mri/mask.mgz --aseg_name /output/005s4168/mri/aseg.auto_noCCseg.mgz --sid 005s4168 --seg_log /output/005s4168/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-005s4168/ses-201209060947300/anat/sub-005s4168_ses-201209060947300_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-005s4168/ses-201209060947300/anat/sub-005s4168_ses-201209060947300_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/005s4168/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4168/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.04 0.03] [-0.05 1.03 0.2 ] [-0.02 -0.2 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 17569.970703125\r\n",
      "rescale:  min: 0.0  max: 9030.96484375  scale: 0.028236185658111177\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4168/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.85batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-005s4168/ses-201209060947300/anat/sub-005s4168_ses-201209060947300_T1w.nii finished in 6.1191 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.04batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-005s4168/ses-201209060947300/anat/sub-005s4168_ses-201209060947300_T1w.nii finished in 6.2380 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.06batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-005s4168/ses-201209060947300/anat/sub-005s4168_ses-201209060947300_T1w.nii finished in 5.9467 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4168/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21135\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4168/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4168/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/005s4168/mri/orig.mgz --rescale /output/005s4168/mri/orig_nu.mgz --aseg /output/005s4168/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/005s4168/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/005s4168/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/005s4168/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 48.06\r\n",
      "- m: 2.1225\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/005s4168/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/005s4168/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/005s4168/stats/aseg+DKT.stats --normfile /output/005s4168/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 005s4168 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/005s4168/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/005s4168/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.58 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:48:53 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-023s4122/ses-201303051114060/anat/sub-023s4122_ses-201303051114060_T1w.nii --asegdkt_segfile /output/023s4122/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/023s4122/mri/orig.mgz --brainmask_name /output/023s4122/mri/mask.mgz --aseg_name /output/023s4122/mri/aseg.auto_noCCseg.mgz --sid 023s4122 --seg_log /output/023s4122/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-023s4122/ses-201303051114060/anat/sub-023s4122_ses-201303051114060_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-023s4122/ses-201303051114060/anat/sub-023s4122_ses-201303051114060_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/023s4122/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4122/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.06 0.06] [-0.08 1.04 0.14] [-0.06 -0.14 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1654.729248046875\r\n",
      "rescale:  min: 0.0  max: 1024.27734375  scale: 0.24895600938161433\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4122/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.77batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-023s4122/ses-201303051114060/anat/sub-023s4122_ses-201303051114060_T1w.nii finished in 6.1314 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.85batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-023s4122/ses-201303051114060/anat/sub-023s4122_ses-201303051114060_T1w.nii finished in 6.2670 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.85batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-023s4122/ses-201303051114060/anat/sub-023s4122_ses-201303051114060_T1w.nii finished in 5.9748 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4122/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  24229\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4122/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4122/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.25\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/023s4122/mri/orig.mgz --rescale /output/023s4122/mri/orig_nu.mgz --aseg /output/023s4122/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/023s4122/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/023s4122/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/023s4122/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 52.75\r\n",
      "- m: 1.9335\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/023s4122/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/023s4122/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/023s4122/stats/aseg+DKT.stats --normfile /output/023s4122/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 023s4122 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/023s4122/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/023s4122/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.94 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:52:04 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-023s4115/ses-201210021043050/anat/sub-023s4115_ses-201210021043050_T1w.nii --asegdkt_segfile /output/023s4115/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/023s4115/mri/orig.mgz --brainmask_name /output/023s4115/mri/mask.mgz --aseg_name /output/023s4115/mri/aseg.auto_noCCseg.mgz --sid 023s4115 --seg_log /output/023s4115/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-023s4115/ses-201210021043050/anat/sub-023s4115_ses-201210021043050_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-023s4115/ses-201210021043050/anat/sub-023s4115_ses-201210021043050_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/023s4115/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4115/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0. -0.01] [ 0. 1.01 0.29] [ 0.01 -0.29 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1516.811767578125\r\n",
      "rescale:  min: 0.0  max: 1001.0957641601562  scale: 0.25472088598229736\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4115/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.77batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-023s4115/ses-201210021043050/anat/sub-023s4115_ses-201210021043050_T1w.nii finished in 6.1311 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.61batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-023s4115/ses-201210021043050/anat/sub-023s4115_ses-201210021043050_T1w.nii finished in 6.3040 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.74batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-023s4115/ses-201210021043050/anat/sub-023s4115_ses-201210021043050_T1w.nii finished in 5.9903 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4115/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19975\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4115/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4115/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.18\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/023s4115/mri/orig.mgz --rescale /output/023s4115/mri/orig_nu.mgz --aseg /output/023s4115/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/023s4115/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/023s4115/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/023s4115/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 64.32\r\n",
      "- m: 1.5858\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/023s4115/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/023s4115/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/023s4115/stats/aseg+DKT.stats --normfile /output/023s4115/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 023s4115 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/023s4115/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/023s4115/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.78 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:55:15 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-037s4146/ses-201209191400510/anat/sub-037s4146_ses-201209191400510_T1w.nii --asegdkt_segfile /output/037s4146/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/037s4146/mri/orig.mgz --brainmask_name /output/037s4146/mri/mask.mgz --aseg_name /output/037s4146/mri/aseg.auto_noCCseg.mgz --sid 037s4146 --seg_log /output/037s4146/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-037s4146/ses-201209191400510/anat/sub-037s4146_ses-201209191400510_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-037s4146/ses-201209191400510/anat/sub-037s4146_ses-201209191400510_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/037s4146/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4146/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 0.01] [-0.03 1.05 0.13] [-0. -0.13 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1380.8267822265625\r\n",
      "rescale:  min: 0.0  max: 929.2964477539062  scale: 0.27440113498370805\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4146/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.90batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-037s4146/ses-201209191400510/anat/sub-037s4146_ses-201209191400510_T1w.nii finished in 6.1119 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.71batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-037s4146/ses-201209191400510/anat/sub-037s4146_ses-201209191400510_T1w.nii finished in 6.2896 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.80batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-037s4146/ses-201209191400510/anat/sub-037s4146_ses-201209191400510_T1w.nii finished in 5.9826 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4146/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20960\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4146/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4146/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.21\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/037s4146/mri/orig.mgz --rescale /output/037s4146/mri/orig_nu.mgz --aseg /output/037s4146/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/037s4146/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/037s4146/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/037s4146/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 64.17\r\n",
      "- m: 1.5896\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/037s4146/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/037s4146/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/037s4146/stats/aseg+DKT.stats --normfile /output/037s4146/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 037s4146 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/037s4146/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/037s4146/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.43 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 13:58:24 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-127s4240/ses-201209251419310/anat/sub-127s4240_ses-201209251419310_T1w.nii --asegdkt_segfile /output/127s4240/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/127s4240/mri/orig.mgz --brainmask_name /output/127s4240/mri/mask.mgz --aseg_name /output/127s4240/mri/aseg.auto_noCCseg.mgz --sid 127s4240 --seg_log /output/127s4240/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-127s4240/ses-201209251419310/anat/sub-127s4240_ses-201209251419310_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-127s4240/ses-201209251419310/anat/sub-127s4240_ses-201209251419310_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/127s4240/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4240/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 -0.03] [ 0.01 1.04 0.17] [ 0.03 -0.17 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 31915.861328125\r\n",
      "rescale:  min: 0.0  max: 7627.890625  scale: 0.03342994971168717\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4240/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.14batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-127s4240/ses-201209251419310/anat/sub-127s4240_ses-201209251419310_T1w.nii finished in 6.0779 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.00batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-127s4240/ses-201209251419310/anat/sub-127s4240_ses-201209251419310_T1w.nii finished in 6.2454 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.10batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-127s4240/ses-201209251419310/anat/sub-127s4240_ses-201209251419310_T1w.nii finished in 5.9408 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4240/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18288\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4240/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4240/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.03\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/127s4240/mri/orig.mgz --rescale /output/127s4240/mri/orig_nu.mgz --aseg /output/127s4240/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/127s4240/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/127s4240/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/127s4240/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 43.09\r\n",
      "- m: 2.3670\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/127s4240/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/127s4240/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/127s4240/stats/aseg+DKT.stats --normfile /output/127s4240/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 127s4240 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/127s4240/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/127s4240/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.51 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:01:33 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s4311/ses-201210151150390/anat/sub-073s4311_ses-201210151150390_T1w.nii --asegdkt_segfile /output/073s4311/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s4311/mri/orig.mgz --brainmask_name /output/073s4311/mri/mask.mgz --aseg_name /output/073s4311/mri/aseg.auto_noCCseg.mgz --sid 073s4311 --seg_log /output/073s4311/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s4311/ses-201210151150390/anat/sub-073s4311_ses-201210151150390_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s4311/ses-201210151150390/anat/sub-073s4311_ses-201210151150390_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s4311/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4311/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 -0.01] [ 0.03 1.05 0.01] [ 0.01 -0.01 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 4161.56494140625\r\n",
      "rescale:  min: 0.0  max: 1847.73486328125  scale: 0.13800681313506483\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4311/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.91batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s4311/ses-201210151150390/anat/sub-073s4311_ses-201210151150390_T1w.nii finished in 6.1111 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.66batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s4311/ses-201210151150390/anat/sub-073s4311_ses-201210151150390_T1w.nii finished in 6.2978 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.94batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s4311/ses-201210151150390/anat/sub-073s4311_ses-201210151150390_T1w.nii finished in 5.9631 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4311/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18654\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4311/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4311/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s4311/mri/orig.mgz --rescale /output/073s4311/mri/orig_nu.mgz --aseg /output/073s4311/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s4311/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s4311/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s4311/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 41.36\r\n",
      "- m: 2.4663\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s4311/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s4311/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s4311/stats/aseg+DKT.stats --normfile /output/073s4311/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s4311 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s4311/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s4311/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.58 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:04:43 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-011s4278/ses-201210251149410/anat/sub-011s4278_ses-201210251149410_T1w.nii --asegdkt_segfile /output/011s4278/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/011s4278/mri/orig.mgz --brainmask_name /output/011s4278/mri/mask.mgz --aseg_name /output/011s4278/mri/aseg.auto_noCCseg.mgz --sid 011s4278 --seg_log /output/011s4278/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-011s4278/ses-201210251149410/anat/sub-011s4278_ses-201210251149410_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-011s4278/ses-201210251149410/anat/sub-011s4278_ses-201210251149410_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/011s4278/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4278/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.18 -0.15 -0.09] [ 0.18 1.03 0.14] [ 0.08 -0.15 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1510.2152099609375\r\n",
      "rescale:  min: 0.0  max: 891.0269775390625  scale: 0.28618662108782317\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4278/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-011s4278/ses-201210251149410/anat/sub-011s4278_ses-201210251149410_T1w.nii finished in 6.0997 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.67batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-011s4278/ses-201210251149410/anat/sub-011s4278_ses-201210251149410_T1w.nii finished in 6.2950 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-011s4278/ses-201210251149410/anat/sub-011s4278_ses-201210251149410_T1w.nii finished in 5.9834 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4278/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18040\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4278/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s4278/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.11\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/011s4278/mri/orig.mgz --rescale /output/011s4278/mri/orig_nu.mgz --aseg /output/011s4278/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/011s4278/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/011s4278/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/011s4278/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 63.79\r\n",
      "- m: 1.5989\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/011s4278/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/011s4278/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/011s4278/stats/aseg+DKT.stats --normfile /output/011s4278/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 011s4278 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/011s4278/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/011s4278/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.05 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:07:54 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-003s4288/ses-201211010924190/anat/sub-003s4288_ses-201211010924190_T1w.nii --asegdkt_segfile /output/003s4288/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/003s4288/mri/orig.mgz --brainmask_name /output/003s4288/mri/mask.mgz --aseg_name /output/003s4288/mri/aseg.auto_noCCseg.mgz --sid 003s4288 --seg_log /output/003s4288/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-003s4288/ses-201211010924190/anat/sub-003s4288_ses-201211010924190_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-003s4288/ses-201211010924190/anat/sub-003s4288_ses-201211010924190_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/003s4288/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4288/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.05 0.04] [ 0.05 1.01 0.08] [-0.06 -0.08 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 9766.3720703125\r\n",
      "rescale:  min: 0.0  max: 6357.908203125  scale: 0.04010753094463741\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4288/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-003s4288/ses-201211010924190/anat/sub-003s4288_ses-201211010924190_T1w.nii finished in 6.1282 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-003s4288/ses-201211010924190/anat/sub-003s4288_ses-201211010924190_T1w.nii finished in 6.2728 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.80batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-003s4288/ses-201211010924190/anat/sub-003s4288_ses-201211010924190_T1w.nii finished in 5.9818 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4288/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18270\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4288/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4288/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.97\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/003s4288/mri/orig.mgz --rescale /output/003s4288/mri/orig_nu.mgz --aseg /output/003s4288/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/003s4288/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/003s4288/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/003s4288/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 37.10\r\n",
      "- m: 2.7490\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/003s4288/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/003s4288/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/003s4288/stats/aseg+DKT.stats --normfile /output/003s4288/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 003s4288 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/003s4288/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/003s4288/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.35 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:11:03 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-153s4297/ses-201110310925210/anat/sub-153s4297_ses-201110310925210_T1w.nii --asegdkt_segfile /output/153s4297/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/153s4297/mri/orig.mgz --brainmask_name /output/153s4297/mri/mask.mgz --aseg_name /output/153s4297/mri/aseg.auto_noCCseg.mgz --sid 153s4297 --seg_log /output/153s4297/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-153s4297/ses-201110310925210/anat/sub-153s4297_ses-201110310925210_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-153s4297/ses-201110310925210/anat/sub-153s4297_ses-201110310925210_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/153s4297/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4297/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 -0.03] [ 0.04 1.05 0.14] [ 0.03 -0.14 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1681.7354736328125\r\n",
      "rescale:  min: 0.0  max: 896.364990234375  scale: 0.28448232893759545\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4297/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.68batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-153s4297/ses-201110310925210/anat/sub-153s4297_ses-201110310925210_T1w.nii finished in 6.1440 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.60batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-153s4297/ses-201110310925210/anat/sub-153s4297_ses-201110310925210_T1w.nii finished in 6.3058 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.85batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-153s4297/ses-201110310925210/anat/sub-153s4297_ses-201110310925210_T1w.nii finished in 5.9756 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4297/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20493\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4297/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4297/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.08\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/153s4297/mri/orig.mgz --rescale /output/153s4297/mri/orig_nu.mgz --aseg /output/153s4297/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/153s4297/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/153s4297/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/153s4297/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 68.97\r\n",
      "- m: 1.4790\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/153s4297/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/153s4297/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/153s4297/stats/aseg+DKT.stats --normfile /output/153s4297/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 153s4297 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/153s4297/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/153s4297/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.69 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:14:13 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-029s4307/ses-201302141348210/anat/sub-029s4307_ses-201302141348210_T1w.nii --asegdkt_segfile /output/029s4307/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/029s4307/mri/orig.mgz --brainmask_name /output/029s4307/mri/mask.mgz --aseg_name /output/029s4307/mri/aseg.auto_noCCseg.mgz --sid 029s4307 --seg_log /output/029s4307/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-029s4307/ses-201302141348210/anat/sub-029s4307_ses-201302141348210_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-029s4307/ses-201302141348210/anat/sub-029s4307_ses-201302141348210_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/029s4307/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4307/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.20e+00 5.08e-03 9.58e-04] [-6.11e-03 1.00e+00 1.61e-01] [-1.64e-04 -1.61e-01 1.00e+00]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 8409.365234375\r\n",
      "rescale:  min: 0.0  max: 5230.625  scale: 0.048751344246624445\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4307/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.10batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-029s4307/ses-201302141348210/anat/sub-029s4307_ses-201302141348210_T1w.nii finished in 6.0828 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.39batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-029s4307/ses-201302141348210/anat/sub-029s4307_ses-201302141348210_T1w.nii finished in 6.3383 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.00batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-029s4307/ses-201302141348210/anat/sub-029s4307_ses-201302141348210_T1w.nii finished in 5.9544 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4307/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18292\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4307/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4307/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.06\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/029s4307/mri/orig.mgz --rescale /output/029s4307/mri/orig_nu.mgz --aseg /output/029s4307/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/029s4307/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/029s4307/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/029s4307/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 46.82\r\n",
      "- m: 2.1787\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/029s4307/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/029s4307/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/029s4307/stats/aseg+DKT.stats --normfile /output/029s4307/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 029s4307 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/029s4307/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/029s4307/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.97 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:17:22 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s4393/ses-201211281030330/anat/sub-073s4393_ses-201211281030330_T1w.nii --asegdkt_segfile /output/073s4393/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s4393/mri/orig.mgz --brainmask_name /output/073s4393/mri/mask.mgz --aseg_name /output/073s4393/mri/aseg.auto_noCCseg.mgz --sid 073s4393 --seg_log /output/073s4393/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s4393/ses-201211281030330/anat/sub-073s4393_ses-201211281030330_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s4393/ses-201211281030330/anat/sub-073s4393_ses-201211281030330_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s4393/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4393/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.08 0.01] [ 0.09 1.05 0.03] [-0.01 -0.03 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 3094.124755859375\r\n",
      "rescale:  min: 0.0  max: 1345.9442138671875  scale: 0.18945807513621243\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4393/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.73batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s4393/ses-201211281030330/anat/sub-073s4393_ses-201211281030330_T1w.nii finished in 6.1372 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.48batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s4393/ses-201211281030330/anat/sub-073s4393_ses-201211281030330_T1w.nii finished in 6.3252 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s4393/ses-201211281030330/anat/sub-073s4393_ses-201211281030330_T1w.nii finished in 5.9831 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4393/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  22254\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4393/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4393/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.34\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s4393/mri/orig.mgz --rescale /output/073s4393/mri/orig_nu.mgz --aseg /output/073s4393/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s4393/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s4393/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s4393/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 56.08\r\n",
      "- m: 1.8187\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s4393/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s4393/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s4393/stats/aseg+DKT.stats --normfile /output/073s4393/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s4393 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s4393/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s4393/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.85 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:20:32 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-007s4387/ses-201301211307270/anat/sub-007s4387_ses-201301211307270_T1w.nii --asegdkt_segfile /output/007s4387/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/007s4387/mri/orig.mgz --brainmask_name /output/007s4387/mri/mask.mgz --aseg_name /output/007s4387/mri/aseg.auto_noCCseg.mgz --sid 007s4387 --seg_log /output/007s4387/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-007s4387/ses-201301211307270/anat/sub-007s4387_ses-201301211307270_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-007s4387/ses-201301211307270/anat/sub-007s4387_ses-201301211307270_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/007s4387/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4387/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.04 0.03] [ 0.04 1.03 0.23] [-0.04 -0.23 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 8303.0\r\n",
      "rescale:  min: 0.0  max: 5604.52490234375  scale: 0.04549895030234621\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4387/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.27batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-007s4387/ses-201301211307270/anat/sub-007s4387_ses-201301211307270_T1w.nii finished in 6.0589 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.06batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-007s4387/ses-201301211307270/anat/sub-007s4387_ses-201301211307270_T1w.nii finished in 6.2354 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.42batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-007s4387/ses-201301211307270/anat/sub-007s4387_ses-201301211307270_T1w.nii finished in 5.8970 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4387/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17245\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4387/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4387/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.01\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/007s4387/mri/orig.mgz --rescale /output/007s4387/mri/orig_nu.mgz --aseg /output/007s4387/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/007s4387/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/007s4387/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/007s4387/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 50.06\r\n",
      "- m: 2.0376\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/007s4387/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/007s4387/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/007s4387/stats/aseg+DKT.stats --normfile /output/007s4387/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 007s4387 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/007s4387/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/007s4387/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.80 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:23:41 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-009s4359/ses-201302131057010/anat/sub-009s4359_ses-201302131057010_T1w.nii --asegdkt_segfile /output/009s4359/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/009s4359/mri/orig.mgz --brainmask_name /output/009s4359/mri/mask.mgz --aseg_name /output/009s4359/mri/aseg.auto_noCCseg.mgz --sid 009s4359 --seg_log /output/009s4359/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-009s4359/ses-201302131057010/anat/sub-009s4359_ses-201302131057010_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-009s4359/ses-201302131057010/anat/sub-009s4359_ses-201302131057010_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/009s4359/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4359/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.05 -0.04] [ 0.07 1.03 0.24] [ 0.03 -0.24 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1808.2010498046875\r\n",
      "rescale:  min: 0.0  max: 1028.866455078125  scale: 0.24784557679124358\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4359/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.27batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-009s4359/ses-201302131057010/anat/sub-009s4359_ses-201302131057010_T1w.nii finished in 6.0581 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.28batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-009s4359/ses-201302131057010/anat/sub-009s4359_ses-201302131057010_T1w.nii finished in 6.2031 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.29batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-009s4359/ses-201302131057010/anat/sub-009s4359_ses-201302131057010_T1w.nii finished in 5.9145 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4359/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17458\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4359/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4359/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.06\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/009s4359/mri/orig.mgz --rescale /output/009s4359/mri/orig_nu.mgz --aseg /output/009s4359/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/009s4359/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/009s4359/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/009s4359/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 62.09\r\n",
      "- m: 1.6429\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/009s4359/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/009s4359/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/009s4359/stats/aseg+DKT.stats --normfile /output/009s4359/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 009s4359 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/009s4359/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/009s4359/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.84 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:26:49 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-141s4423/ses-201112201416050/anat/sub-141s4423_ses-201112201416050_T1w.nii --asegdkt_segfile /output/141s4423/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/141s4423/mri/orig.mgz --brainmask_name /output/141s4423/mri/mask.mgz --aseg_name /output/141s4423/mri/aseg.auto_noCCseg.mgz --sid 141s4423 --seg_log /output/141s4423/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-141s4423/ses-201112201416050/anat/sub-141s4423_ses-201112201416050_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-141s4423/ses-201112201416050/anat/sub-141s4423_ses-201112201416050_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/141s4423/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4423/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 0.01] [ 0.02 1.01 0.32] [-0.02 -0.32 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1440.89990234375\r\n",
      "rescale:  min: 0.0  max: 821.3129272460938  scale: 0.31047849308183745\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4423/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.34batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-141s4423/ses-201112201416050/anat/sub-141s4423_ses-201112201416050_T1w.nii finished in 6.0495 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.10batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-141s4423/ses-201112201416050/anat/sub-141s4423_ses-201112201416050_T1w.nii finished in 6.2294 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.35batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-141s4423/ses-201112201416050/anat/sub-141s4423_ses-201112201416050_T1w.nii finished in 5.9068 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4423/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  22313\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4423/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4423/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.24\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/141s4423/mri/orig.mgz --rescale /output/141s4423/mri/orig_nu.mgz --aseg /output/141s4423/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/141s4423/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/141s4423/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/141s4423/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 73.85\r\n",
      "- m: 1.3811\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/141s4423/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/141s4423/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/141s4423/stats/aseg+DKT.stats --normfile /output/141s4423/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 141s4423 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/141s4423/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/141s4423/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.60 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:29:58 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-068s4424/ses-201112231355220/anat/sub-068s4424_ses-201112231355220_T1w.nii --asegdkt_segfile /output/068s4424/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/068s4424/mri/orig.mgz --brainmask_name /output/068s4424/mri/mask.mgz --aseg_name /output/068s4424/mri/aseg.auto_noCCseg.mgz --sid 068s4424 --seg_log /output/068s4424/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-068s4424/ses-201112231355220/anat/sub-068s4424_ses-201112231355220_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-068s4424/ses-201112231355220/anat/sub-068s4424_ses-201112231355220_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/068s4424/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4424/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.06 -0.02] [-0.05 1.01 0.3 ] [ 0.04 -0.3 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2051.19580078125\r\n",
      "rescale:  min: 0.0  max: 976.3692016601562  scale: 0.2611716956725122\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4424/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.26batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-068s4424/ses-201112231355220/anat/sub-068s4424_ses-201112231355220_T1w.nii finished in 6.0601 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.87batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-068s4424/ses-201112231355220/anat/sub-068s4424_ses-201112231355220_T1w.nii finished in 6.2653 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.36batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-068s4424/ses-201112231355220/anat/sub-068s4424_ses-201112231355220_T1w.nii finished in 5.9044 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4424/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18009\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4424/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4424/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.95\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/068s4424/mri/orig.mgz --rescale /output/068s4424/mri/orig_nu.mgz --aseg /output/068s4424/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/068s4424/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/068s4424/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/068s4424/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 78.01\r\n",
      "- m: 1.3075\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/068s4424/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/068s4424/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/068s4424/stats/aseg+DKT.stats --normfile /output/068s4424/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 068s4424 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/068s4424/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/068s4424/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.42 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:33:04 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-141s4456/ses-201201170945340/anat/sub-141s4456_ses-201201170945340_T1w.nii --asegdkt_segfile /output/141s4456/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/141s4456/mri/orig.mgz --brainmask_name /output/141s4456/mri/mask.mgz --aseg_name /output/141s4456/mri/aseg.auto_noCCseg.mgz --sid 141s4456 --seg_log /output/141s4456/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-141s4456/ses-201201170945340/anat/sub-141s4456_ses-201201170945340_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-141s4456/ses-201201170945340/anat/sub-141s4456_ses-201201170945340_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/141s4456/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4456/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.05 -0.04] [-0.05 1.04 0.15] [ 0.06 -0.14 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1629.239013671875\r\n",
      "rescale:  min: 0.0  max: 907.4861450195312  scale: 0.2809960255586181\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4456/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.31batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-141s4456/ses-201201170945340/anat/sub-141s4456_ses-201201170945340_T1w.nii finished in 6.0531 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.89batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-141s4456/ses-201201170945340/anat/sub-141s4456_ses-201201170945340_T1w.nii finished in 6.2608 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.39batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-141s4456/ses-201201170945340/anat/sub-141s4456_ses-201201170945340_T1w.nii finished in 5.9010 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4456/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20883\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4456/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/141s4456/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.16\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/141s4456/mri/orig.mgz --rescale /output/141s4456/mri/orig_nu.mgz --aseg /output/141s4456/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/141s4456/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/141s4456/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/141s4456/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 64.60\r\n",
      "- m: 1.5789\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/141s4456/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/141s4456/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/141s4456/stats/aseg+DKT.stats --normfile /output/141s4456/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 141s4456 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/141s4456/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/141s4456/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.42 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:36:12 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-116s4453/ses-201301161110340/anat/sub-116s4453_ses-201301161110340_T1w.nii --asegdkt_segfile /output/116s4453/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/116s4453/mri/orig.mgz --brainmask_name /output/116s4453/mri/mask.mgz --aseg_name /output/116s4453/mri/aseg.auto_noCCseg.mgz --sid 116s4453 --seg_log /output/116s4453/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-116s4453/ses-201301161110340/anat/sub-116s4453_ses-201301161110340_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-116s4453/ses-201301161110340/anat/sub-116s4453_ses-201301161110340_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/116s4453/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4453/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 -0.06] [ 0.03 1.03 0.23] [ 0.07 -0.23 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1228.59033203125\r\n",
      "rescale:  min: 0.0  max: 887.042236328125  scale: 0.2874722189729793\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4453/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.38batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-116s4453/ses-201301161110340/anat/sub-116s4453_ses-201301161110340_T1w.nii finished in 6.0431 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.61batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-116s4453/ses-201301161110340/anat/sub-116s4453_ses-201301161110340_T1w.nii finished in 6.3042 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.14batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-116s4453/ses-201301161110340/anat/sub-116s4453_ses-201301161110340_T1w.nii finished in 5.9350 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4453/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20068\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4453/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4453/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.13\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/116s4453/mri/orig.mgz --rescale /output/116s4453/mri/orig_nu.mgz --aseg /output/116s4453/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/116s4453/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/116s4453/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/116s4453/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 68.77\r\n",
      "- m: 1.4833\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/116s4453/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/116s4453/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/116s4453/stats/aseg+DKT.stats --normfile /output/116s4453/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 116s4453 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/116s4453/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/116s4453/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.63 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:39:20 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-137s4466/ses-201303011445310/anat/sub-137s4466_ses-201303011445310_T1w.nii --asegdkt_segfile /output/137s4466/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/137s4466/mri/orig.mgz --brainmask_name /output/137s4466/mri/mask.mgz --aseg_name /output/137s4466/mri/aseg.auto_noCCseg.mgz --sid 137s4466 --seg_log /output/137s4466/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-137s4466/ses-201303011445310/anat/sub-137s4466_ses-201303011445310_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-137s4466/ses-201303011445310/anat/sub-137s4466_ses-201303011445310_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/137s4466/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4466/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 -0.01] [ 0.01 1. 0.33] [ 0.01 -0.33 1. ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 881.8307495117188\r\n",
      "rescale:  min: 0.0  max: 532.6257934570312  scale: 0.4787601410455758\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4466/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.31batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-137s4466/ses-201303011445310/anat/sub-137s4466_ses-201303011445310_T1w.nii finished in 6.0529 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.05batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-137s4466/ses-201303011445310/anat/sub-137s4466_ses-201303011445310_T1w.nii finished in 6.2373 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.41batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-137s4466/ses-201303011445310/anat/sub-137s4466_ses-201303011445310_T1w.nii finished in 5.8979 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4466/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20632\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4466/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4466/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/137s4466/mri/orig.mgz --rescale /output/137s4466/mri/orig_nu.mgz --aseg /output/137s4466/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/137s4466/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/137s4466/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/137s4466/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 69.32\r\n",
      "- m: 1.4714\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/137s4466/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/137s4466/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/137s4466/stats/aseg+DKT.stats --normfile /output/137s4466/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 137s4466 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/137s4466/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/137s4466/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.29 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:42:27 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-126s4507/ses-201202031005260/anat/sub-126s4507_ses-201202031005260_T1w.nii --asegdkt_segfile /output/126s4507/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/126s4507/mri/orig.mgz --brainmask_name /output/126s4507/mri/mask.mgz --aseg_name /output/126s4507/mri/aseg.auto_noCCseg.mgz --sid 126s4507 --seg_log /output/126s4507/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-126s4507/ses-201202031005260/anat/sub-126s4507_ses-201202031005260_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-126s4507/ses-201202031005260/anat/sub-126s4507_ses-201202031005260_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/126s4507/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4507/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.08 -0.04] [ 0.11 0.97 0.29] [ 0.01 -0.29 0.97]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 17251.837890625\r\n",
      "rescale:  min: 0.0  max: 8953.7041015625  scale: 0.028479833274309373\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4507/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.30batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-126s4507/ses-201202031005260/anat/sub-126s4507_ses-201202031005260_T1w.nii finished in 6.0548 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-126s4507/ses-201202031005260/anat/sub-126s4507_ses-201202031005260_T1w.nii finished in 6.2576 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.49batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-126s4507/ses-201202031005260/anat/sub-126s4507_ses-201202031005260_T1w.nii finished in 5.8872 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4507/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20737\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4507/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4507/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.21\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/126s4507/mri/orig.mgz --rescale /output/126s4507/mri/orig_nu.mgz --aseg /output/126s4507/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/126s4507/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/126s4507/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/126s4507/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 49.54\r\n",
      "- m: 2.0589\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/126s4507/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/126s4507/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/126s4507/stats/aseg+DKT.stats --normfile /output/126s4507/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 126s4507 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/126s4507/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/126s4507/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.78 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:45:35 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-007s4516/ses-201303131230280/anat/sub-007s4516_ses-201303131230280_T1w.nii --asegdkt_segfile /output/007s4516/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/007s4516/mri/orig.mgz --brainmask_name /output/007s4516/mri/mask.mgz --aseg_name /output/007s4516/mri/aseg.auto_noCCseg.mgz --sid 007s4516 --seg_log /output/007s4516/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-007s4516/ses-201303131230280/anat/sub-007s4516_ses-201303131230280_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-007s4516/ses-201303131230280/anat/sub-007s4516_ses-201303131230280_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/007s4516/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4516/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.20e+00 1.04e-03 -6.44e-02] [ 8.90e-03 1.04e+00 1.45e-01] [ 7.28e-02 -1.45e-01 1.04e+00]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 13012.404296875\r\n",
      "rescale:  min: 0.0  max: 6831.51220703125  scale: 0.03732702105655968\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4516/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.27batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-007s4516/ses-201303131230280/anat/sub-007s4516_ses-201303131230280_T1w.nii finished in 6.0583 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-007s4516/ses-201303131230280/anat/sub-007s4516_ses-201303131230280_T1w.nii finished in 6.2457 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.50batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-007s4516/ses-201303131230280/anat/sub-007s4516_ses-201303131230280_T1w.nii finished in 5.8856 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4516/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19693\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4516/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4516/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.13\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/007s4516/mri/orig.mgz --rescale /output/007s4516/mri/orig_nu.mgz --aseg /output/007s4516/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/007s4516/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/007s4516/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/007s4516/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 54.03\r\n",
      "- m: 1.8878\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/007s4516/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/007s4516/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/007s4516/stats/aseg+DKT.stats --normfile /output/007s4516/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 007s4516 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/007s4516/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/007s4516/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.76 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:48:44 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-029s4385/ses-201303051514580/anat/sub-029s4385_ses-201303051514580_T1w.nii --asegdkt_segfile /output/029s4385/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/029s4385/mri/orig.mgz --brainmask_name /output/029s4385/mri/mask.mgz --aseg_name /output/029s4385/mri/aseg.auto_noCCseg.mgz --sid 029s4385 --seg_log /output/029s4385/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-029s4385/ses-201303051514580/anat/sub-029s4385_ses-201303051514580_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-029s4385/ses-201303051514580/anat/sub-029s4385_ses-201303051514580_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/029s4385/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4385/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 0.05] [ 0. 1. 0.19] [-0.06 -0.19 1. ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 7332.5654296875\r\n",
      "rescale:  min: 0.0  max: 5081.4677734375  scale: 0.05018235111771616\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4385/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.96batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-029s4385/ses-201303051514580/anat/sub-029s4385_ses-201303051514580_T1w.nii finished in 6.1035 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.94batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-029s4385/ses-201303051514580/anat/sub-029s4385_ses-201303051514580_T1w.nii finished in 6.2543 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-029s4385/ses-201303051514580/anat/sub-029s4385_ses-201303051514580_T1w.nii finished in 5.9618 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4385/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20332\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4385/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/029s4385/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.99\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/029s4385/mri/orig.mgz --rescale /output/029s4385/mri/orig_nu.mgz --aseg /output/029s4385/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/029s4385/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/029s4385/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/029s4385/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 51.66\r\n",
      "- m: 1.9746\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/029s4385/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/029s4385/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/029s4385/stats/aseg+DKT.stats --normfile /output/029s4385/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 029s4385 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/029s4385/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/029s4385/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.15 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:51:52 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-128s4571/ses-201306191711580/anat/sub-128s4571_ses-201306191711580_T1w.nii --asegdkt_segfile /output/128s4571/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/128s4571/mri/orig.mgz --brainmask_name /output/128s4571/mri/mask.mgz --aseg_name /output/128s4571/mri/aseg.auto_noCCseg.mgz --sid 128s4571 --seg_log /output/128s4571/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-128s4571/ses-201306191711580/anat/sub-128s4571_ses-201306191711580_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-128s4571/ses-201306191711580/anat/sub-128s4571_ses-201306191711580_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/128s4571/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4571/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0x1.0\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.01 -0.01] [-0.01 0.98 0.21] [ 0.02 -0.2 0.98]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2893.417724609375\r\n",
      "rescale:  min: 0.0  max: 1105.2855224609375  scale: 0.2307096173957278\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4571/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-128s4571/ses-201306191711580/anat/sub-128s4571_ses-201306191711580_T1w.nii finished in 6.1232 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.51batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-128s4571/ses-201306191711580/anat/sub-128s4571_ses-201306191711580_T1w.nii finished in 6.3199 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.80batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-128s4571/ses-201306191711580/anat/sub-128s4571_ses-201306191711580_T1w.nii finished in 5.9822 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4571/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  23385\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4571/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 1 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4571/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.27\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/128s4571/mri/orig.mgz --rescale /output/128s4571/mri/orig_nu.mgz --aseg /output/128s4571/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/128s4571/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/128s4571/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/128s4571/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 45.00\r\n",
      "- m: 2.2665\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/128s4571/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/128s4571/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/128s4571/stats/aseg+DKT.stats --normfile /output/128s4571/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 128s4571 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/128s4571/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/128s4571/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.19 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:55:02 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-109s4531/ses-201303251311530/anat/sub-109s4531_ses-201303251311530_T1w.nii --asegdkt_segfile /output/109s4531/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/109s4531/mri/orig.mgz --brainmask_name /output/109s4531/mri/mask.mgz --aseg_name /output/109s4531/mri/aseg.auto_noCCseg.mgz --sid 109s4531 --seg_log /output/109s4531/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-109s4531/ses-201303251311530/anat/sub-109s4531_ses-201303251311530_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-109s4531/ses-201303251311530/anat/sub-109s4531_ses-201303251311530_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/109s4531/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/109s4531/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.12 -0.08] [ 0.15 1.03 0.16] [ 0.07 -0.17 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 9954.927734375\r\n",
      "rescale:  min: 0.0  max: 6729.53125  scale: 0.03789268383291927\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/109s4531/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.94batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-109s4531/ses-201303251311530/anat/sub-109s4531_ses-201303251311530_T1w.nii finished in 6.1063 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.19batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-109s4531/ses-201303251311530/anat/sub-109s4531_ses-201303251311530_T1w.nii finished in 6.3710 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.88batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-109s4531/ses-201303251311530/anat/sub-109s4531_ses-201303251311530_T1w.nii finished in 5.9704 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/109s4531/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18139\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/109s4531/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/109s4531/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.12\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/109s4531/mri/orig.mgz --rescale /output/109s4531/mri/orig_nu.mgz --aseg /output/109s4531/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/109s4531/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/109s4531/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/109s4531/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 49.11\r\n",
      "- m: 2.0770\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/109s4531/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/109s4531/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/109s4531/stats/aseg+DKT.stats --normfile /output/109s4531/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 109s4531 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/109s4531/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/109s4531/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.22 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 14:58:11 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-007s4620/ses-201304151230160/anat/sub-007s4620_ses-201304151230160_T1w.nii --asegdkt_segfile /output/007s4620/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/007s4620/mri/orig.mgz --brainmask_name /output/007s4620/mri/mask.mgz --aseg_name /output/007s4620/mri/aseg.auto_noCCseg.mgz --sid 007s4620 --seg_log /output/007s4620/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-007s4620/ses-201304151230160/anat/sub-007s4620_ses-201304151230160_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-007s4620/ses-201304151230160/anat/sub-007s4620_ses-201304151230160_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/007s4620/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4620/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.20e+00 -5.76e-04 -1.63e-02] [ 3.18e-03 1.04e+00 1.44e-01] [ 1.83e-02 -1.44e-01 1.04e+00]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 19259.978515625\r\n",
      "rescale:  min: 0.0  max: 7126.19189453125  scale: 0.03578348769918629\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4620/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.78batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-007s4620/ses-201304151230160/anat/sub-007s4620_ses-201304151230160_T1w.nii finished in 6.1303 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.13batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-007s4620/ses-201304151230160/anat/sub-007s4620_ses-201304151230160_T1w.nii finished in 6.3795 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.96batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-007s4620/ses-201304151230160/anat/sub-007s4620_ses-201304151230160_T1w.nii finished in 5.9604 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4620/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20508\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4620/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4620/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.04\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/007s4620/mri/orig.mgz --rescale /output/007s4620/mri/orig_nu.mgz --aseg /output/007s4620/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/007s4620/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/007s4620/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/007s4620/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 53.35\r\n",
      "- m: 1.9120\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/007s4620/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/007s4620/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/007s4620/stats/aseg+DKT.stats --normfile /output/007s4620/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 007s4620 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/007s4620/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/007s4620/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.94 seconds using up to 1 threads.\r\n"
     ]
    }
   ],
   "execution_count": 247
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T07:26:21.219020Z",
     "start_time": "2025-01-07T02:28:33.212245Z"
    }
   },
   "cell_type": "code",
   "source": [
    "!docker run --gpus all -v /media/jfallmann/T9/University/master_thesis/dataset/mri/bids:/data -v /media/jfallmann/T9/University/master_thesis/dataset/mri/processed:/output -v /media/jfallmann/T9/University/master_thesis/license:/fs_license --entrypoint \"/fastsurfer/brun_fastsurfer.sh\" --rm --user $(id -u):$(id -g) deepmi/fastsurfer:latest --fs_license /fs_license/license.txt --subject_list /data/fastsurfer_batch2.txt --sd /output --parallel --3T --seg_only --no_cereb --no_hypothal"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/fastsurfer/brun_fastsurfer.sh --fs_license /fs_license/license.txt --subject_list /data/fastsurfer_batch2.txt --sd /output --parallel --3T --seg_only --no_cereb --no_hypothal\r\n",
      "Tue, 07 Jan 2025 02:28:33 +0000\r\n",
      "\r\n",
      "INFO: run_fastsurfer not explicitly specified, using $FASTSURFER_HOME/run_fastsurfer.sh.\r\n",
      "072s4610=/data/sub-072s4610/ses-201304090730570/anat/sub-072s4610_ses-201304090730570_T1w.nii 099s4104=/data/sub-099s4104/ses-201207110807380/anat/sub-099s4104_ses-201207110807380_T1w.nii 128s4609=/data/sub-128s4609/ses-201204041828370/anat/sub-128s4609_ses-201204041828370_T1w.nii 041s4629=/data/sub-041s4629/ses-201304221158430/anat/sub-041s4629_ses-201304221158430_T1w.nii 070s4692=/data/sub-070s4692/ses-201205081035540/anat/sub-070s4692_ses-201205081035540_T1w.nii 037s4030=/data/sub-037s4030/ses-201205141108430/anat/sub-037s4030_ses-201205141108430_T1w.nii 014s4080=/data/sub-014s4080/ses-201206211033430/anat/sub-014s4080_ses-201206211033430_T1w.nii 016s4009=/data/sub-016s4009/ses-201207180940140/anat/sub-016s4009_ses-201207180940140_T1w.nii 941s4036=/data/sub-941s4036/ses-201208171318460/anat/sub-941s4036_ses-201208171318460_T1w.nii 116s4167=/data/sub-116s4167/ses-201208291116580/anat/sub-116s4167_ses-201208291116580_T1w.nii 127s4148=/data/sub-127s4148/ses-201208300908040/anat/sub-127s4148_ses-201208300908040_T1w.nii 072s2083=/data/sub-072s2083/ses-201209111026390/anat/sub-072s2083_ses-201209111026390_T1w.nii 123s2055=/data/sub-123s2055/ses-201209241436050/anat/sub-123s2055_ses-201209241436050_T1w.nii 005s4185=/data/sub-005s4185/ses-201209271137370/anat/sub-005s4185_ses-201209271137370_T1w.nii 128s2036=/data/sub-128s2036/ses-201210060910010/anat/sub-128s2036_ses-201210060910010_T1w.nii 073s0089=/data/sub-073s0089/ses-201210091114110/anat/sub-073s0089_ses-201210091114110_T1w.nii 099s2042=/data/sub-099s2042/ses-201208210808420/anat/sub-099s2042_ses-201208210808420_T1w.nii 024s4280=/data/sub-024s4280/ses-201211051025130/anat/sub-024s4280_ses-201211051025130_T1w.nii 128s2045=/data/sub-128s2045/ses-201210181449010/anat/sub-128s2045_ses-201210181449010_T1w.nii 067s2196=/data/sub-067s2196/ses-201211281153460/anat/sub-067s2196_ses-201211281153460_T1w.nii 014s4328=/data/sub-014s4328/ses-201211081101260/anat/sub-014s4328_ses-201211081101260_T1w.nii 021s4276=/data/sub-021s4276/ses-201210221004540/anat/sub-021s4276_ses-201210221004540_T1w.nii 067s2195=/data/sub-067s2195/ses-201212051217390/anat/sub-067s2195_ses-201212051217390_T1w.nii 035s2074=/data/sub-035s2074/ses-201211271413050/anat/sub-035s2074_ses-201211271413050_T1w.nii 094s2201=/data/sub-094s2201/ses-201301151022330/anat/sub-094s2201_ses-201301151022330_T1w.nii 072s4445=/data/sub-072s4445/ses-201301240955020/anat/sub-072s4445_ses-201301240955020_T1w.nii 037s4381=/data/sub-037s4381/ses-201301251132420/anat/sub-037s4381_ses-201301251132420_T1w.nii 021s4419=/data/sub-021s4419/ses-201212171032370/anat/sub-021s4419_ses-201212171032370_T1w.nii 128s2151=/data/sub-128s2151/ses-201302051739350/anat/sub-128s2151_ses-201302051739350_T1w.nii 011s2274=/data/sub-011s2274/ses-201302061432250/anat/sub-011s2274_ses-201302061432250_T1w.nii 022s4291=/data/sub-022s4291/ses-201301301429580/anat/sub-022s4291_ses-201301301429580_T1w.nii 023s4448=/data/sub-023s4448/ses-201302281128010/anat/sub-023s4448_ses-201302281128010_T1w.nii 099s4565=/data/sub-099s4565/ses-201302261036210/anat/sub-099s4565_ses-201302261036210_T1w.nii 073s4614=/data/sub-073s4614/ses-201303120850540/anat/sub-073s4614_ses-201303120850540_T1w.nii 035s4464=/data/sub-035s4464/ses-201302211415500/anat/sub-035s4464_ses-201302211415500_T1w.nii 072s4007=/data/sub-072s4007/ses-201203221108250/anat/sub-072s4007_ses-201203221108250_T1w.nii 014s4576=/data/sub-014s4576/ses-201304041124060/anat/sub-014s4576_ses-201304041124060_T1w.nii 073s2225=/data/sub-073s2225/ses-201304291337190/anat/sub-073s2225_ses-201304291337190_T1w.nii 067s2301=/data/sub-067s2301/ses-201303151036450/anat/sub-067s2301_ses-201303151036450_T1w.nii 073s2153=/data/sub-073s2153/ses-201304011234170/anat/sub-073s2153_ses-201304011234170_T1w.nii 094s4560=/data/sub-094s4560/ses-201303211430590/anat/sub-094s4560_ses-201303211430590_T1w.nii 007s4568=/data/sub-007s4568/ses-201305131446180/anat/sub-007s4568_ses-201305131446180_T1w.nii 022s2379=/data/sub-022s2379/ses-201305311149020/anat/sub-022s2379_ses-201305311149020_T1w.nii 135s4598=/data/sub-135s4598/ses-201303281045290/anat/sub-135s4598_ses-201303281045290_T1w.nii 114s2392=/data/sub-114s2392/ses-201308081147530/anat/sub-114s2392_ses-201308081147530_T1w.nii 082s2307=/data/sub-082s2307/ses-201306271218100/anat/sub-082s2307_ses-201306271218100_T1w.nii 073s2182=/data/sub-073s2182/ses-201312041652010/anat/sub-073s2182_ses-201312041652010_T1w.nii 068s4174=/data/sub-068s4174/ses-201210291153430/anat/sub-068s4174_ses-201210291153430_T1w.nii 116s4010=/data/sub-116s4010/ses-201204181119380/anat/sub-116s4010_ses-201204181119380_T1w.nii 023s4034=/data/sub-023s4034/ses-201207261314580/anat/sub-023s4034_ses-201207261314580_T1w.nii 014s4079=/data/sub-014s4079/ses-201206211135200/anat/sub-014s4079_ses-201206211135200_T1w.nii 153s4159=/data/sub-153s4159/ses-201208281128460/anat/sub-153s4159_ses-201208281128460_T1w.nii 033s4179=/data/sub-033s4179/ses-201209051633410/anat/sub-033s4179_ses-201209051633410_T1w.nii 033s4176=/data/sub-033s4176/ses-201209071405590/anat/sub-033s4176_ses-201209071405590_T1w.nii 073s4216=/data/sub-073s4216/ses-201209170922460/anat/sub-073s4216_ses-201209170922460_T1w.nii 003s4152=/data/sub-003s4152/ses-201108300937460/anat/sub-003s4152_ses-201108300937460_T1w.nii 127s4210=/data/sub-127s4210/ses-201210011338490/anat/sub-127s4210_ses-201210011338490_T1w.nii 021s4254=/data/sub-021s4254/ses-201210031003160/anat/sub-021s4254_ses-201210031003160_T1w.nii 098s4201=/data/sub-098s4201/ses-201109141625250/anat/sub-098s4201_ses-201109141625250_T1w.nii 129s4287=/data/sub-129s4287/ses-201110201035370/anat/sub-129s4287_ses-201110201035370_T1w.nii 094s4282=/data/sub-094s4282/ses-201110200929470/anat/sub-094s4282_ses-201110200929470_T1w.nii 116s4338=/data/sub-116s4338/ses-201211071317040/anat/sub-116s4338_ses-201211071317040_T1w.nii 073s4360=/data/sub-073s4360/ses-201302251204230/anat/sub-073s4360_ses-201302251204230_T1w.nii 129s4369=/data/sub-129s4369/ses-201111170919350/anat/sub-129s4369_ses-201111170919350_T1w.nii 003s4350=/data/sub-003s4350/ses-201212061620080/anat/sub-003s4350_ses-201212061620080_T1w.nii 035s4256=/data/sub-035s4256/ses-201301141553360/anat/sub-035s4256_ses-201301141553360_T1w.nii 032s4429=/data/sub-032s4429/ses-201302071257230/anat/sub-032s4429_ses-201302071257230_T1w.nii 037s4410=/data/sub-037s4410/ses-201212191007000/anat/sub-037s4410_ses-201212191007000_T1w.nii 099s4480=/data/sub-099s4480/ses-201302260748180/anat/sub-099s4480_ses-201302260748180_T1w.nii 098s4506=/data/sub-098s4506/ses-201202241316090/anat/sub-098s4506_ses-201202241316090_T1w.nii 014s4615=/data/sub-014s4615/ses-201304111133030/anat/sub-014s4615_ses-201304111133030_T1w.nii 127s4645=/data/sub-127s4645/ses-201204041407240/anat/sub-127s4645_ses-201204041407240_T1w.nii 137s4672=/data/sub-137s4672/ses-201204160838030/anat/sub-137s4672_ses-201204160838030_T1w.nii 126s4686=/data/sub-126s4686/ses-201204271102150/anat/sub-126s4686_ses-201204271102150_T1w.nii 137s0668=/data/sub-137s0668/ses-201208221249470/anat/sub-137s0668_ses-201208221249470_T1w.nii 037s0566=/data/sub-037s0566/ses-201212061253070/anat/sub-037s0566_ses-201212061253070_T1w.nii 016s4591=/data/sub-016s4591/ses-201304011455320/anat/sub-016s4591_ses-201304011455320_T1w.nii 036s4562=/data/sub-036s4562/ses-201304041541000/anat/sub-036s4562_ses-201304041541000_T1w.nii 021s4659=/data/sub-021s4659/ses-201305090858240/anat/sub-021s4659_ses-201305090858240_T1w.nii 014s4668=/data/sub-014s4668/ses-201305301058270/anat/sub-014s4668_ses-201305301058270_T1w.nii 068s4431=/data/sub-068s4431/ses-201202061525130/anat/sub-068s4431_ses-201202061525130_T1w.nii 127s4604=/data/sub-127s4604/ses-201203211543180/anat/sub-127s4604_ses-201203211543180_T1w.nii 128s4607=/data/sub-128s4607/ses-201304121516540/anat/sub-128s4607_ses-201304121516540_T1w.nii 009s4543=/data/sub-009s4543/ses-201303201410350/anat/sub-009s4543_ses-201303201410350_T1w.nii 052s4626=/data/sub-052s4626/ses-201204031330200/anat/sub-052s4626_ses-201204031330200_T1w.nii 128s4603=/data/sub-128s4603/ses-201204070856120/anat/sub-128s4603_ses-201204070856120_T1w.nii 128s4653=/data/sub-128s4653/ses-201305311807210/anat/sub-128s4653_ses-201305311807210_T1w.nii 941s1202=/data/sub-941s1202/ses-201207021018000/anat/sub-941s1202_ses-201207021018000_T1w.nii 137s0722=/data/sub-137s0722/ses-201208311552330/anat/sub-137s0722_ses-201208311552330_T1w.nii 023s2068=/data/sub-023s2068/ses-201210171218320/anat/sub-023s2068_ses-201210171218320_T1w.nii 037s1078=/data/sub-037s1078/ses-201211281227570/anat/sub-037s1078_ses-201211281227570_T1w.nii 137s0994=/data/sub-137s0994/ses-201211141033240/anat/sub-137s0994_ses-201211141033240_T1w.nii 941s1195=/data/sub-941s1195/ses-201303071323450/anat/sub-941s1195_ses-201303071323450_T1w.nii 073s0746=/data/sub-073s0746/ses-201303081132280/anat/sub-073s0746_ses-201303081132280_T1w.nii 131s0123=/data/sub-131s0123/ses-201303211338380/anat/sub-131s0123_ses-201303211338380_T1w.nii\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:28:33 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-072s4610/ses-201304090730570/anat/sub-072s4610_ses-201304090730570_T1w.nii --asegdkt_segfile /output/072s4610/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/072s4610/mri/orig.mgz --brainmask_name /output/072s4610/mri/mask.mgz --aseg_name /output/072s4610/mri/aseg.auto_noCCseg.mgz --sid 072s4610 --seg_log /output/072s4610/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-072s4610/ses-201304090730570/anat/sub-072s4610_ses-201304090730570_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-072s4610/ses-201304090730570/anat/sub-072s4610_ses-201304090730570_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/072s4610/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4610/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 -0.07] [ 0.02 1.05 0.06] [ 0.08 -0.06 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2186.599609375\r\n",
      "rescale:  min: 0.0  max: 994.90283203125  scale: 0.2563064369606603\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4610/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.77batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-072s4610/ses-201304090730570/anat/sub-072s4610_ses-201304090730570_T1w.nii finished in 5.9875 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.28batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-072s4610/ses-201304090730570/anat/sub-072s4610_ses-201304090730570_T1w.nii finished in 6.0562 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.96batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-072s4610/ses-201304090730570/anat/sub-072s4610_ses-201304090730570_T1w.nii finished in 5.8243 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4610/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  15238\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4610/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4610/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.9\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/072s4610/mri/orig.mgz --rescale /output/072s4610/mri/orig_nu.mgz --aseg /output/072s4610/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/072s4610/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/072s4610/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/072s4610/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 49.23\r\n",
      "- m: 2.0720\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/072s4610/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/072s4610/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/072s4610/stats/aseg+DKT.stats --normfile /output/072s4610/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 072s4610 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/072s4610/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/072s4610/stats/aseg+DKT.stats.\r\n",
      "Calculation took 8.88 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:31:41 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-099s4104/ses-201207110807380/anat/sub-099s4104_ses-201207110807380_T1w.nii --asegdkt_segfile /output/099s4104/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/099s4104/mri/orig.mgz --brainmask_name /output/099s4104/mri/mask.mgz --aseg_name /output/099s4104/mri/aseg.auto_noCCseg.mgz --sid 099s4104 --seg_log /output/099s4104/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-099s4104/ses-201207110807380/anat/sub-099s4104_ses-201207110807380_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-099s4104/ses-201207110807380/anat/sub-099s4104_ses-201207110807380_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/099s4104/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4104/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.04 0. ] [ 0.04 1.02 0.25] [-0.01 -0.25 1.02]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1848.9176025390625\r\n",
      "rescale:  min: 0.0  max: 1155.573486328125  scale: 0.22066965278882555\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4104/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.83batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-099s4104/ses-201207110807380/anat/sub-099s4104_ses-201207110807380_T1w.nii finished in 5.9797 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.31batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-099s4104/ses-201207110807380/anat/sub-099s4104_ses-201207110807380_T1w.nii finished in 6.0518 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.91batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-099s4104/ses-201207110807380/anat/sub-099s4104_ses-201207110807380_T1w.nii finished in 5.8305 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4104/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20337\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4104/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 1 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4104/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.06\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/099s4104/mri/orig.mgz --rescale /output/099s4104/mri/orig_nu.mgz --aseg /output/099s4104/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/099s4104/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/099s4104/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/099s4104/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 64.87\r\n",
      "- m: 1.5724\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/099s4104/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/099s4104/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/099s4104/stats/aseg+DKT.stats --normfile /output/099s4104/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 099s4104 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/099s4104/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/099s4104/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.67 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:34:49 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-128s4609/ses-201204041828370/anat/sub-128s4609_ses-201204041828370_T1w.nii --asegdkt_segfile /output/128s4609/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/128s4609/mri/orig.mgz --brainmask_name /output/128s4609/mri/mask.mgz --aseg_name /output/128s4609/mri/aseg.auto_noCCseg.mgz --sid 128s4609 --seg_log /output/128s4609/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-128s4609/ses-201204041828370/anat/sub-128s4609_ses-201204041828370_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-128s4609/ses-201204041828370/anat/sub-128s4609_ses-201204041828370_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/128s4609/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4609/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.06 -0.08] [ 0.08 1.05 0.07] [ 0.09 -0.07 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2515.130615234375\r\n",
      "rescale:  min: 0.0  max: 1219.83837890625  scale: 0.20904408683111117\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4609/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.94batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-128s4609/ses-201204041828370/anat/sub-128s4609_ses-201204041828370_T1w.nii finished in 5.9644 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.45batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-128s4609/ses-201204041828370/anat/sub-128s4609_ses-201204041828370_T1w.nii finished in 6.0321 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-128s4609/ses-201204041828370/anat/sub-128s4609_ses-201204041828370_T1w.nii finished in 5.8282 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4609/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18417\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4609/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4609/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.01\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/128s4609/mri/orig.mgz --rescale /output/128s4609/mri/orig_nu.mgz --aseg /output/128s4609/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/128s4609/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/128s4609/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/128s4609/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 44.10\r\n",
      "- m: 2.3128\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/128s4609/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/128s4609/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/128s4609/stats/aseg+DKT.stats --normfile /output/128s4609/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 128s4609 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/128s4609/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/128s4609/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.82 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:37:56 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-041s4629/ses-201304221158430/anat/sub-041s4629_ses-201304221158430_T1w.nii --asegdkt_segfile /output/041s4629/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/041s4629/mri/orig.mgz --brainmask_name /output/041s4629/mri/mask.mgz --aseg_name /output/041s4629/mri/aseg.auto_noCCseg.mgz --sid 041s4629 --seg_log /output/041s4629/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-041s4629/ses-201304221158430/anat/sub-041s4629_ses-201304221158430_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-041s4629/ses-201304221158430/anat/sub-041s4629_ses-201304221158430_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/041s4629/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4629/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.01 0. ] [-0.01 1.05 0.04] [-0. -0.04 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2653.8046875\r\n",
      "rescale:  min: 0.0  max: 1042.9451904296875  scale: 0.24449990501892188\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4629/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-041s4629/ses-201304221158430/anat/sub-041s4629_ses-201304221158430_T1w.nii finished in 5.9629 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.14batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-041s4629/ses-201304221158430/anat/sub-041s4629_ses-201304221158430_T1w.nii finished in 6.0755 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.01batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-041s4629/ses-201304221158430/anat/sub-041s4629_ses-201304221158430_T1w.nii finished in 5.8177 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4629/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18490\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4629/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/041s4629/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.06\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/041s4629/mri/orig.mgz --rescale /output/041s4629/mri/orig_nu.mgz --aseg /output/041s4629/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/041s4629/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/041s4629/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/041s4629/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 50.84\r\n",
      "- m: 2.0065\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/041s4629/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/041s4629/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/041s4629/stats/aseg+DKT.stats --normfile /output/041s4629/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 041s4629 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/041s4629/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/041s4629/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.08 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:41:05 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-070s4692/ses-201205081035540/anat/sub-070s4692_ses-201205081035540_T1w.nii --asegdkt_segfile /output/070s4692/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/070s4692/mri/orig.mgz --brainmask_name /output/070s4692/mri/mask.mgz --aseg_name /output/070s4692/mri/aseg.auto_noCCseg.mgz --sid 070s4692 --seg_log /output/070s4692/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-070s4692/ses-201205081035540/anat/sub-070s4692_ses-201205081035540_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-070s4692/ses-201205081035540/anat/sub-070s4692_ses-201205081035540_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/070s4692/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/070s4692/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.07 -0.07] [ 0.09 1.02 0.24] [ 0.06 -0.25 1.02]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2685.487060546875\r\n",
      "rescale:  min: 0.0  max: 1127.904541015625  scale: 0.22608296245565648\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/070s4692/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-070s4692/ses-201205081035540/anat/sub-070s4692_ses-201205081035540_T1w.nii finished in 5.9573 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.42batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-070s4692/ses-201205081035540/anat/sub-070s4692_ses-201205081035540_T1w.nii finished in 6.0360 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-070s4692/ses-201205081035540/anat/sub-070s4692_ses-201205081035540_T1w.nii finished in 5.8197 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/070s4692/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21454\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/070s4692/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/070s4692/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.22\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/070s4692/mri/orig.mgz --rescale /output/070s4692/mri/orig_nu.mgz --aseg /output/070s4692/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/070s4692/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/070s4692/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/070s4692/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 54.03\r\n",
      "- m: 1.8879\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/070s4692/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/070s4692/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/070s4692/stats/aseg+DKT.stats --normfile /output/070s4692/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 070s4692 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/070s4692/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/070s4692/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.30 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:44:14 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-037s4030/ses-201205141108430/anat/sub-037s4030_ses-201205141108430_T1w.nii --asegdkt_segfile /output/037s4030/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/037s4030/mri/orig.mgz --brainmask_name /output/037s4030/mri/mask.mgz --aseg_name /output/037s4030/mri/aseg.auto_noCCseg.mgz --sid 037s4030 --seg_log /output/037s4030/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-037s4030/ses-201205141108430/anat/sub-037s4030_ses-201205141108430_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-037s4030/ses-201205141108430/anat/sub-037s4030_ses-201205141108430_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/037s4030/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4030/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 0.02] [ 0.03 1.04 0.19] [-0.03 -0.19 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1597.4573974609375\r\n",
      "rescale:  min: 0.0  max: 961.6693725585938  scale: 0.26516389860847217\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4030/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.12batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-037s4030/ses-201205141108430/anat/sub-037s4030_ses-201205141108430_T1w.nii finished in 5.9395 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.19batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-037s4030/ses-201205141108430/anat/sub-037s4030_ses-201205141108430_T1w.nii finished in 6.0683 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.02batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-037s4030/ses-201205141108430/anat/sub-037s4030_ses-201205141108430_T1w.nii finished in 5.8162 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4030/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19723\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4030/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4030/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.14\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/037s4030/mri/orig.mgz --rescale /output/037s4030/mri/orig_nu.mgz --aseg /output/037s4030/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/037s4030/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/037s4030/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/037s4030/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 65.80\r\n",
      "- m: 1.5501\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/037s4030/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/037s4030/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/037s4030/stats/aseg+DKT.stats --normfile /output/037s4030/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 037s4030 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/037s4030/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/037s4030/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.35 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:47:23 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-014s4080/ses-201206211033430/anat/sub-014s4080_ses-201206211033430_T1w.nii --asegdkt_segfile /output/014s4080/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/014s4080/mri/orig.mgz --brainmask_name /output/014s4080/mri/mask.mgz --aseg_name /output/014s4080/mri/aseg.auto_noCCseg.mgz --sid 014s4080 --seg_log /output/014s4080/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-014s4080/ses-201206211033430/anat/sub-014s4080_ses-201206211033430_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-014s4080/ses-201206211033430/anat/sub-014s4080_ses-201206211033430_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/014s4080/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4080/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.05 -0.03] [-0.05 1.03 0.23] [ 0.04 -0.22 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2542.625\r\n",
      "rescale:  min: 0.0  max: 1039.93359375  scale: 0.24520796475135506\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4080/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.00batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-014s4080/ses-201206211033430/anat/sub-014s4080_ses-201206211033430_T1w.nii finished in 5.9556 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-014s4080/ses-201206211033430/anat/sub-014s4080_ses-201206211033430_T1w.nii finished in 6.1056 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.16batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-014s4080/ses-201206211033430/anat/sub-014s4080_ses-201206211033430_T1w.nii finished in 5.7975 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4080/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16613\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4080/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4080/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.01\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/014s4080/mri/orig.mgz --rescale /output/014s4080/mri/orig_nu.mgz --aseg /output/014s4080/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/014s4080/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/014s4080/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/014s4080/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 44.34\r\n",
      "- m: 2.3004\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/014s4080/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/014s4080/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/014s4080/stats/aseg+DKT.stats --normfile /output/014s4080/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 014s4080 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/014s4080/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/014s4080/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.31 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:50:30 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-016s4009/ses-201207180940140/anat/sub-016s4009_ses-201207180940140_T1w.nii --asegdkt_segfile /output/016s4009/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/016s4009/mri/orig.mgz --brainmask_name /output/016s4009/mri/mask.mgz --aseg_name /output/016s4009/mri/aseg.auto_noCCseg.mgz --sid 016s4009 --seg_log /output/016s4009/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-016s4009/ses-201207180940140/anat/sub-016s4009_ses-201207180940140_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-016s4009/ses-201207180940140/anat/sub-016s4009_ses-201207180940140_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/016s4009/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4009/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.14 -0.04] [ 0.16 1. 0.09] [ 0.03 -0.09 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 10649.162109375\r\n",
      "rescale:  min: 0.0  max: 6943.25390625  scale: 0.03672629626441583\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4009/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.17batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-016s4009/ses-201207180940140/anat/sub-016s4009_ses-201207180940140_T1w.nii finished in 5.9317 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.00batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-016s4009/ses-201207180940140/anat/sub-016s4009_ses-201207180940140_T1w.nii finished in 6.0963 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.94batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-016s4009/ses-201207180940140/anat/sub-016s4009_ses-201207180940140_T1w.nii finished in 5.8265 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4009/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17812\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4009/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4009/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.94\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/016s4009/mri/orig.mgz --rescale /output/016s4009/mri/orig_nu.mgz --aseg /output/016s4009/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/016s4009/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/016s4009/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/016s4009/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 46.45\r\n",
      "- m: 2.1961\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/016s4009/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/016s4009/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/016s4009/stats/aseg+DKT.stats --normfile /output/016s4009/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 016s4009 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/016s4009/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/016s4009/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.84 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:53:38 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-941s4036/ses-201208171318460/anat/sub-941s4036_ses-201208171318460_T1w.nii --asegdkt_segfile /output/941s4036/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/941s4036/mri/orig.mgz --brainmask_name /output/941s4036/mri/mask.mgz --aseg_name /output/941s4036/mri/aseg.auto_noCCseg.mgz --sid 941s4036 --seg_log /output/941s4036/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-941s4036/ses-201208171318460/anat/sub-941s4036_ses-201208171318460_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-941s4036/ses-201208171318460/anat/sub-941s4036_ses-201208171318460_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/941s4036/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s4036/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 0.05] [ 0.01 1. 0.33] [-0.06 -0.33 1. ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1554.5533447265625\r\n",
      "rescale:  min: 0.0  max: 912.5228271484375  scale: 0.2794450641819614\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s4036/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-941s4036/ses-201208171318460/anat/sub-941s4036_ses-201208171318460_T1w.nii finished in 5.9673 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.85batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-941s4036/ses-201208171318460/anat/sub-941s4036_ses-201208171318460_T1w.nii finished in 6.1178 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.03batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-941s4036/ses-201208171318460/anat/sub-941s4036_ses-201208171318460_T1w.nii finished in 5.8155 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s4036/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21029\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s4036/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s4036/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.06\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/941s4036/mri/orig.mgz --rescale /output/941s4036/mri/orig_nu.mgz --aseg /output/941s4036/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/941s4036/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/941s4036/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/941s4036/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 62.76\r\n",
      "- m: 1.6252\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/941s4036/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/941s4036/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/941s4036/stats/aseg+DKT.stats --normfile /output/941s4036/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 941s4036 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/941s4036/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/941s4036/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.28 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:56:48 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-116s4167/ses-201208291116580/anat/sub-116s4167_ses-201208291116580_T1w.nii --asegdkt_segfile /output/116s4167/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/116s4167/mri/orig.mgz --brainmask_name /output/116s4167/mri/mask.mgz --aseg_name /output/116s4167/mri/aseg.auto_noCCseg.mgz --sid 116s4167 --seg_log /output/116s4167/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-116s4167/ses-201208291116580/anat/sub-116s4167_ses-201208291116580_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-116s4167/ses-201208291116580/anat/sub-116s4167_ses-201208291116580_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/116s4167/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4167/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.05 0.02] [ 0.05 1.05 0.11] [-0.03 -0.11 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 4335.46044921875\r\n",
      "rescale:  min: 0.0  max: 1244.277099609375  scale: 0.20493827305835172\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4167/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.98batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-116s4167/ses-201208291116580/anat/sub-116s4167_ses-201208291116580_T1w.nii finished in 5.9584 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.60batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-116s4167/ses-201208291116580/anat/sub-116s4167_ses-201208291116580_T1w.nii finished in 6.1552 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.08batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-116s4167/ses-201208291116580/anat/sub-116s4167_ses-201208291116580_T1w.nii finished in 5.8087 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4167/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18781\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4167/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4167/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.05\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/116s4167/mri/orig.mgz --rescale /output/116s4167/mri/orig_nu.mgz --aseg /output/116s4167/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/116s4167/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/116s4167/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/116s4167/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 45.39\r\n",
      "- m: 2.2470\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/116s4167/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/116s4167/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/116s4167/stats/aseg+DKT.stats --normfile /output/116s4167/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 116s4167 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/116s4167/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/116s4167/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.97 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 02:59:56 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-127s4148/ses-201208300908040/anat/sub-127s4148_ses-201208300908040_T1w.nii --asegdkt_segfile /output/127s4148/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/127s4148/mri/orig.mgz --brainmask_name /output/127s4148/mri/mask.mgz --aseg_name /output/127s4148/mri/aseg.auto_noCCseg.mgz --sid 127s4148 --seg_log /output/127s4148/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-127s4148/ses-201208300908040/anat/sub-127s4148_ses-201208300908040_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-127s4148/ses-201208300908040/anat/sub-127s4148_ses-201208300908040_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/127s4148/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4148/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.01 -0.11] [ 0.05 0.99 0.36] [ 0.12 -0.37 0.98]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 22914.69921875\r\n",
      "rescale:  min: 0.0  max: 8753.4150390625  scale: 0.029131487409434064\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4148/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.04batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-127s4148/ses-201208300908040/anat/sub-127s4148_ses-201208300908040_T1w.nii finished in 5.9508 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-127s4148/ses-201208300908040/anat/sub-127s4148_ses-201208300908040_T1w.nii finished in 6.1067 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.08batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-127s4148/ses-201208300908040/anat/sub-127s4148_ses-201208300908040_T1w.nii finished in 5.8087 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4148/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  22463\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4148/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4148/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.14\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/127s4148/mri/orig.mgz --rescale /output/127s4148/mri/orig_nu.mgz --aseg /output/127s4148/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/127s4148/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/127s4148/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/127s4148/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 41.43\r\n",
      "- m: 2.4620\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/127s4148/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/127s4148/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/127s4148/stats/aseg+DKT.stats --normfile /output/127s4148/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 127s4148 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/127s4148/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/127s4148/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.22 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:03:03 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-072s2083/ses-201209111026390/anat/sub-072s2083_ses-201209111026390_T1w.nii --asegdkt_segfile /output/072s2083/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/072s2083/mri/orig.mgz --brainmask_name /output/072s2083/mri/mask.mgz --aseg_name /output/072s2083/mri/aseg.auto_noCCseg.mgz --sid 072s2083 --seg_log /output/072s2083/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-072s2083/ses-201209111026390/anat/sub-072s2083_ses-201209111026390_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-072s2083/ses-201209111026390/anat/sub-072s2083_ses-201209111026390_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/072s2083/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s2083/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.01 0.03] [-0.01 1.05 0.11] [-0.03 -0.11 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1365.5318603515625\r\n",
      "rescale:  min: 0.0  max: 846.6297607421875  scale: 0.30119423132073386\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s2083/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-072s2083/ses-201209111026390/anat/sub-072s2083_ses-201209111026390_T1w.nii finished in 5.9624 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.97batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-072s2083/ses-201209111026390/anat/sub-072s2083_ses-201209111026390_T1w.nii finished in 6.1000 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.08batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-072s2083/ses-201209111026390/anat/sub-072s2083_ses-201209111026390_T1w.nii finished in 5.8079 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s2083/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  15545\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s2083/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s2083/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.94\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/072s2083/mri/orig.mgz --rescale /output/072s2083/mri/orig_nu.mgz --aseg /output/072s2083/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/072s2083/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/072s2083/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/072s2083/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 69.79\r\n",
      "- m: 1.4616\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/072s2083/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/072s2083/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/072s2083/stats/aseg+DKT.stats --normfile /output/072s2083/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 072s2083 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/072s2083/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/072s2083/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.09 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:06:11 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-123s2055/ses-201209241436050/anat/sub-123s2055_ses-201209241436050_T1w.nii --asegdkt_segfile /output/123s2055/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/123s2055/mri/orig.mgz --brainmask_name /output/123s2055/mri/mask.mgz --aseg_name /output/123s2055/mri/aseg.auto_noCCseg.mgz --sid 123s2055 --seg_log /output/123s2055/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-123s2055/ses-201209241436050/anat/sub-123s2055_ses-201209241436050_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-123s2055/ses-201209241436050/anat/sub-123s2055_ses-201209241436050_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/123s2055/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s2055/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999998x1.0546874x1.0546874\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.05 0.04] [ 0.05 1.04 0.17] [-0.05 -0.16 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 4960.91259765625\r\n",
      "rescale:  min: 0.0  max: 1567.6484375  scale: 0.16266402204735397\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s2055/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.05batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-123s2055/ses-201209241436050/anat/sub-123s2055_ses-201209241436050_T1w.nii finished in 5.9492 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.78batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-123s2055/ses-201209241436050/anat/sub-123s2055_ses-201209241436050_T1w.nii finished in 6.1283 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.06batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-123s2055/ses-201209241436050/anat/sub-123s2055_ses-201209241436050_T1w.nii finished in 5.8113 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s2055/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17688\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s2055/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/123s2055/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.11\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/123s2055/mri/orig.mgz --rescale /output/123s2055/mri/orig_nu.mgz --aseg /output/123s2055/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/123s2055/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/123s2055/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/123s2055/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 49.40\r\n",
      "- m: 2.0650\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/123s2055/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/123s2055/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/123s2055/stats/aseg+DKT.stats --normfile /output/123s2055/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 123s2055 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/123s2055/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/123s2055/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.79 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:09:21 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-005s4185/ses-201209271137370/anat/sub-005s4185_ses-201209271137370_T1w.nii --asegdkt_segfile /output/005s4185/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/005s4185/mri/orig.mgz --brainmask_name /output/005s4185/mri/mask.mgz --aseg_name /output/005s4185/mri/aseg.auto_noCCseg.mgz --sid 005s4185 --seg_log /output/005s4185/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-005s4185/ses-201209271137370/anat/sub-005s4185_ses-201209271137370_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-005s4185/ses-201209271137370/anat/sub-005s4185_ses-201209271137370_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/005s4185/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4185/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 -0.03] [ 0.04 0.97 0.3 ] [ 0.02 -0.3 0.97]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 10204.6083984375\r\n",
      "rescale:  min: 0.0  max: 6449.3125  scale: 0.039539098159687563\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4185/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.02batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-005s4185/ses-201209271137370/anat/sub-005s4185_ses-201209271137370_T1w.nii finished in 5.9531 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.02batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-005s4185/ses-201209271137370/anat/sub-005s4185_ses-201209271137370_T1w.nii finished in 6.0934 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.22batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-005s4185/ses-201209271137370/anat/sub-005s4185_ses-201209271137370_T1w.nii finished in 5.7898 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4185/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17367\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4185/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/005s4185/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.02\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/005s4185/mri/orig.mgz --rescale /output/005s4185/mri/orig_nu.mgz --aseg /output/005s4185/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/005s4185/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/005s4185/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/005s4185/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 55.85\r\n",
      "- m: 1.8264\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/005s4185/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/005s4185/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/005s4185/stats/aseg+DKT.stats --normfile /output/005s4185/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 005s4185 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/005s4185/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/005s4185/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.63 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:12:28 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-128s2036/ses-201210060910010/anat/sub-128s2036_ses-201210060910010_T1w.nii --asegdkt_segfile /output/128s2036/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/128s2036/mri/orig.mgz --brainmask_name /output/128s2036/mri/mask.mgz --aseg_name /output/128s2036/mri/aseg.auto_noCCseg.mgz --sid 128s2036 --seg_log /output/128s2036/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-128s2036/ses-201210060910010/anat/sub-128s2036_ses-201210060910010_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-128s2036/ses-201210060910010/anat/sub-128s2036_ses-201210060910010_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/128s2036/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2036/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.05 0.04] [ 0.05 1.05 0.04] [-0.05 -0.03 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1244.12744140625\r\n",
      "rescale:  min: 0.0  max: 782.55615234375  scale: 0.3258552108194113\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2036/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-128s2036/ses-201210060910010/anat/sub-128s2036_ses-201210060910010_T1w.nii finished in 5.9814 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-128s2036/ses-201210060910010/anat/sub-128s2036_ses-201210060910010_T1w.nii finished in 6.1073 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.12batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-128s2036/ses-201210060910010/anat/sub-128s2036_ses-201210060910010_T1w.nii finished in 5.8037 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2036/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19908\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2036/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2036/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.04\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/128s2036/mri/orig.mgz --rescale /output/128s2036/mri/orig_nu.mgz --aseg /output/128s2036/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/128s2036/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/128s2036/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/128s2036/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 73.14\r\n",
      "- m: 1.3946\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/128s2036/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/128s2036/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/128s2036/stats/aseg+DKT.stats --normfile /output/128s2036/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 128s2036 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/128s2036/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/128s2036/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.45 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:15:36 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s0089/ses-201210091114110/anat/sub-073s0089_ses-201210091114110_T1w.nii --asegdkt_segfile /output/073s0089/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s0089/mri/orig.mgz --brainmask_name /output/073s0089/mri/mask.mgz --aseg_name /output/073s0089/mri/aseg.auto_noCCseg.mgz --sid 073s0089 --seg_log /output/073s0089/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s0089/ses-201210091114110/anat/sub-073s0089_ses-201210091114110_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s0089/ses-201210091114110/anat/sub-073s0089_ses-201210091114110_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s0089/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0089/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.06 -0.04] [ 0.06 1.05 0.03] [ 0.05 -0.03 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1807.5618896484375\r\n",
      "rescale:  min: 0.0  max: 1015.8497924804688  scale: 0.2510213634806671\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0089/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.81batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s0089/ses-201210091114110/anat/sub-073s0089_ses-201210091114110_T1w.nii finished in 5.9819 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.76batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s0089/ses-201210091114110/anat/sub-073s0089_ses-201210091114110_T1w.nii finished in 6.1315 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.98batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s0089/ses-201210091114110/anat/sub-073s0089_ses-201210091114110_T1w.nii finished in 5.8215 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0089/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20381\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0089/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0089/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s0089/mri/orig.mgz --rescale /output/073s0089/mri/orig_nu.mgz --aseg /output/073s0089/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s0089/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s0089/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s0089/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 51.17\r\n",
      "- m: 1.9932\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s0089/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s0089/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s0089/stats/aseg+DKT.stats --normfile /output/073s0089/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s0089 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s0089/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s0089/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.44 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:18:44 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-099s2042/ses-201208210808420/anat/sub-099s2042_ses-201208210808420_T1w.nii --asegdkt_segfile /output/099s2042/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/099s2042/mri/orig.mgz --brainmask_name /output/099s2042/mri/mask.mgz --aseg_name /output/099s2042/mri/aseg.auto_noCCseg.mgz --sid 099s2042 --seg_log /output/099s2042/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-099s2042/ses-201208210808420/anat/sub-099s2042_ses-201208210808420_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-099s2042/ses-201208210808420/anat/sub-099s2042_ses-201208210808420_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/099s2042/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s2042/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 0.04] [-0.05 0.95 0.46] [-0.03 -0.46 0.95]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2273.005126953125\r\n",
      "rescale:  min: 0.0  max: 1081.950439453125  scale: 0.23568547199711892\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s2042/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.19batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-099s2042/ses-201208210808420/anat/sub-099s2042_ses-201208210808420_T1w.nii finished in 5.9302 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.76batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-099s2042/ses-201208210808420/anat/sub-099s2042_ses-201208210808420_T1w.nii finished in 6.1306 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.09batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-099s2042/ses-201208210808420/anat/sub-099s2042_ses-201208210808420_T1w.nii finished in 5.8066 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s2042/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20563\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s2042/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s2042/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.12\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/099s2042/mri/orig.mgz --rescale /output/099s2042/mri/orig_nu.mgz --aseg /output/099s2042/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/099s2042/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/099s2042/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/099s2042/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 68.28\r\n",
      "- m: 1.4939\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/099s2042/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/099s2042/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/099s2042/stats/aseg+DKT.stats --normfile /output/099s2042/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 099s2042 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/099s2042/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/099s2042/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.74 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:21:53 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-024s4280/ses-201211051025130/anat/sub-024s4280_ses-201211051025130_T1w.nii --asegdkt_segfile /output/024s4280/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/024s4280/mri/orig.mgz --brainmask_name /output/024s4280/mri/mask.mgz --aseg_name /output/024s4280/mri/aseg.auto_noCCseg.mgz --sid 024s4280 --seg_log /output/024s4280/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-024s4280/ses-201211051025130/anat/sub-024s4280_ses-201211051025130_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-024s4280/ses-201211051025130/anat/sub-024s4280_ses-201211051025130_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/024s4280/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/024s4280/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.05 0.02] [-0.06 1. 0.34] [-0. -0.34 1. ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2123.400146484375\r\n",
      "rescale:  min: 0.0  max: 908.8152465820312  scale: 0.28058508146626177\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/024s4280/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.96batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-024s4280/ses-201211051025130/anat/sub-024s4280_ses-201211051025130_T1w.nii finished in 5.9608 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-024s4280/ses-201211051025130/anat/sub-024s4280_ses-201211051025130_T1w.nii finished in 6.1265 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.13batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-024s4280/ses-201211051025130/anat/sub-024s4280_ses-201211051025130_T1w.nii finished in 5.8018 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/024s4280/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16502\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/024s4280/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/024s4280/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.01\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/024s4280/mri/orig.mgz --rescale /output/024s4280/mri/orig_nu.mgz --aseg /output/024s4280/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/024s4280/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/024s4280/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/024s4280/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 53.17\r\n",
      "- m: 1.9184\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/024s4280/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/024s4280/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/024s4280/stats/aseg+DKT.stats --normfile /output/024s4280/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 024s4280 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/024s4280/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/024s4280/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.43 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:25:00 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-128s2045/ses-201210181449010/anat/sub-128s2045_ses-201210181449010_T1w.nii --asegdkt_segfile /output/128s2045/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/128s2045/mri/orig.mgz --brainmask_name /output/128s2045/mri/mask.mgz --aseg_name /output/128s2045/mri/aseg.auto_noCCseg.mgz --sid 128s2045 --seg_log /output/128s2045/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-128s2045/ses-201210181449010/anat/sub-128s2045_ses-201210181449010_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-128s2045/ses-201210181449010/anat/sub-128s2045_ses-201210181449010_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/128s2045/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2045/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.05 0. ] [ 0.06 1.02 0.24] [-0.02 -0.24 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1361.7677001953125\r\n",
      "rescale:  min: 0.0  max: 780.2929077148438  scale: 0.3268003559673378\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2045/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.96batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-128s2045/ses-201210181449010/anat/sub-128s2045_ses-201210181449010_T1w.nii finished in 5.9619 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.80batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-128s2045/ses-201210181449010/anat/sub-128s2045_ses-201210181449010_T1w.nii finished in 6.1255 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-128s2045/ses-201210181449010/anat/sub-128s2045_ses-201210181449010_T1w.nii finished in 5.8296 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2045/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  22819\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2045/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 2 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2045/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.26\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/128s2045/mri/orig.mgz --rescale /output/128s2045/mri/orig_nu.mgz --aseg /output/128s2045/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/128s2045/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/128s2045/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/128s2045/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 74.15\r\n",
      "- m: 1.3757\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/128s2045/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/128s2045/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/128s2045/stats/aseg+DKT.stats --normfile /output/128s2045/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 128s2045 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/128s2045/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/128s2045/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.79 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:28:10 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-067s2196/ses-201211281153460/anat/sub-067s2196_ses-201211281153460_T1w.nii --asegdkt_segfile /output/067s2196/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/067s2196/mri/orig.mgz --brainmask_name /output/067s2196/mri/mask.mgz --aseg_name /output/067s2196/mri/aseg.auto_noCCseg.mgz --sid 067s2196 --seg_log /output/067s2196/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-067s2196/ses-201211281153460/anat/sub-067s2196_ses-201211281153460_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-067s2196/ses-201211281153460/anat/sub-067s2196_ses-201211281153460_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/067s2196/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2196/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.08 -0.06] [ 0.1 1.04 0.13] [ 0.05 -0.14 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1648.957275390625\r\n",
      "rescale:  min: 0.0  max: 971.23583984375  scale: 0.26255209037696114\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2196/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.03batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-067s2196/ses-201211281153460/anat/sub-067s2196_ses-201211281153460_T1w.nii finished in 5.9522 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.03batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-067s2196/ses-201211281153460/anat/sub-067s2196_ses-201211281153460_T1w.nii finished in 6.0924 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-067s2196/ses-201211281153460/anat/sub-067s2196_ses-201211281153460_T1w.nii finished in 5.8278 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2196/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17613\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2196/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2196/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.97\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/067s2196/mri/orig.mgz --rescale /output/067s2196/mri/orig_nu.mgz --aseg /output/067s2196/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/067s2196/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/067s2196/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/067s2196/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 69.90\r\n",
      "- m: 1.4593\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/067s2196/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/067s2196/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/067s2196/stats/aseg+DKT.stats --normfile /output/067s2196/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 067s2196 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/067s2196/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/067s2196/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.43 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:31:18 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-014s4328/ses-201211081101260/anat/sub-014s4328_ses-201211081101260_T1w.nii --asegdkt_segfile /output/014s4328/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/014s4328/mri/orig.mgz --brainmask_name /output/014s4328/mri/mask.mgz --aseg_name /output/014s4328/mri/aseg.auto_noCCseg.mgz --sid 014s4328 --seg_log /output/014s4328/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-014s4328/ses-201211081101260/anat/sub-014s4328_ses-201211081101260_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-014s4328/ses-201211081101260/anat/sub-014s4328_ses-201211081101260_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/014s4328/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4328/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.06 -0.04] [ 0.07 1.04 0.16] [ 0.03 -0.16 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1640.4227294921875\r\n",
      "rescale:  min: 0.0  max: 887.4686889648438  scale: 0.2873340808197252\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4328/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-014s4328/ses-201211081101260/anat/sub-014s4328_ses-201211081101260_T1w.nii finished in 5.9568 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.69batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-014s4328/ses-201211081101260/anat/sub-014s4328_ses-201211081101260_T1w.nii finished in 6.1416 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.12batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-014s4328/ses-201211081101260/anat/sub-014s4328_ses-201211081101260_T1w.nii finished in 5.8033 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4328/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16272\r\n",
      "  Found 3 connected component(s)!\r\n",
      "  Selecting largest component!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4328/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4328/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.87\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/014s4328/mri/orig.mgz --rescale /output/014s4328/mri/orig_nu.mgz --aseg /output/014s4328/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/014s4328/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/014s4328/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/014s4328/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 58.58\r\n",
      "- m: 1.7413\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/014s4328/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/014s4328/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/014s4328/stats/aseg+DKT.stats --normfile /output/014s4328/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 014s4328 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/014s4328/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/014s4328/stats/aseg+DKT.stats.\r\n",
      "Calculation took 13.89 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:34:31 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-021s4276/ses-201210221004540/anat/sub-021s4276_ses-201210221004540_T1w.nii --asegdkt_segfile /output/021s4276/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/021s4276/mri/orig.mgz --brainmask_name /output/021s4276/mri/mask.mgz --aseg_name /output/021s4276/mri/aseg.auto_noCCseg.mgz --sid 021s4276 --seg_log /output/021s4276/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-021s4276/ses-201210221004540/anat/sub-021s4276_ses-201210221004540_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-021s4276/ses-201210221004540/anat/sub-021s4276_ses-201210221004540_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/021s4276/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4276/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 0.01] [ 0.02 1.01 0.08] [-0.02 -0.08 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 13015.5498046875\r\n",
      "rescale:  min: 0.0  max: 8460.107421875  scale: 0.030141461246775133\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4276/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.03batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-021s4276/ses-201210221004540/anat/sub-021s4276_ses-201210221004540_T1w.nii finished in 5.9520 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.80batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-021s4276/ses-201210221004540/anat/sub-021s4276_ses-201210221004540_T1w.nii finished in 6.1255 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.22batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-021s4276/ses-201210221004540/anat/sub-021s4276_ses-201210221004540_T1w.nii finished in 5.7907 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4276/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18848\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4276/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4276/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.06\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/021s4276/mri/orig.mgz --rescale /output/021s4276/mri/orig_nu.mgz --aseg /output/021s4276/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/021s4276/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/021s4276/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/021s4276/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 52.63\r\n",
      "- m: 1.9382\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/021s4276/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/021s4276/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/021s4276/stats/aseg+DKT.stats --normfile /output/021s4276/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 021s4276 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/021s4276/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/021s4276/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.39 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:37:38 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-067s2195/ses-201212051217390/anat/sub-067s2195_ses-201212051217390_T1w.nii --asegdkt_segfile /output/067s2195/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/067s2195/mri/orig.mgz --brainmask_name /output/067s2195/mri/mask.mgz --aseg_name /output/067s2195/mri/aseg.auto_noCCseg.mgz --sid 067s2195 --seg_log /output/067s2195/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-067s2195/ses-201212051217390/anat/sub-067s2195_ses-201212051217390_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-067s2195/ses-201212051217390/anat/sub-067s2195_ses-201212051217390_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/067s2195/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2195/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.13 0.01] [ 0.13 0.98 0.37] [-0.06 -0.37 0.99]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1537.500244140625\r\n",
      "rescale:  min: 0.0  max: 956.3251342773438  scale: 0.2666457158345976\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2195/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.86batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-067s2195/ses-201212051217390/anat/sub-067s2195_ses-201212051217390_T1w.nii finished in 5.9756 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.59batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-067s2195/ses-201212051217390/anat/sub-067s2195_ses-201212051217390_T1w.nii finished in 6.1562 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.03batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-067s2195/ses-201212051217390/anat/sub-067s2195_ses-201212051217390_T1w.nii finished in 5.8153 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2195/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19736\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2195/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2195/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.21\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/067s2195/mri/orig.mgz --rescale /output/067s2195/mri/orig_nu.mgz --aseg /output/067s2195/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/067s2195/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/067s2195/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/067s2195/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 61.08\r\n",
      "- m: 1.6700\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/067s2195/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/067s2195/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/067s2195/stats/aseg+DKT.stats --normfile /output/067s2195/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 067s2195 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/067s2195/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/067s2195/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.06 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:40:46 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-035s2074/ses-201211271413050/anat/sub-035s2074_ses-201211271413050_T1w.nii --asegdkt_segfile /output/035s2074/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/035s2074/mri/orig.mgz --brainmask_name /output/035s2074/mri/mask.mgz --aseg_name /output/035s2074/mri/aseg.auto_noCCseg.mgz --sid 035s2074 --seg_log /output/035s2074/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-035s2074/ses-201211271413050/anat/sub-035s2074_ses-201211271413050_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-035s2074/ses-201211271413050/anat/sub-035s2074_ses-201211271413050_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/035s2074/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s2074/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.08 -0.05] [-0.08 1.05 0.12] [ 0.07 -0.11 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1911.9207763671875\r\n",
      "rescale:  min: 0.0  max: 879.4835815429688  scale: 0.28994287710593436\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s2074/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.98batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-035s2074/ses-201211271413050/anat/sub-035s2074_ses-201211271413050_T1w.nii finished in 5.9593 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.42batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-035s2074/ses-201211271413050/anat/sub-035s2074_ses-201211271413050_T1w.nii finished in 6.1819 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.01batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-035s2074/ses-201211271413050/anat/sub-035s2074_ses-201211271413050_T1w.nii finished in 5.8177 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s2074/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19294\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s2074/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 2 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s2074/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/035s2074/mri/orig.mgz --rescale /output/035s2074/mri/orig_nu.mgz --aseg /output/035s2074/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/035s2074/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/035s2074/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/035s2074/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 52.96\r\n",
      "- m: 1.9258\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/035s2074/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/035s2074/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/035s2074/stats/aseg+DKT.stats --normfile /output/035s2074/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 035s2074 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/035s2074/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/035s2074/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.09 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:43:54 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-094s2201/ses-201301151022330/anat/sub-094s2201_ses-201301151022330_T1w.nii --asegdkt_segfile /output/094s2201/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/094s2201/mri/orig.mgz --brainmask_name /output/094s2201/mri/mask.mgz --aseg_name /output/094s2201/mri/aseg.auto_noCCseg.mgz --sid 094s2201 --seg_log /output/094s2201/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-094s2201/ses-201301151022330/anat/sub-094s2201_ses-201301151022330_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-094s2201/ses-201301151022330/anat/sub-094s2201_ses-201301151022330_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/094s2201/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s2201/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.07 0.05] [-0.09 0.99 0.19] [-0.05 -0.19 1. ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 6744.10205078125\r\n",
      "rescale:  min: 0.0  max: 4808.544921875  scale: 0.05303059535535495\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s2201/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.08batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-094s2201/ses-201301151022330/anat/sub-094s2201_ses-201301151022330_T1w.nii finished in 5.9442 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.68batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-094s2201/ses-201301151022330/anat/sub-094s2201_ses-201301151022330_T1w.nii finished in 6.1426 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.19batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-094s2201/ses-201301151022330/anat/sub-094s2201_ses-201301151022330_T1w.nii finished in 5.7935 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s2201/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19931\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s2201/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s2201/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.05\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/094s2201/mri/orig.mgz --rescale /output/094s2201/mri/orig_nu.mgz --aseg /output/094s2201/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/094s2201/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/094s2201/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/094s2201/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 58.09\r\n",
      "- m: 1.7558\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/094s2201/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/094s2201/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/094s2201/stats/aseg+DKT.stats --normfile /output/094s2201/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 094s2201 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/094s2201/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/094s2201/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.93 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:47:00 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-072s4445/ses-201301240955020/anat/sub-072s4445_ses-201301240955020_T1w.nii --asegdkt_segfile /output/072s4445/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/072s4445/mri/orig.mgz --brainmask_name /output/072s4445/mri/mask.mgz --aseg_name /output/072s4445/mri/aseg.auto_noCCseg.mgz --sid 072s4445 --seg_log /output/072s4445/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-072s4445/ses-201301240955020/anat/sub-072s4445_ses-201301240955020_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-072s4445/ses-201301240955020/anat/sub-072s4445_ses-201301240955020_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/072s4445/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4445/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.05 0.01] [-0.05 1.05 -0.04] [-0.02 0.04 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2125.89892578125\r\n",
      "rescale:  min: 0.0  max: 1037.438720703125  scale: 0.2457976504165697\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4445/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.01batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-072s4445/ses-201301240955020/anat/sub-072s4445_ses-201301240955020_T1w.nii finished in 5.9547 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.45batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-072s4445/ses-201301240955020/anat/sub-072s4445_ses-201301240955020_T1w.nii finished in 6.1770 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.90batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-072s4445/ses-201301240955020/anat/sub-072s4445_ses-201301240955020_T1w.nii finished in 5.8328 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4445/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21507\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4445/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4445/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.01\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/072s4445/mri/orig.mgz --rescale /output/072s4445/mri/orig_nu.mgz --aseg /output/072s4445/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/072s4445/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/072s4445/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/072s4445/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 40.07\r\n",
      "- m: 2.5453\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/072s4445/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/072s4445/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/072s4445/stats/aseg+DKT.stats --normfile /output/072s4445/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 072s4445 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/072s4445/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/072s4445/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.31 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:50:08 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-037s4381/ses-201301251132420/anat/sub-037s4381_ses-201301251132420_T1w.nii --asegdkt_segfile /output/037s4381/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/037s4381/mri/orig.mgz --brainmask_name /output/037s4381/mri/mask.mgz --aseg_name /output/037s4381/mri/aseg.auto_noCCseg.mgz --sid 037s4381 --seg_log /output/037s4381/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-037s4381/ses-201301251132420/anat/sub-037s4381_ses-201301251132420_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-037s4381/ses-201301251132420/anat/sub-037s4381_ses-201301251132420_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/037s4381/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4381/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.01 0.04] [-0.01 1.05 0.13] [-0.05 -0.13 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1226.2855224609375\r\n",
      "rescale:  min: 0.0  max: 840.0055541992188  scale: 0.30356942132733006\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4381/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.89batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-037s4381/ses-201301251132420/anat/sub-037s4381_ses-201301251132420_T1w.nii finished in 5.9708 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.19batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-037s4381/ses-201301251132420/anat/sub-037s4381_ses-201301251132420_T1w.nii finished in 6.2161 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.11batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-037s4381/ses-201301251132420/anat/sub-037s4381_ses-201301251132420_T1w.nii finished in 5.8050 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4381/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17099\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4381/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4381/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.03\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/037s4381/mri/orig.mgz --rescale /output/037s4381/mri/orig_nu.mgz --aseg /output/037s4381/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/037s4381/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/037s4381/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/037s4381/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 61.85\r\n",
      "- m: 1.6492\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/037s4381/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/037s4381/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/037s4381/stats/aseg+DKT.stats --normfile /output/037s4381/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 037s4381 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/037s4381/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/037s4381/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.22 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:53:14 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-021s4419/ses-201212171032370/anat/sub-021s4419_ses-201212171032370_T1w.nii --asegdkt_segfile /output/021s4419/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/021s4419/mri/orig.mgz --brainmask_name /output/021s4419/mri/mask.mgz --aseg_name /output/021s4419/mri/aseg.auto_noCCseg.mgz --sid 021s4419 --seg_log /output/021s4419/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-021s4419/ses-201212171032370/anat/sub-021s4419_ses-201212171032370_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-021s4419/ses-201212171032370/anat/sub-021s4419_ses-201212171032370_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/021s4419/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4419/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 0.03] [ 0.02 1.03 0.22] [-0.04 -0.22 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 14241.962890625\r\n",
      "rescale:  min: 0.0  max: 8516.693359375  scale: 0.02994119774422796\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4419/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-021s4419/ses-201212171032370/anat/sub-021s4419_ses-201212171032370_T1w.nii finished in 5.9622 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.65batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-021s4419/ses-201212171032370/anat/sub-021s4419_ses-201212171032370_T1w.nii finished in 6.1470 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.20batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-021s4419/ses-201212171032370/anat/sub-021s4419_ses-201212171032370_T1w.nii finished in 5.7930 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4419/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18692\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4419/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4419/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.97\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/021s4419/mri/orig.mgz --rescale /output/021s4419/mri/orig_nu.mgz --aseg /output/021s4419/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/021s4419/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/021s4419/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/021s4419/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 44.74\r\n",
      "- m: 2.2797\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/021s4419/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/021s4419/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/021s4419/stats/aseg+DKT.stats --normfile /output/021s4419/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 021s4419 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/021s4419/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/021s4419/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.45 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:56:21 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-128s2151/ses-201302051739350/anat/sub-128s2151_ses-201302051739350_T1w.nii --asegdkt_segfile /output/128s2151/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/128s2151/mri/orig.mgz --brainmask_name /output/128s2151/mri/mask.mgz --aseg_name /output/128s2151/mri/aseg.auto_noCCseg.mgz --sid 128s2151 --seg_log /output/128s2151/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-128s2151/ses-201302051739350/anat/sub-128s2151_ses-201302051739350_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-128s2151/ses-201302051739350/anat/sub-128s2151_ses-201302051739350_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/128s2151/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2151/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.09 -0.03] [ 0.1 1.04 0.13] [ 0.02 -0.13 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2750.54541015625\r\n",
      "rescale:  min: 0.0  max: 1419.2813720703125  scale: 0.17966839065042492\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2151/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.17batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-128s2151/ses-201302051739350/anat/sub-128s2151_ses-201302051739350_T1w.nii finished in 5.9317 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.31batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-128s2151/ses-201302051739350/anat/sub-128s2151_ses-201302051739350_T1w.nii finished in 6.1973 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.07batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-128s2151/ses-201302051739350/anat/sub-128s2151_ses-201302051739350_T1w.nii finished in 5.8101 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2151/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  0\r\n",
      "  Found 30 connected component(s)!\r\n",
      "  Selecting largest component!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2151/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s2151/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.01\r\n",
      "[WARNING: run_prediction.py:  766]: Total segmentation volume is too small. Segmentation may be corrupted.\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/128s2151/mri/orig.mgz --rescale /output/128s2151/mri/orig_nu.mgz --aseg /output/128s2151/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/128s2151/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/128s2151/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/128s2151/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 28.90\r\n",
      "- m: 3.5295\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/128s2151/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/128s2151/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/128s2151/stats/aseg+DKT.stats --normfile /output/128s2151/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 128s2151 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/128s2151/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [17, 18, 26, 27, 28, 58, 59, 60, 10, 11, 12, 13, 49, 50, 51, 52, 53, 54] for merged label 10001 exist in the segmentation.\r\n",
      "None of the labels [4, 5, 31, 43, 44, 63] for merged label 10002 exist in the segmentation.\r\n",
      "None of the labels [14, 15, 72, 24] for merged label 10003 exist in the segmentation.\r\n",
      "None of the labels [7, 46] for merged label 10005 exist in the segmentation.\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/128s2151/stats/aseg+DKT.stats.\r\n",
      "Calculation took 4.98 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 03:59:23 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-011s2274/ses-201302061432250/anat/sub-011s2274_ses-201302061432250_T1w.nii --asegdkt_segfile /output/011s2274/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/011s2274/mri/orig.mgz --brainmask_name /output/011s2274/mri/mask.mgz --aseg_name /output/011s2274/mri/aseg.auto_noCCseg.mgz --sid 011s2274 --seg_log /output/011s2274/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-011s2274/ses-201302061432250/anat/sub-011s2274_ses-201302061432250_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-011s2274/ses-201302061432250/anat/sub-011s2274_ses-201302061432250_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/011s2274/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s2274/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.06 -0.03] [ 0.07 1. 0.34] [ 0.01 -0.34 1. ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1913.1497802734375\r\n",
      "rescale:  min: 0.0  max: 1067.53759765625  scale: 0.23886746523948724\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s2274/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.78batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-011s2274/ses-201302061432250/anat/sub-011s2274_ses-201302061432250_T1w.nii finished in 5.9866 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.45batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-011s2274/ses-201302061432250/anat/sub-011s2274_ses-201302061432250_T1w.nii finished in 6.1769 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.10batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-011s2274/ses-201302061432250/anat/sub-011s2274_ses-201302061432250_T1w.nii finished in 5.8056 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s2274/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  24410\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s2274/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/011s2274/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.5\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/011s2274/mri/orig.mgz --rescale /output/011s2274/mri/orig_nu.mgz --aseg /output/011s2274/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/011s2274/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/011s2274/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/011s2274/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 70.28\r\n",
      "- m: 1.4514\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/011s2274/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/011s2274/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/011s2274/stats/aseg+DKT.stats --normfile /output/011s2274/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 011s2274 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/011s2274/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/011s2274/stats/aseg+DKT.stats.\r\n",
      "Calculation took 12.74 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:02:33 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-022s4291/ses-201301301429580/anat/sub-022s4291_ses-201301301429580_T1w.nii --asegdkt_segfile /output/022s4291/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/022s4291/mri/orig.mgz --brainmask_name /output/022s4291/mri/mask.mgz --aseg_name /output/022s4291/mri/aseg.auto_noCCseg.mgz --sid 022s4291 --seg_log /output/022s4291/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-022s4291/ses-201301301429580/anat/sub-022s4291_ses-201301301429580_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-022s4291/ses-201301301429580/anat/sub-022s4291_ses-201301301429580_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/022s4291/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s4291/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.02 0.04] [-0.03 1.05 0.02] [-0.04 -0.02 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1155.8516845703125\r\n",
      "rescale:  min: 0.0  max: 748.9918823242188  scale: 0.34045762847082134\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s4291/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-022s4291/ses-201301301429580/anat/sub-022s4291_ses-201301301429580_T1w.nii finished in 5.9623 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.17batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-022s4291/ses-201301301429580/anat/sub-022s4291_ses-201301301429580_T1w.nii finished in 6.2194 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.06batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-022s4291/ses-201301301429580/anat/sub-022s4291_ses-201301301429580_T1w.nii finished in 5.8109 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s4291/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18185\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s4291/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s4291/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.98\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/022s4291/mri/orig.mgz --rescale /output/022s4291/mri/orig_nu.mgz --aseg /output/022s4291/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/022s4291/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/022s4291/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/022s4291/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 72.96\r\n",
      "- m: 1.3980\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/022s4291/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/022s4291/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/022s4291/stats/aseg+DKT.stats --normfile /output/022s4291/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 022s4291 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/022s4291/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/022s4291/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.29 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:05:40 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-023s4448/ses-201302281128010/anat/sub-023s4448_ses-201302281128010_T1w.nii --asegdkt_segfile /output/023s4448/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/023s4448/mri/orig.mgz --brainmask_name /output/023s4448/mri/mask.mgz --aseg_name /output/023s4448/mri/aseg.auto_noCCseg.mgz --sid 023s4448 --seg_log /output/023s4448/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-023s4448/ses-201302281128010/anat/sub-023s4448_ses-201302281128010_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-023s4448/ses-201302281128010/anat/sub-023s4448_ses-201302281128010_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/023s4448/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4448/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.04 0.03] [-0.05 1.04 0.17] [-0.03 -0.17 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2530.333984375\r\n",
      "rescale:  min: 0.0  max: 890.6775512695312  scale: 0.28629889642613604\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4448/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.98batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-023s4448/ses-201302281128010/anat/sub-023s4448_ses-201302281128010_T1w.nii finished in 5.9588 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.25batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-023s4448/ses-201302281128010/anat/sub-023s4448_ses-201302281128010_T1w.nii finished in 6.2068 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.10batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-023s4448/ses-201302281128010/anat/sub-023s4448_ses-201302281128010_T1w.nii finished in 5.8063 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4448/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19744\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4448/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4448/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.07\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/023s4448/mri/orig.mgz --rescale /output/023s4448/mri/orig_nu.mgz --aseg /output/023s4448/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/023s4448/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/023s4448/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/023s4448/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 54.40\r\n",
      "- m: 1.8750\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/023s4448/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/023s4448/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/023s4448/stats/aseg+DKT.stats --normfile /output/023s4448/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 023s4448 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/023s4448/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/023s4448/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.67 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:08:46 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-099s4565/ses-201302261036210/anat/sub-099s4565_ses-201302261036210_T1w.nii --asegdkt_segfile /output/099s4565/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/099s4565/mri/orig.mgz --brainmask_name /output/099s4565/mri/mask.mgz --aseg_name /output/099s4565/mri/aseg.auto_noCCseg.mgz --sid 099s4565 --seg_log /output/099s4565/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-099s4565/ses-201302261036210/anat/sub-099s4565_ses-201302261036210_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-099s4565/ses-201302261036210/anat/sub-099s4565_ses-201302261036210_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/099s4565/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4565/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 0.03] [-0.04 1.04 0.19] [-0.03 -0.19 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1415.3187255859375\r\n",
      "rescale:  min: 0.0  max: 951.0941772460938  scale: 0.2681122501857345\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4565/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.73batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-099s4565/ses-201302261036210/anat/sub-099s4565_ses-201302261036210_T1w.nii finished in 5.9940 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.15batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-099s4565/ses-201302261036210/anat/sub-099s4565_ses-201302261036210_T1w.nii finished in 6.2223 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.80batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-099s4565/ses-201302261036210/anat/sub-099s4565_ses-201302261036210_T1w.nii finished in 5.8454 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4565/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21692\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4565/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4565/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.01\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/099s4565/mri/orig.mgz --rescale /output/099s4565/mri/orig_nu.mgz --aseg /output/099s4565/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/099s4565/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/099s4565/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/099s4565/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 68.20\r\n",
      "- m: 1.4956\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/099s4565/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/099s4565/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/099s4565/stats/aseg+DKT.stats --normfile /output/099s4565/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 099s4565 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/099s4565/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/099s4565/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.51 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:11:53 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s4614/ses-201303120850540/anat/sub-073s4614_ses-201303120850540_T1w.nii --asegdkt_segfile /output/073s4614/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s4614/mri/orig.mgz --brainmask_name /output/073s4614/mri/mask.mgz --aseg_name /output/073s4614/mri/aseg.auto_noCCseg.mgz --sid 073s4614 --seg_log /output/073s4614/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s4614/ses-201303120850540/anat/sub-073s4614_ses-201303120850540_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s4614/ses-201303120850540/anat/sub-073s4614_ses-201303120850540_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s4614/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4614/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.04 0.05] [ 0.03 1.03 0.21] [-0.06 -0.21 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2174.423828125\r\n",
      "rescale:  min: 0.0  max: 1252.4681396484375  scale: 0.20359799337616477\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4614/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.16batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s4614/ses-201303120850540/anat/sub-073s4614_ses-201303120850540_T1w.nii finished in 5.9338 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.59batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s4614/ses-201303120850540/anat/sub-073s4614_ses-201303120850540_T1w.nii finished in 6.1570 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.03batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s4614/ses-201303120850540/anat/sub-073s4614_ses-201303120850540_T1w.nii finished in 5.8145 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4614/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16512\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4614/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4614/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.99\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s4614/mri/orig.mgz --rescale /output/073s4614/mri/orig_nu.mgz --aseg /output/073s4614/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s4614/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s4614/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s4614/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 48.95\r\n",
      "- m: 2.0837\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s4614/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s4614/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s4614/stats/aseg+DKT.stats --normfile /output/073s4614/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s4614 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s4614/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s4614/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.04 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:14:59 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-035s4464/ses-201302211415500/anat/sub-035s4464_ses-201302211415500_T1w.nii --asegdkt_segfile /output/035s4464/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/035s4464/mri/orig.mgz --brainmask_name /output/035s4464/mri/mask.mgz --aseg_name /output/035s4464/mri/aseg.auto_noCCseg.mgz --sid 035s4464 --seg_log /output/035s4464/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-035s4464/ses-201302211415500/anat/sub-035s4464_ses-201302211415500_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-035s4464/ses-201302211415500/anat/sub-035s4464_ses-201302211415500_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/035s4464/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4464/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.02 -0.02] [-0.02 1.05 0.11] [ 0.02 -0.11 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1026.1246337890625\r\n",
      "rescale:  min: 0.0  max: 699.8170166015625  scale: 0.3643809652390648\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4464/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.11batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-035s4464/ses-201302211415500/anat/sub-035s4464_ses-201302211415500_T1w.nii finished in 5.9403 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.44batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-035s4464/ses-201302211415500/anat/sub-035s4464_ses-201302211415500_T1w.nii finished in 6.1791 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-035s4464/ses-201302211415500/anat/sub-035s4464_ses-201302211415500_T1w.nii finished in 5.8289 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4464/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21541\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4464/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4464/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.12\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/035s4464/mri/orig.mgz --rescale /output/035s4464/mri/orig_nu.mgz --aseg /output/035s4464/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/035s4464/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/035s4464/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/035s4464/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 55.25\r\n",
      "- m: 1.8462\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/035s4464/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/035s4464/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/035s4464/stats/aseg+DKT.stats --normfile /output/035s4464/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 035s4464 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/035s4464/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/035s4464/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.03 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:18:06 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-072s4007/ses-201203221108250/anat/sub-072s4007_ses-201203221108250_T1w.nii --asegdkt_segfile /output/072s4007/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/072s4007/mri/orig.mgz --brainmask_name /output/072s4007/mri/mask.mgz --aseg_name /output/072s4007/mri/aseg.auto_noCCseg.mgz --sid 072s4007 --seg_log /output/072s4007/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-072s4007/ses-201203221108250/anat/sub-072s4007_ses-201203221108250_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-072s4007/ses-201203221108250/anat/sub-072s4007_ses-201203221108250_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/072s4007/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4007/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19e+00 -1.52e-02 1.12e-01] [ 1.19e-03 1.05e+00 1.32e-01] [-1.28e-01 -1.31e-01 1.04e+00]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1584.9539794921875\r\n",
      "rescale:  min: 0.0  max: 955.7272338867188  scale: 0.266812528678266\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4007/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.09batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-072s4007/ses-201203221108250/anat/sub-072s4007_ses-201203221108250_T1w.nii finished in 5.9433 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.49batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-072s4007/ses-201203221108250/anat/sub-072s4007_ses-201203221108250_T1w.nii finished in 6.1705 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.15batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-072s4007/ses-201203221108250/anat/sub-072s4007_ses-201203221108250_T1w.nii finished in 5.7992 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4007/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19232\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4007/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/072s4007/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.08\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/072s4007/mri/orig.mgz --rescale /output/072s4007/mri/orig_nu.mgz --aseg /output/072s4007/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/072s4007/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/072s4007/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/072s4007/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 70.77\r\n",
      "- m: 1.4413\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/072s4007/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/072s4007/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/072s4007/stats/aseg+DKT.stats --normfile /output/072s4007/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 072s4007 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/072s4007/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/072s4007/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.86 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:21:13 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-014s4576/ses-201304041124060/anat/sub-014s4576_ses-201304041124060_T1w.nii --asegdkt_segfile /output/014s4576/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/014s4576/mri/orig.mgz --brainmask_name /output/014s4576/mri/mask.mgz --aseg_name /output/014s4576/mri/aseg.auto_noCCseg.mgz --sid 014s4576 --seg_log /output/014s4576/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-014s4576/ses-201304041124060/anat/sub-014s4576_ses-201304041124060_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-014s4576/ses-201304041124060/anat/sub-014s4576_ses-201304041124060_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/014s4576/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4576/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 0.08] [-0.01 1.04 0.17] [-0.09 -0.17 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 3475.876708984375\r\n",
      "rescale:  min: 0.0  max: 1272.1708984375  scale: 0.2004447675333518\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4576/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.83batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-014s4576/ses-201304041124060/anat/sub-014s4576_ses-201304041124060_T1w.nii finished in 5.9797 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.48batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-014s4576/ses-201304041124060/anat/sub-014s4576_ses-201304041124060_T1w.nii finished in 6.1725 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.02batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-014s4576/ses-201304041124060/anat/sub-014s4576_ses-201304041124060_T1w.nii finished in 5.8160 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4576/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19858\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4576/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4576/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.96\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/014s4576/mri/orig.mgz --rescale /output/014s4576/mri/orig_nu.mgz --aseg /output/014s4576/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/014s4576/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/014s4576/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/014s4576/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 38.68\r\n",
      "- m: 2.6371\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/014s4576/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/014s4576/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/014s4576/stats/aseg+DKT.stats --normfile /output/014s4576/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 014s4576 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/014s4576/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/014s4576/stats/aseg+DKT.stats.\r\n",
      "Calculation took 8.77 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:24:18 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s2225/ses-201304291337190/anat/sub-073s2225_ses-201304291337190_T1w.nii --asegdkt_segfile /output/073s2225/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s2225/mri/orig.mgz --brainmask_name /output/073s2225/mri/mask.mgz --aseg_name /output/073s2225/mri/aseg.auto_noCCseg.mgz --sid 073s2225 --seg_log /output/073s2225/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s2225/ses-201304291337190/anat/sub-073s2225_ses-201304291337190_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s2225/ses-201304291337190/anat/sub-073s2225_ses-201304291337190_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s2225/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2225/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.09 -0.04] [-0.09 1.03 0.23] [ 0.07 -0.23 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2384.7900390625\r\n",
      "rescale:  min: 0.0  max: 1314.019287109375  scale: 0.19406107848002582\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2225/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.90batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s2225/ses-201304291337190/anat/sub-073s2225_ses-201304291337190_T1w.nii finished in 5.9703 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.55batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s2225/ses-201304291337190/anat/sub-073s2225_ses-201304291337190_T1w.nii finished in 6.1622 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.15batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s2225/ses-201304291337190/anat/sub-073s2225_ses-201304291337190_T1w.nii finished in 5.7990 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2225/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20260\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2225/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2225/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.24\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s2225/mri/orig.mgz --rescale /output/073s2225/mri/orig_nu.mgz --aseg /output/073s2225/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s2225/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s2225/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s2225/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 42.91\r\n",
      "- m: 2.3773\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s2225/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s2225/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s2225/stats/aseg+DKT.stats --normfile /output/073s2225/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s2225 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s2225/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s2225/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.19 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:27:26 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-067s2301/ses-201303151036450/anat/sub-067s2301_ses-201303151036450_T1w.nii --asegdkt_segfile /output/067s2301/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/067s2301/mri/orig.mgz --brainmask_name /output/067s2301/mri/mask.mgz --aseg_name /output/067s2301/mri/aseg.auto_noCCseg.mgz --sid 067s2301 --seg_log /output/067s2301/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-067s2301/ses-201303151036450/anat/sub-067s2301_ses-201303151036450_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-067s2301/ses-201303151036450/anat/sub-067s2301_ses-201303151036450_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/067s2301/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2301/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999998x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.01 -0.02] [-0.01 1.03 0.21] [ 0.02 -0.21 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1479.0706787109375\r\n",
      "rescale:  min: 0.0  max: 951.0424194335938  scale: 0.26812684144190824\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2301/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-067s2301/ses-201303151036450/anat/sub-067s2301_ses-201303151036450_T1w.nii finished in 5.9662 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.38batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-067s2301/ses-201303151036450/anat/sub-067s2301_ses-201303151036450_T1w.nii finished in 6.1870 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.11batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-067s2301/ses-201303151036450/anat/sub-067s2301_ses-201303151036450_T1w.nii finished in 5.8051 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2301/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20465\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2301/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/067s2301/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.16\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/067s2301/mri/orig.mgz --rescale /output/067s2301/mri/orig_nu.mgz --aseg /output/067s2301/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/067s2301/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/067s2301/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/067s2301/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 56.97\r\n",
      "- m: 1.7903\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/067s2301/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/067s2301/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/067s2301/stats/aseg+DKT.stats --normfile /output/067s2301/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 067s2301 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/067s2301/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/067s2301/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.96 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:30:33 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s2153/ses-201304011234170/anat/sub-073s2153_ses-201304011234170_T1w.nii --asegdkt_segfile /output/073s2153/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s2153/mri/orig.mgz --brainmask_name /output/073s2153/mri/mask.mgz --aseg_name /output/073s2153/mri/aseg.auto_noCCseg.mgz --sid 073s2153 --seg_log /output/073s2153/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s2153/ses-201304011234170/anat/sub-073s2153_ses-201304011234170_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s2153/ses-201304011234170/anat/sub-073s2153_ses-201304011234170_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s2153/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2153/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 0.01] [-0.04 1.05 0.1 ] [-0. -0.1 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2770.88623046875\r\n",
      "rescale:  min: 0.0  max: 1318.94189453125  scale: 0.19333679600087814\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2153/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.86batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s2153/ses-201304011234170/anat/sub-073s2153_ses-201304011234170_T1w.nii finished in 5.9749 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s2153/ses-201304011234170/anat/sub-073s2153_ses-201304011234170_T1w.nii finished in 6.2728 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.03batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s2153/ses-201304011234170/anat/sub-073s2153_ses-201304011234170_T1w.nii finished in 5.8157 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2153/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19156\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2153/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2153/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.99\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s2153/mri/orig.mgz --rescale /output/073s2153/mri/orig_nu.mgz --aseg /output/073s2153/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s2153/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s2153/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s2153/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 43.90\r\n",
      "- m: 2.3234\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s2153/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s2153/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s2153/stats/aseg+DKT.stats --normfile /output/073s2153/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s2153 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s2153/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s2153/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.01 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:33:39 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-094s4560/ses-201303211430590/anat/sub-094s4560_ses-201303211430590_T1w.nii --asegdkt_segfile /output/094s4560/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/094s4560/mri/orig.mgz --brainmask_name /output/094s4560/mri/mask.mgz --aseg_name /output/094s4560/mri/aseg.auto_noCCseg.mgz --sid 094s4560 --seg_log /output/094s4560/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-094s4560/ses-201303211430590/anat/sub-094s4560_ses-201303211430590_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-094s4560/ses-201303211430590/anat/sub-094s4560_ses-201303211430590_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/094s4560/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4560/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.08 -0.01] [ 0.1 1.01 0.01] [ 0.02 -0.01 1.02]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 6941.71875\r\n",
      "rescale:  min: 0.0  max: 4644.009765625  scale: 0.054909445257310216\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4560/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.94batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-094s4560/ses-201303211430590/anat/sub-094s4560_ses-201303211430590_T1w.nii finished in 5.9642 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.45batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-094s4560/ses-201303211430590/anat/sub-094s4560_ses-201303211430590_T1w.nii finished in 6.1771 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.33batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-094s4560/ses-201303211430590/anat/sub-094s4560_ses-201303211430590_T1w.nii finished in 5.7751 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4560/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18603\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4560/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4560/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.94\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/094s4560/mri/orig.mgz --rescale /output/094s4560/mri/orig_nu.mgz --aseg /output/094s4560/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/094s4560/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/094s4560/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/094s4560/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 62.16\r\n",
      "- m: 1.6408\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/094s4560/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/094s4560/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/094s4560/stats/aseg+DKT.stats --normfile /output/094s4560/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 094s4560 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/094s4560/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/094s4560/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.67 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:36:46 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-007s4568/ses-201305131446180/anat/sub-007s4568_ses-201305131446180_T1w.nii --asegdkt_segfile /output/007s4568/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/007s4568/mri/orig.mgz --brainmask_name /output/007s4568/mri/mask.mgz --aseg_name /output/007s4568/mri/aseg.auto_noCCseg.mgz --sid 007s4568 --seg_log /output/007s4568/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-007s4568/ses-201305131446180/anat/sub-007s4568_ses-201305131446180_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-007s4568/ses-201305131446180/anat/sub-007s4568_ses-201305131446180_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/007s4568/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4568/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.06 -0.01] [ 0.07 1.01 0.05] [ 0.01 -0.05 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 11486.9521484375\r\n",
      "rescale:  min: 0.0  max: 6053.6240234375  scale: 0.0421235278260972\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4568/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.76batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-007s4568/ses-201305131446180/anat/sub-007s4568_ses-201305131446180_T1w.nii finished in 5.9896 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.10batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-007s4568/ses-201305131446180/anat/sub-007s4568_ses-201305131446180_T1w.nii finished in 6.2301 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.01batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-007s4568/ses-201305131446180/anat/sub-007s4568_ses-201305131446180_T1w.nii finished in 5.8183 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4568/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16481\r\n",
      "  Found 2 connected component(s)!\r\n",
      "  Selecting largest component!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4568/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/007s4568/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.09\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/007s4568/mri/orig.mgz --rescale /output/007s4568/mri/orig_nu.mgz --aseg /output/007s4568/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/007s4568/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/007s4568/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/007s4568/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 33.88\r\n",
      "- m: 3.0104\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/007s4568/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/007s4568/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/007s4568/stats/aseg+DKT.stats --normfile /output/007s4568/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 007s4568 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/007s4568/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/007s4568/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.01 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:39:56 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-022s2379/ses-201305311149020/anat/sub-022s2379_ses-201305311149020_T1w.nii --asegdkt_segfile /output/022s2379/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/022s2379/mri/orig.mgz --brainmask_name /output/022s2379/mri/mask.mgz --aseg_name /output/022s2379/mri/aseg.auto_noCCseg.mgz --sid 022s2379 --seg_log /output/022s2379/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-022s2379/ses-201305311149020/anat/sub-022s2379_ses-201305311149020_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-022s2379/ses-201305311149020/anat/sub-022s2379_ses-201305311149020_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/022s2379/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s2379/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0x1.0\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.20e+00 4.72e-02 1.07e-02] [-5.80e-02 9.74e-01 2.23e-01] [ 1.60e-04 -2.23e-01 9.75e-01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 4273.5615234375\r\n",
      "rescale:  min: 0.0  max: 1709.424560546875  scale: 0.14917300586720306\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s2379/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.86batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-022s2379/ses-201305311149020/anat/sub-022s2379_ses-201305311149020_T1w.nii finished in 5.9754 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.30batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-022s2379/ses-201305311149020/anat/sub-022s2379_ses-201305311149020_T1w.nii finished in 6.1992 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.07batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-022s2379/ses-201305311149020/anat/sub-022s2379_ses-201305311149020_T1w.nii finished in 5.8095 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s2379/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  0\r\n",
      "  Found 21 connected component(s)!\r\n",
      "  Selecting largest component!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s2379/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/022s2379/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.02\r\n",
      "[WARNING: run_prediction.py:  766]: Total segmentation volume is too small. Segmentation may be corrupted.\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/022s2379/mri/orig.mgz --rescale /output/022s2379/mri/orig_nu.mgz --aseg /output/022s2379/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/022s2379/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/022s2379/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/022s2379/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 28.80\r\n",
      "- m: 3.5412\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/022s2379/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/022s2379/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/022s2379/stats/aseg+DKT.stats --normfile /output/022s2379/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 022s2379 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/022s2379/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [17, 18, 26, 27, 28, 58, 59, 60, 10, 11, 12, 13, 49, 50, 51, 52, 53, 54] for merged label 10001 exist in the segmentation.\r\n",
      "None of the labels [4, 5, 31, 43, 44, 63] for merged label 10002 exist in the segmentation.\r\n",
      "None of the labels [14, 15, 72, 24] for merged label 10003 exist in the segmentation.\r\n",
      "None of the labels [7, 46] for merged label 10005 exist in the segmentation.\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/022s2379/stats/aseg+DKT.stats.\r\n",
      "Calculation took 4.24 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:42:57 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-135s4598/ses-201303281045290/anat/sub-135s4598_ses-201303281045290_T1w.nii --asegdkt_segfile /output/135s4598/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/135s4598/mri/orig.mgz --brainmask_name /output/135s4598/mri/mask.mgz --aseg_name /output/135s4598/mri/aseg.auto_noCCseg.mgz --sid 135s4598 --seg_log /output/135s4598/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-135s4598/ses-201303281045290/anat/sub-135s4598_ses-201303281045290_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-135s4598/ses-201303281045290/anat/sub-135s4598_ses-201303281045290_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/135s4598/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/135s4598/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.07 -0.03] [ 0.08 1.03 0.2 ] [ 0.02 -0.2 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1271.974853515625\r\n",
      "rescale:  min: 0.0  max: 931.0855712890625  scale: 0.27387386064522456\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/135s4598/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.87batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-135s4598/ses-201303281045290/anat/sub-135s4598_ses-201303281045290_T1w.nii finished in 5.9738 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.51batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-135s4598/ses-201303281045290/anat/sub-135s4598_ses-201303281045290_T1w.nii finished in 6.1684 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.14batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-135s4598/ses-201303281045290/anat/sub-135s4598_ses-201303281045290_T1w.nii finished in 5.8002 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/135s4598/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19149\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/135s4598/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 4 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/135s4598/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.14\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/135s4598/mri/orig.mgz --rescale /output/135s4598/mri/orig_nu.mgz --aseg /output/135s4598/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/135s4598/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/135s4598/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/135s4598/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 68.83\r\n",
      "- m: 1.4818\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/135s4598/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/135s4598/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/135s4598/stats/aseg+DKT.stats --normfile /output/135s4598/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 135s4598 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/135s4598/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/135s4598/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.34 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:46:05 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-114s2392/ses-201308081147530/anat/sub-114s2392_ses-201308081147530_T1w.nii --asegdkt_segfile /output/114s2392/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/114s2392/mri/orig.mgz --brainmask_name /output/114s2392/mri/mask.mgz --aseg_name /output/114s2392/mri/aseg.auto_noCCseg.mgz --sid 114s2392 --seg_log /output/114s2392/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-114s2392/ses-201308081147530/anat/sub-114s2392_ses-201308081147530_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-114s2392/ses-201308081147530/anat/sub-114s2392_ses-201308081147530_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/114s2392/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/114s2392/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0x1.0\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.05 0.07] [ 0.04 0.97 0.23] [-0.09 -0.23 0.97]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1637.5384521484375\r\n",
      "rescale:  min: 0.0  max: 989.0732421875  scale: 0.25781710506698685\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/114s2392/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.75batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-114s2392/ses-201308081147530/anat/sub-114s2392_ses-201308081147530_T1w.nii finished in 5.9908 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.47batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-114s2392/ses-201308081147530/anat/sub-114s2392_ses-201308081147530_T1w.nii finished in 6.1738 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.16batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-114s2392/ses-201308081147530/anat/sub-114s2392_ses-201308081147530_T1w.nii finished in 5.7975 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/114s2392/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19009\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/114s2392/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/114s2392/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.12\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/114s2392/mri/orig.mgz --rescale /output/114s2392/mri/orig_nu.mgz --aseg /output/114s2392/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/114s2392/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/114s2392/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/114s2392/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 57.08\r\n",
      "- m: 1.7871\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/114s2392/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/114s2392/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/114s2392/stats/aseg+DKT.stats --normfile /output/114s2392/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 114s2392 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/114s2392/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/114s2392/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.68 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:49:12 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-082s2307/ses-201306271218100/anat/sub-082s2307_ses-201306271218100_T1w.nii --asegdkt_segfile /output/082s2307/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/082s2307/mri/orig.mgz --brainmask_name /output/082s2307/mri/mask.mgz --aseg_name /output/082s2307/mri/aseg.auto_noCCseg.mgz --sid 082s2307 --seg_log /output/082s2307/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-082s2307/ses-201306271218100/anat/sub-082s2307_ses-201306271218100_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-082s2307/ses-201306271218100/anat/sub-082s2307_ses-201306271218100_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/082s2307/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/082s2307/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0x1.0\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.09 -0.07] [ 0.13 0.98 0.17] [ 0.07 -0.18 0.98]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 4416.787109375\r\n",
      "rescale:  min: 0.0  max: 1408.955078125  scale: 0.1809851882143377\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/082s2307/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.10batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-082s2307/ses-201306271218100/anat/sub-082s2307_ses-201306271218100_T1w.nii finished in 5.9421 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.04batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-082s2307/ses-201306271218100/anat/sub-082s2307_ses-201306271218100_T1w.nii finished in 6.2379 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-082s2307/ses-201306271218100/anat/sub-082s2307_ses-201306271218100_T1w.nii finished in 5.8292 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/082s2307/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18680\r\n",
      "  Found 2 connected component(s)!\r\n",
      "  Selecting largest component!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/082s2307/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/082s2307/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/082s2307/mri/orig.mgz --rescale /output/082s2307/mri/orig_nu.mgz --aseg /output/082s2307/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/082s2307/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/082s2307/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/082s2307/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 39.44\r\n",
      "- m: 2.5860\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/082s2307/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/082s2307/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/082s2307/stats/aseg+DKT.stats --normfile /output/082s2307/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 082s2307 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/082s2307/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/082s2307/stats/aseg+DKT.stats.\r\n",
      "Calculation took 14.41 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:52:24 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s2182/ses-201312041652010/anat/sub-073s2182_ses-201312041652010_T1w.nii --asegdkt_segfile /output/073s2182/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s2182/mri/orig.mgz --brainmask_name /output/073s2182/mri/mask.mgz --aseg_name /output/073s2182/mri/aseg.auto_noCCseg.mgz --sid 073s2182 --seg_log /output/073s2182/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s2182/ses-201312041652010/anat/sub-073s2182_ses-201312041652010_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s2182/ses-201312041652010/anat/sub-073s2182_ses-201312041652010_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s2182/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2182/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0x1.0\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 -0.06] [ 0.02 0.99 0.15] [ 0.07 -0.15 0.99]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1365.08984375\r\n",
      "rescale:  min: 0.0  max: 921.4356689453125  scale: 0.2767420543768144\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2182/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.58batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s2182/ses-201312041652010/anat/sub-073s2182_ses-201312041652010_T1w.nii finished in 6.0147 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.91batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s2182/ses-201312041652010/anat/sub-073s2182_ses-201312041652010_T1w.nii finished in 6.2594 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.07batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s2182/ses-201312041652010/anat/sub-073s2182_ses-201312041652010_T1w.nii finished in 5.8096 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2182/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20164\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2182/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s2182/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.17\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s2182/mri/orig.mgz --rescale /output/073s2182/mri/orig_nu.mgz --aseg /output/073s2182/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s2182/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s2182/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s2182/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 50.91\r\n",
      "- m: 2.0035\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s2182/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s2182/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s2182/stats/aseg+DKT.stats --normfile /output/073s2182/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s2182 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s2182/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s2182/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.19 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:55:33 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-068s4174/ses-201210291153430/anat/sub-068s4174_ses-201210291153430_T1w.nii --asegdkt_segfile /output/068s4174/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/068s4174/mri/orig.mgz --brainmask_name /output/068s4174/mri/mask.mgz --aseg_name /output/068s4174/mri/aseg.auto_noCCseg.mgz --sid 068s4174 --seg_log /output/068s4174/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-068s4174/ses-201210291153430/anat/sub-068s4174_ses-201210291153430_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-068s4174/ses-201210291153430/anat/sub-068s4174_ses-201210291153430_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/068s4174/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4174/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 0.05] [ 0.02 0.99 0.35] [-0.07 -0.35 0.99]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 3227.57470703125\r\n",
      "rescale:  min: 0.0  max: 1358.8089599609375  scale: 0.18766434981951446\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4174/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-068s4174/ses-201210291153430/anat/sub-068s4174_ses-201210291153430_T1w.nii finished in 5.9815 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.18batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-068s4174/ses-201210291153430/anat/sub-068s4174_ses-201210291153430_T1w.nii finished in 6.2176 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-068s4174/ses-201210291153430/anat/sub-068s4174_ses-201210291153430_T1w.nii finished in 5.8285 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4174/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21301\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4174/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4174/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.05\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/068s4174/mri/orig.mgz --rescale /output/068s4174/mri/orig_nu.mgz --aseg /output/068s4174/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/068s4174/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/068s4174/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/068s4174/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 55.01\r\n",
      "- m: 1.8542\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/068s4174/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/068s4174/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/068s4174/stats/aseg+DKT.stats --normfile /output/068s4174/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 068s4174 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/068s4174/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/068s4174/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.10 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 04:58:40 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-116s4010/ses-201204181119380/anat/sub-116s4010_ses-201204181119380_T1w.nii --asegdkt_segfile /output/116s4010/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/116s4010/mri/orig.mgz --brainmask_name /output/116s4010/mri/mask.mgz --aseg_name /output/116s4010/mri/aseg.auto_noCCseg.mgz --sid 116s4010 --seg_log /output/116s4010/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-116s4010/ses-201204181119380/anat/sub-116s4010_ses-201204181119380_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-116s4010/ses-201204181119380/anat/sub-116s4010_ses-201204181119380_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/116s4010/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4010/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0. -0.01] [ 0. 1.05 0.02] [ 0.01 -0.02 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 4457.16748046875\r\n",
      "rescale:  min: 0.0  max: 1292.57861328125  scale: 0.19728007053487817\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4010/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-116s4010/ses-201204181119380/anat/sub-116s4010_ses-201204181119380_T1w.nii finished in 5.9815 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.67batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-116s4010/ses-201204181119380/anat/sub-116s4010_ses-201204181119380_T1w.nii finished in 6.1444 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.20batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-116s4010/ses-201204181119380/anat/sub-116s4010_ses-201204181119380_T1w.nii finished in 5.7921 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4010/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17800\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4010/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4010/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.95\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/116s4010/mri/orig.mgz --rescale /output/116s4010/mri/orig_nu.mgz --aseg /output/116s4010/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/116s4010/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/116s4010/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/116s4010/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 48.01\r\n",
      "- m: 2.1245\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/116s4010/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/116s4010/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/116s4010/stats/aseg+DKT.stats --normfile /output/116s4010/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 116s4010 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/116s4010/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/116s4010/stats/aseg+DKT.stats.\r\n",
      "Calculation took 8.77 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:01:47 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-023s4034/ses-201207261314580/anat/sub-023s4034_ses-201207261314580_T1w.nii --asegdkt_segfile /output/023s4034/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/023s4034/mri/orig.mgz --brainmask_name /output/023s4034/mri/mask.mgz --aseg_name /output/023s4034/mri/aseg.auto_noCCseg.mgz --sid 023s4034 --seg_log /output/023s4034/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-023s4034/ses-201207261314580/anat/sub-023s4034_ses-201207261314580_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-023s4034/ses-201207261314580/anat/sub-023s4034_ses-201207261314580_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/023s4034/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4034/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.02 0.03] [-0.03 1.05 0.08] [-0.03 -0.08 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1530.1785888671875\r\n",
      "rescale:  min: 0.0  max: 858.43017578125  scale: 0.29705386319618443\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4034/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.97batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-023s4034/ses-201207261314580/anat/sub-023s4034_ses-201207261314580_T1w.nii finished in 5.9603 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.31batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-023s4034/ses-201207261314580/anat/sub-023s4034_ses-201207261314580_T1w.nii finished in 6.1978 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.06batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-023s4034/ses-201207261314580/anat/sub-023s4034_ses-201207261314580_T1w.nii finished in 5.8112 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4034/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18848\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4034/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s4034/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.94\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/023s4034/mri/orig.mgz --rescale /output/023s4034/mri/orig_nu.mgz --aseg /output/023s4034/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/023s4034/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/023s4034/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/023s4034/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 49.26\r\n",
      "- m: 2.0707\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/023s4034/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/023s4034/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/023s4034/stats/aseg+DKT.stats --normfile /output/023s4034/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 023s4034 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/023s4034/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/023s4034/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.00 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:04:54 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-014s4079/ses-201206211135200/anat/sub-014s4079_ses-201206211135200_T1w.nii --asegdkt_segfile /output/014s4079/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/014s4079/mri/orig.mgz --brainmask_name /output/014s4079/mri/mask.mgz --aseg_name /output/014s4079/mri/aseg.auto_noCCseg.mgz --sid 014s4079 --seg_log /output/014s4079/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-014s4079/ses-201206211135200/anat/sub-014s4079_ses-201206211135200_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-014s4079/ses-201206211135200/anat/sub-014s4079_ses-201206211135200_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/014s4079/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4079/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 -0.07] [ 0.05 1.03 0.22] [ 0.08 -0.23 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2149.95166015625\r\n",
      "rescale:  min: 0.0  max: 915.87939453125  scale: 0.27842093786869154\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4079/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.67batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-014s4079/ses-201206211135200/anat/sub-014s4079_ses-201206211135200_T1w.nii finished in 6.0025 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.37batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-014s4079/ses-201206211135200/anat/sub-014s4079_ses-201206211135200_T1w.nii finished in 6.1882 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.97batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-014s4079/ses-201206211135200/anat/sub-014s4079_ses-201206211135200_T1w.nii finished in 5.8226 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4079/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19962\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4079/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4079/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.04\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/014s4079/mri/orig.mgz --rescale /output/014s4079/mri/orig_nu.mgz --aseg /output/014s4079/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/014s4079/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/014s4079/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/014s4079/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 53.66\r\n",
      "- m: 1.9007\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/014s4079/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/014s4079/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/014s4079/stats/aseg+DKT.stats --normfile /output/014s4079/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 014s4079 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/014s4079/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/014s4079/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.59 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:08:02 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-153s4159/ses-201208281128460/anat/sub-153s4159_ses-201208281128460_T1w.nii --asegdkt_segfile /output/153s4159/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/153s4159/mri/orig.mgz --brainmask_name /output/153s4159/mri/mask.mgz --aseg_name /output/153s4159/mri/aseg.auto_noCCseg.mgz --sid 153s4159 --seg_log /output/153s4159/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-153s4159/ses-201208281128460/anat/sub-153s4159_ses-201208281128460_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-153s4159/ses-201208281128460/anat/sub-153s4159_ses-201208281128460_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/153s4159/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4159/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 -0.02] [ 0.02 1.03 0.21] [ 0.02 -0.21 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1630.9403076171875\r\n",
      "rescale:  min: 0.0  max: 952.4691162109375  scale: 0.26772521613554\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4159/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.87batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-153s4159/ses-201208281128460/anat/sub-153s4159_ses-201208281128460_T1w.nii finished in 5.9741 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.23batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-153s4159/ses-201208281128460/anat/sub-153s4159_ses-201208281128460_T1w.nii finished in 6.2097 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-153s4159/ses-201208281128460/anat/sub-153s4159_ses-201208281128460_T1w.nii finished in 5.8252 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4159/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18346\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4159/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/153s4159/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.08\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/153s4159/mri/orig.mgz --rescale /output/153s4159/mri/orig_nu.mgz --aseg /output/153s4159/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/153s4159/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/153s4159/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/153s4159/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 71.56\r\n",
      "- m: 1.4254\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/153s4159/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/153s4159/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/153s4159/stats/aseg+DKT.stats --normfile /output/153s4159/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 153s4159 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/153s4159/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/153s4159/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.58 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:11:10 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-033s4179/ses-201209051633410/anat/sub-033s4179_ses-201209051633410_T1w.nii --asegdkt_segfile /output/033s4179/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/033s4179/mri/orig.mgz --brainmask_name /output/033s4179/mri/mask.mgz --aseg_name /output/033s4179/mri/aseg.auto_noCCseg.mgz --sid 033s4179 --seg_log /output/033s4179/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-033s4179/ses-201209051633410/anat/sub-033s4179_ses-201209051633410_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-033s4179/ses-201209051633410/anat/sub-033s4179_ses-201209051633410_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/033s4179/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4179/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.02 -0.02] [-0.02 1.04 0.16] [ 0.03 -0.15 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1903.8275146484375\r\n",
      "rescale:  min: 0.0  max: 1227.96875  scale: 0.2076600076345591\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4179/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-033s4179/ses-201209051633410/anat/sub-033s4179_ses-201209051633410_T1w.nii finished in 5.9858 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.30batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-033s4179/ses-201209051633410/anat/sub-033s4179_ses-201209051633410_T1w.nii finished in 6.1987 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-033s4179/ses-201209051633410/anat/sub-033s4179_ses-201209051633410_T1w.nii finished in 5.8292 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4179/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  22019\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4179/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4179/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.21\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/033s4179/mri/orig.mgz --rescale /output/033s4179/mri/orig_nu.mgz --aseg /output/033s4179/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/033s4179/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/033s4179/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/033s4179/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 66.12\r\n",
      "- m: 1.5426\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/033s4179/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/033s4179/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/033s4179/stats/aseg+DKT.stats --normfile /output/033s4179/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 033s4179 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/033s4179/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/033s4179/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.49 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:14:19 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-033s4176/ses-201209071405590/anat/sub-033s4176_ses-201209071405590_T1w.nii --asegdkt_segfile /output/033s4176/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/033s4176/mri/orig.mgz --brainmask_name /output/033s4176/mri/mask.mgz --aseg_name /output/033s4176/mri/aseg.auto_noCCseg.mgz --sid 033s4176 --seg_log /output/033s4176/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-033s4176/ses-201209071405590/anat/sub-033s4176_ses-201209071405590_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-033s4176/ses-201209071405590/anat/sub-033s4176_ses-201209071405590_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/033s4176/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4176/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 0.01] [-0.04 0.96 0.44] [ 0.01 -0.44 0.96]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1795.6005859375\r\n",
      "rescale:  min: 0.0  max: 1023.4923095703125  scale: 0.24914696243008932\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4176/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.07batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-033s4176/ses-201209071405590/anat/sub-033s4176_ses-201209071405590_T1w.nii finished in 5.9468 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.18batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-033s4176/ses-201209071405590/anat/sub-033s4176_ses-201209071405590_T1w.nii finished in 6.2172 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.09batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-033s4176/ses-201209071405590/anat/sub-033s4176_ses-201209071405590_T1w.nii finished in 5.8066 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4176/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20261\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4176/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/033s4176/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.13\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/033s4176/mri/orig.mgz --rescale /output/033s4176/mri/orig_nu.mgz --aseg /output/033s4176/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/033s4176/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/033s4176/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/033s4176/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 76.44\r\n",
      "- m: 1.3345\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/033s4176/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/033s4176/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/033s4176/stats/aseg+DKT.stats --normfile /output/033s4176/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 033s4176 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/033s4176/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/033s4176/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.50 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:17:27 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s4216/ses-201209170922460/anat/sub-073s4216_ses-201209170922460_T1w.nii --asegdkt_segfile /output/073s4216/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s4216/mri/orig.mgz --brainmask_name /output/073s4216/mri/mask.mgz --aseg_name /output/073s4216/mri/aseg.auto_noCCseg.mgz --sid 073s4216 --seg_log /output/073s4216/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s4216/ses-201209170922460/anat/sub-073s4216_ses-201209170922460_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s4216/ses-201209170922460/anat/sub-073s4216_ses-201209170922460_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s4216/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4216/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.1 -0.01] [ 0.11 1.03 0.22] [-0.01 -0.22 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1816.495361328125\r\n",
      "rescale:  min: 0.0  max: 1048.1177978515625  scale: 0.24329326390859918\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4216/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.75batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s4216/ses-201209170922460/anat/sub-073s4216_ses-201209170922460_T1w.nii finished in 5.9906 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.05batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s4216/ses-201209170922460/anat/sub-073s4216_ses-201209170922460_T1w.nii finished in 6.2366 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.00batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s4216/ses-201209170922460/anat/sub-073s4216_ses-201209170922460_T1w.nii finished in 5.8186 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4216/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  22285\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4216/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4216/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.27\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s4216/mri/orig.mgz --rescale /output/073s4216/mri/orig_nu.mgz --aseg /output/073s4216/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s4216/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s4216/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s4216/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 57.69\r\n",
      "- m: 1.7682\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s4216/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s4216/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s4216/stats/aseg+DKT.stats --normfile /output/073s4216/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s4216 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s4216/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s4216/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.87 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:20:37 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-003s4152/ses-201108300937460/anat/sub-003s4152_ses-201108300937460_T1w.nii --asegdkt_segfile /output/003s4152/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/003s4152/mri/orig.mgz --brainmask_name /output/003s4152/mri/mask.mgz --aseg_name /output/003s4152/mri/aseg.auto_noCCseg.mgz --sid 003s4152 --seg_log /output/003s4152/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-003s4152/ses-201108300937460/anat/sub-003s4152_ses-201108300937460_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-003s4152/ses-201108300937460/anat/sub-003s4152_ses-201108300937460_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/003s4152/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4152/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 -0.02] [ 0.03 1.01 0.09] [ 0.02 -0.09 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 30595.05078125\r\n",
      "rescale:  min: 0.0  max: 7159.24169921875  scale: 0.03561829739982473\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4152/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-003s4152/ses-201108300937460/anat/sub-003s4152_ses-201108300937460_T1w.nii finished in 5.9813 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.15batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-003s4152/ses-201108300937460/anat/sub-003s4152_ses-201108300937460_T1w.nii finished in 6.2218 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.01batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-003s4152/ses-201108300937460/anat/sub-003s4152_ses-201108300937460_T1w.nii finished in 5.8177 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4152/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19887\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4152/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4152/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.05\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/003s4152/mri/orig.mgz --rescale /output/003s4152/mri/orig_nu.mgz --aseg /output/003s4152/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/003s4152/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/003s4152/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/003s4152/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 50.53\r\n",
      "- m: 2.0185\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/003s4152/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/003s4152/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/003s4152/stats/aseg+DKT.stats --normfile /output/003s4152/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 003s4152 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/003s4152/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/003s4152/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.75 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:23:45 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-127s4210/ses-201210011338490/anat/sub-127s4210_ses-201210011338490_T1w.nii --asegdkt_segfile /output/127s4210/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/127s4210/mri/orig.mgz --brainmask_name /output/127s4210/mri/mask.mgz --aseg_name /output/127s4210/mri/aseg.auto_noCCseg.mgz --sid 127s4210 --seg_log /output/127s4210/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-127s4210/ses-201210011338490/anat/sub-127s4210_ses-201210011338490_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-127s4210/ses-201210011338490/anat/sub-127s4210_ses-201210011338490_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/127s4210/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4210/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.20e+00 -1.90e-02 -1.67e-04] [ 2.13e-02 1.04e+00 1.85e-01] [-3.62e-03 -1.85e-01 1.04e+00]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 13775.1015625\r\n",
      "rescale:  min: 0.0  max: 6625.82373046875  scale: 0.03848578084372911\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4210/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-127s4210/ses-201210011338490/anat/sub-127s4210_ses-201210011338490_T1w.nii finished in 5.9658 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.42batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-127s4210/ses-201210011338490/anat/sub-127s4210_ses-201210011338490_T1w.nii finished in 6.1818 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.19batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-127s4210/ses-201210011338490/anat/sub-127s4210_ses-201210011338490_T1w.nii finished in 5.7937 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4210/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  23712\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4210/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4210/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.23\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/127s4210/mri/orig.mgz --rescale /output/127s4210/mri/orig_nu.mgz --aseg /output/127s4210/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/127s4210/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/127s4210/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/127s4210/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 59.29\r\n",
      "- m: 1.7205\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/127s4210/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/127s4210/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/127s4210/stats/aseg+DKT.stats --normfile /output/127s4210/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 127s4210 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/127s4210/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/127s4210/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.46 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:26:53 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-021s4254/ses-201210031003160/anat/sub-021s4254_ses-201210031003160_T1w.nii --asegdkt_segfile /output/021s4254/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/021s4254/mri/orig.mgz --brainmask_name /output/021s4254/mri/mask.mgz --aseg_name /output/021s4254/mri/aseg.auto_noCCseg.mgz --sid 021s4254 --seg_log /output/021s4254/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-021s4254/ses-201210031003160/anat/sub-021s4254_ses-201210031003160_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-021s4254/ses-201210031003160/anat/sub-021s4254_ses-201210031003160_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/021s4254/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4254/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.07 -0.06] [ 0.1 0.97 0.41] [ 0.04 -0.42 0.97]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 26414.693359375\r\n",
      "rescale:  min: 0.0  max: 8928.166015625  scale: 0.028561296861385612\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4254/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.03batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-021s4254/ses-201210031003160/anat/sub-021s4254_ses-201210031003160_T1w.nii finished in 5.9522 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.98batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-021s4254/ses-201210031003160/anat/sub-021s4254_ses-201210031003160_T1w.nii finished in 6.2474 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.35batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-021s4254/ses-201210031003160/anat/sub-021s4254_ses-201210031003160_T1w.nii finished in 5.7739 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4254/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17141\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4254/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4254/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.91\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/021s4254/mri/orig.mgz --rescale /output/021s4254/mri/orig_nu.mgz --aseg /output/021s4254/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/021s4254/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/021s4254/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/021s4254/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 46.68\r\n",
      "- m: 2.1851\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/021s4254/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/021s4254/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/021s4254/stats/aseg+DKT.stats --normfile /output/021s4254/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 021s4254 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/021s4254/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/021s4254/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.72 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:30:01 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-098s4201/ses-201109141625250/anat/sub-098s4201_ses-201109141625250_T1w.nii --asegdkt_segfile /output/098s4201/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/098s4201/mri/orig.mgz --brainmask_name /output/098s4201/mri/mask.mgz --aseg_name /output/098s4201/mri/aseg.auto_noCCseg.mgz --sid 098s4201 --seg_log /output/098s4201/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-098s4201/ses-201109141625250/anat/sub-098s4201_ses-201109141625250_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-098s4201/ses-201109141625250/anat/sub-098s4201_ses-201109141625250_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/098s4201/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4201/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.08 -0. ] [ 0.09 0.99 0.23] [-0.02 -0.23 0.99]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 9073.5625\r\n",
      "rescale:  min: 0.0  max: 6024.845703125  scale: 0.04232473536504598\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4201/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.11batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-098s4201/ses-201109141625250/anat/sub-098s4201_ses-201109141625250_T1w.nii finished in 5.9406 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.91batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-098s4201/ses-201109141625250/anat/sub-098s4201_ses-201109141625250_T1w.nii finished in 6.2591 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.17batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-098s4201/ses-201109141625250/anat/sub-098s4201_ses-201109141625250_T1w.nii finished in 5.7963 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4201/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16371\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4201/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4201/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.95\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/098s4201/mri/orig.mgz --rescale /output/098s4201/mri/orig_nu.mgz --aseg /output/098s4201/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/098s4201/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/098s4201/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/098s4201/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 73.69\r\n",
      "- m: 1.3841\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/098s4201/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/098s4201/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/098s4201/stats/aseg+DKT.stats --normfile /output/098s4201/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 098s4201 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/098s4201/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/098s4201/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.40 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:33:09 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-129s4287/ses-201110201035370/anat/sub-129s4287_ses-201110201035370_T1w.nii --asegdkt_segfile /output/129s4287/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/129s4287/mri/orig.mgz --brainmask_name /output/129s4287/mri/mask.mgz --aseg_name /output/129s4287/mri/aseg.auto_noCCseg.mgz --sid 129s4287 --seg_log /output/129s4287/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-129s4287/ses-201110201035370/anat/sub-129s4287_ses-201110201035370_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-129s4287/ses-201110201035370/anat/sub-129s4287_ses-201110201035370_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/129s4287/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4287/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.20e+00 -7.57e-04 2.83e-02] [-7.11e-04 1.05e+00 5.14e-02] [-3.23e-02 -5.14e-02 1.05e+00]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 9542.0244140625\r\n",
      "rescale:  min: 0.0  max: 6832.08935546875  scale: 0.03732386781444612\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4287/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-129s4287/ses-201110201035370/anat/sub-129s4287_ses-201110201035370_T1w.nii finished in 5.9807 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.87batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-129s4287/ses-201110201035370/anat/sub-129s4287_ses-201110201035370_T1w.nii finished in 6.2651 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.19batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-129s4287/ses-201110201035370/anat/sub-129s4287_ses-201110201035370_T1w.nii finished in 5.7937 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4287/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17759\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4287/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4287/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.99\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/129s4287/mri/orig.mgz --rescale /output/129s4287/mri/orig_nu.mgz --aseg /output/129s4287/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/129s4287/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/129s4287/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/129s4287/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 43.87\r\n",
      "- m: 2.3252\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/129s4287/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/129s4287/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/129s4287/stats/aseg+DKT.stats --normfile /output/129s4287/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 129s4287 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/129s4287/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/129s4287/stats/aseg+DKT.stats.\r\n",
      "Calculation took 13.39 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:36:20 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-094s4282/ses-201110200929470/anat/sub-094s4282_ses-201110200929470_T1w.nii --asegdkt_segfile /output/094s4282/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/094s4282/mri/orig.mgz --brainmask_name /output/094s4282/mri/mask.mgz --aseg_name /output/094s4282/mri/aseg.auto_noCCseg.mgz --sid 094s4282 --seg_log /output/094s4282/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-094s4282/ses-201110200929470/anat/sub-094s4282_ses-201110200929470_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-094s4282/ses-201110200929470/anat/sub-094s4282_ses-201110200929470_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/094s4282/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4282/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.09 0. ] [ 0.09 0.89 0.48] [-0.05 -0.48 0.9 ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 48417.40625\r\n",
      "rescale:  min: 0.0  max: 13363.2041015625  scale: 0.01908224989021787\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4282/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.02batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-094s4282/ses-201110200929470/anat/sub-094s4282_ses-201110200929470_T1w.nii finished in 5.9527 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.35batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-094s4282/ses-201110200929470/anat/sub-094s4282_ses-201110200929470_T1w.nii finished in 6.1913 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.33batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-094s4282/ses-201110200929470/anat/sub-094s4282_ses-201110200929470_T1w.nii finished in 5.7755 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4282/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21370\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4282/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/094s4282/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.23\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/094s4282/mri/orig.mgz --rescale /output/094s4282/mri/orig_nu.mgz --aseg /output/094s4282/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/094s4282/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/094s4282/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/094s4282/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 40.26\r\n",
      "- m: 2.5337\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/094s4282/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/094s4282/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/094s4282/stats/aseg+DKT.stats --normfile /output/094s4282/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 094s4282 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/094s4282/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/094s4282/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.88 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:39:28 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-116s4338/ses-201211071317040/anat/sub-116s4338_ses-201211071317040_T1w.nii --asegdkt_segfile /output/116s4338/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/116s4338/mri/orig.mgz --brainmask_name /output/116s4338/mri/mask.mgz --aseg_name /output/116s4338/mri/aseg.auto_noCCseg.mgz --sid 116s4338 --seg_log /output/116s4338/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-116s4338/ses-201211071317040/anat/sub-116s4338_ses-201211071317040_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-116s4338/ses-201211071317040/anat/sub-116s4338_ses-201211071317040_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/116s4338/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4338/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.06 -0. ] [-0.06 1.05 0.05] [ 0.01 -0.05 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1900.215087890625\r\n",
      "rescale:  min: 0.0  max: 1168.63232421875  scale: 0.21820378806522547\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4338/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.14batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-116s4338/ses-201211071317040/anat/sub-116s4338_ses-201211071317040_T1w.nii finished in 5.9360 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 39.42batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-116s4338/ses-201211071317040/anat/sub-116s4338_ses-201211071317040_T1w.nii finished in 6.4953 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.04batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-116s4338/ses-201211071317040/anat/sub-116s4338_ses-201211071317040_T1w.nii finished in 5.8144 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4338/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16554\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4338/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/116s4338/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.98\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/116s4338/mri/orig.mgz --rescale /output/116s4338/mri/orig_nu.mgz --aseg /output/116s4338/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/116s4338/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/116s4338/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/116s4338/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 57.32\r\n",
      "- m: 1.7795\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/116s4338/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/116s4338/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/116s4338/stats/aseg+DKT.stats --normfile /output/116s4338/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 116s4338 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/116s4338/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/116s4338/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.36 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:42:37 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s4360/ses-201302251204230/anat/sub-073s4360_ses-201302251204230_T1w.nii --asegdkt_segfile /output/073s4360/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s4360/mri/orig.mgz --brainmask_name /output/073s4360/mri/mask.mgz --aseg_name /output/073s4360/mri/aseg.auto_noCCseg.mgz --sid 073s4360 --seg_log /output/073s4360/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s4360/ses-201302251204230/anat/sub-073s4360_ses-201302251204230_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s4360/ses-201302251204230/anat/sub-073s4360_ses-201302251204230_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s4360/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4360/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.1 0.04] [ 0.1 1.04 0.15] [-0.06 -0.15 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 3841.888916015625\r\n",
      "rescale:  min: 0.0  max: 1402.2894287109375  scale: 0.18184548409126225\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4360/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s4360/ses-201302251204230/anat/sub-073s4360_ses-201302251204230_T1w.nii finished in 5.9566 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.28batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s4360/ses-201302251204230/anat/sub-073s4360_ses-201302251204230_T1w.nii finished in 6.2020 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.07batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s4360/ses-201302251204230/anat/sub-073s4360_ses-201302251204230_T1w.nii finished in 5.8100 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4360/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20377\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4360/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s4360/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.11\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s4360/mri/orig.mgz --rescale /output/073s4360/mri/orig_nu.mgz --aseg /output/073s4360/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s4360/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s4360/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s4360/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 39.64\r\n",
      "- m: 2.5732\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s4360/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s4360/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s4360/stats/aseg+DKT.stats --normfile /output/073s4360/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s4360 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s4360/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s4360/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.09 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:45:44 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-129s4369/ses-201111170919350/anat/sub-129s4369_ses-201111170919350_T1w.nii --asegdkt_segfile /output/129s4369/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/129s4369/mri/orig.mgz --brainmask_name /output/129s4369/mri/mask.mgz --aseg_name /output/129s4369/mri/aseg.auto_noCCseg.mgz --sid 129s4369 --seg_log /output/129s4369/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-129s4369/ses-201111170919350/anat/sub-129s4369_ses-201111170919350_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-129s4369/ses-201111170919350/anat/sub-129s4369_ses-201111170919350_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/129s4369/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4369/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 -0.06] [ 0.04 1.02 0.27] [ 0.06 -0.27 1.02]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 27515.015625\r\n",
      "rescale:  min: 0.0  max: 7484.08447265625  scale: 0.03407230382442429\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4369/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.67batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-129s4369/ses-201111170919350/anat/sub-129s4369_ses-201111170919350_T1w.nii finished in 6.0025 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.08batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-129s4369/ses-201111170919350/anat/sub-129s4369_ses-201111170919350_T1w.nii finished in 6.2333 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.17batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-129s4369/ses-201111170919350/anat/sub-129s4369_ses-201111170919350_T1w.nii finished in 5.7966 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4369/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18708\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4369/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/129s4369/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.04\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/129s4369/mri/orig.mgz --rescale /output/129s4369/mri/orig_nu.mgz --aseg /output/129s4369/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/129s4369/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/129s4369/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/129s4369/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 48.17\r\n",
      "- m: 2.1173\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/129s4369/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/129s4369/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/129s4369/stats/aseg+DKT.stats --normfile /output/129s4369/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 129s4369 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/129s4369/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/129s4369/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.74 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:48:53 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-003s4350/ses-201212061620080/anat/sub-003s4350_ses-201212061620080_T1w.nii --asegdkt_segfile /output/003s4350/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/003s4350/mri/orig.mgz --brainmask_name /output/003s4350/mri/mask.mgz --aseg_name /output/003s4350/mri/aseg.auto_noCCseg.mgz --sid 003s4350 --seg_log /output/003s4350/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-003s4350/ses-201212061620080/anat/sub-003s4350_ses-201212061620080_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-003s4350/ses-201212061620080/anat/sub-003s4350_ses-201212061620080_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/003s4350/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4350/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.04 0.02] [ 0.04 1.05 0.1 ] [-0.03 -0.1 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 16898.330078125\r\n",
      "rescale:  min: 0.0  max: 6674.84033203125  scale: 0.03820316102189067\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4350/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-003s4350/ses-201212061620080/anat/sub-003s4350_ses-201212061620080_T1w.nii finished in 5.9575 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.05batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-003s4350/ses-201212061620080/anat/sub-003s4350_ses-201212061620080_T1w.nii finished in 6.2366 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.24batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-003s4350/ses-201212061620080/anat/sub-003s4350_ses-201212061620080_T1w.nii finished in 5.7871 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4350/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21018\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4350/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/003s4350/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.16\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/003s4350/mri/orig.mgz --rescale /output/003s4350/mri/orig_nu.mgz --aseg /output/003s4350/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/003s4350/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/003s4350/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/003s4350/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 50.12\r\n",
      "- m: 2.0349\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/003s4350/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/003s4350/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/003s4350/stats/aseg+DKT.stats --normfile /output/003s4350/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 003s4350 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/003s4350/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/003s4350/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.36 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:52:01 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-035s4256/ses-201301141553360/anat/sub-035s4256_ses-201301141553360_T1w.nii --asegdkt_segfile /output/035s4256/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/035s4256/mri/orig.mgz --brainmask_name /output/035s4256/mri/mask.mgz --aseg_name /output/035s4256/mri/aseg.auto_noCCseg.mgz --sid 035s4256 --seg_log /output/035s4256/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-035s4256/ses-201301141553360/anat/sub-035s4256_ses-201301141553360_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-035s4256/ses-201301141553360/anat/sub-035s4256_ses-201301141553360_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/035s4256/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4256/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 -0.01] [ 0.04 1.05 0.04] [ 0.01 -0.05 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1772.45654296875\r\n",
      "rescale:  min: 0.0  max: 1143.2344970703125  scale: 0.22305135180356328\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4256/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.65batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-035s4256/ses-201301141553360/anat/sub-035s4256_ses-201301141553360_T1w.nii finished in 6.0052 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.88batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-035s4256/ses-201301141553360/anat/sub-035s4256_ses-201301141553360_T1w.nii finished in 6.2628 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.16batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-035s4256/ses-201301141553360/anat/sub-035s4256_ses-201301141553360_T1w.nii finished in 5.7980 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4256/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19393\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4256/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/035s4256/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.12\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/035s4256/mri/orig.mgz --rescale /output/035s4256/mri/orig_nu.mgz --aseg /output/035s4256/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/035s4256/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/035s4256/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/035s4256/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 55.11\r\n",
      "- m: 1.8509\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/035s4256/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/035s4256/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/035s4256/stats/aseg+DKT.stats --normfile /output/035s4256/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 035s4256 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/035s4256/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/035s4256/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.88 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:55:09 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-032s4429/ses-201302071257230/anat/sub-032s4429_ses-201302071257230_T1w.nii --asegdkt_segfile /output/032s4429/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/032s4429/mri/orig.mgz --brainmask_name /output/032s4429/mri/mask.mgz --aseg_name /output/032s4429/mri/aseg.auto_noCCseg.mgz --sid 032s4429 --seg_log /output/032s4429/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-032s4429/ses-201302071257230/anat/sub-032s4429_ses-201302071257230_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-032s4429/ses-201302071257230/anat/sub-032s4429_ses-201302071257230_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/032s4429/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/032s4429/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.08 -0.02] [ 0.09 1.02 0.24] [-0. -0.25 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1133.9569091796875\r\n",
      "rescale:  min: 0.0  max: 758.6171875  scale: 0.3361379154094106\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/032s4429/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.81batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-032s4429/ses-201302071257230/anat/sub-032s4429_ses-201302071257230_T1w.nii finished in 5.9820 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.87batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-032s4429/ses-201302071257230/anat/sub-032s4429_ses-201302071257230_T1w.nii finished in 6.2646 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.96batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-032s4429/ses-201302071257230/anat/sub-032s4429_ses-201302071257230_T1w.nii finished in 5.8240 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/032s4429/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19182\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/032s4429/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/032s4429/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.17\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/032s4429/mri/orig.mgz --rescale /output/032s4429/mri/orig_nu.mgz --aseg /output/032s4429/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/032s4429/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/032s4429/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/032s4429/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 69.41\r\n",
      "- m: 1.4695\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/032s4429/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/032s4429/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/032s4429/stats/aseg+DKT.stats --normfile /output/032s4429/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 032s4429 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/032s4429/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/032s4429/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.07 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 05:58:17 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-037s4410/ses-201212191007000/anat/sub-037s4410_ses-201212191007000_T1w.nii --asegdkt_segfile /output/037s4410/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/037s4410/mri/orig.mgz --brainmask_name /output/037s4410/mri/mask.mgz --aseg_name /output/037s4410/mri/aseg.auto_noCCseg.mgz --sid 037s4410 --seg_log /output/037s4410/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-037s4410/ses-201212191007000/anat/sub-037s4410_ses-201212191007000_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-037s4410/ses-201212191007000/anat/sub-037s4410_ses-201212191007000_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/037s4410/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4410/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.08 0.01] [ 0.08 1.02 0.27] [-0.04 -0.27 1.02]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1163.7442626953125\r\n",
      "rescale:  min: 0.0  max: 798.3285522460938  scale: 0.319417361789903\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4410/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.60batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-037s4410/ses-201212191007000/anat/sub-037s4410_ses-201212191007000_T1w.nii finished in 6.0113 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.21batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-037s4410/ses-201212191007000/anat/sub-037s4410_ses-201212191007000_T1w.nii finished in 6.2132 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.06batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-037s4410/ses-201212191007000/anat/sub-037s4410_ses-201212191007000_T1w.nii finished in 5.8112 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4410/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17257\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4410/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s4410/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.99\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/037s4410/mri/orig.mgz --rescale /output/037s4410/mri/orig_nu.mgz --aseg /output/037s4410/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/037s4410/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/037s4410/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/037s4410/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 80.93\r\n",
      "- m: 1.2604\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/037s4410/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/037s4410/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/037s4410/stats/aseg+DKT.stats --normfile /output/037s4410/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 037s4410 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/037s4410/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/037s4410/stats/aseg+DKT.stats.\r\n",
      "Calculation took 8.97 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:01:23 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-099s4480/ses-201302260748180/anat/sub-099s4480_ses-201302260748180_T1w.nii --asegdkt_segfile /output/099s4480/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/099s4480/mri/orig.mgz --brainmask_name /output/099s4480/mri/mask.mgz --aseg_name /output/099s4480/mri/aseg.auto_noCCseg.mgz --sid 099s4480 --seg_log /output/099s4480/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-099s4480/ses-201302260748180/anat/sub-099s4480_ses-201302260748180_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-099s4480/ses-201302260748180/anat/sub-099s4480_ses-201302260748180_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/099s4480/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4480/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19e+00 -4.60e-04 9.66e-02] [-2.77e-03 1.05e+00 3.15e-02] [-1.10e-01 -3.16e-02 1.05e+00]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 3718.048583984375\r\n",
      "rescale:  min: 0.0  max: 1260.41845703125  scale: 0.20231376220927372\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4480/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.00batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-099s4480/ses-201302260748180/anat/sub-099s4480_ses-201302260748180_T1w.nii finished in 5.9554 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-099s4480/ses-201302260748180/anat/sub-099s4480_ses-201302260748180_T1w.nii finished in 6.2569 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.98batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-099s4480/ses-201302260748180/anat/sub-099s4480_ses-201302260748180_T1w.nii finished in 5.8220 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4480/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16855\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4480/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/099s4480/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.95\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/099s4480/mri/orig.mgz --rescale /output/099s4480/mri/orig_nu.mgz --aseg /output/099s4480/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/099s4480/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/099s4480/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/099s4480/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 48.73\r\n",
      "- m: 2.0932\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/099s4480/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/099s4480/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/099s4480/stats/aseg+DKT.stats --normfile /output/099s4480/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 099s4480 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/099s4480/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/099s4480/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.58 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:04:31 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-098s4506/ses-201202241316090/anat/sub-098s4506_ses-201202241316090_T1w.nii --asegdkt_segfile /output/098s4506/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/098s4506/mri/orig.mgz --brainmask_name /output/098s4506/mri/mask.mgz --aseg_name /output/098s4506/mri/aseg.auto_noCCseg.mgz --sid 098s4506 --seg_log /output/098s4506/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-098s4506/ses-201202241316090/anat/sub-098s4506_ses-201202241316090_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-098s4506/ses-201202241316090/anat/sub-098s4506_ses-201202241316090_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/098s4506/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4506/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 -0.04] [ 0.03 1.01 0.14] [ 0.04 -0.14 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 12201.638671875\r\n",
      "rescale:  min: 0.0  max: 6466.86865234375  scale: 0.03943175804376076\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4506/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.06batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-098s4506/ses-201202241316090/anat/sub-098s4506_ses-201202241316090_T1w.nii finished in 5.9479 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.97batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-098s4506/ses-201202241316090/anat/sub-098s4506_ses-201202241316090_T1w.nii finished in 6.2488 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.25batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-098s4506/ses-201202241316090/anat/sub-098s4506_ses-201202241316090_T1w.nii finished in 5.7864 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4506/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19506\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4506/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/098s4506/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.11\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/098s4506/mri/orig.mgz --rescale /output/098s4506/mri/orig_nu.mgz --aseg /output/098s4506/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/098s4506/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/098s4506/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/098s4506/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 58.61\r\n",
      "- m: 1.7402\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/098s4506/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/098s4506/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/098s4506/stats/aseg+DKT.stats --normfile /output/098s4506/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 098s4506 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/098s4506/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/098s4506/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.94 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:07:39 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-014s4615/ses-201304111133030/anat/sub-014s4615_ses-201304111133030_T1w.nii --asegdkt_segfile /output/014s4615/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/014s4615/mri/orig.mgz --brainmask_name /output/014s4615/mri/mask.mgz --aseg_name /output/014s4615/mri/aseg.auto_noCCseg.mgz --sid 014s4615 --seg_log /output/014s4615/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-014s4615/ses-201304111133030/anat/sub-014s4615_ses-201304111133030_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-014s4615/ses-201304111133030/anat/sub-014s4615_ses-201304111133030_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/014s4615/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4615/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 -0.01] [ 0.03 1.04 0.18] [ 0.01 -0.18 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1731.8115234375\r\n",
      "rescale:  min: 0.0  max: 1108.359375  scale: 0.23006978219496724\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4615/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.87batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-014s4615/ses-201304111133030/anat/sub-014s4615_ses-201304111133030_T1w.nii finished in 5.9740 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.88batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-014s4615/ses-201304111133030/anat/sub-014s4615_ses-201304111133030_T1w.nii finished in 6.2638 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.29batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-014s4615/ses-201304111133030/anat/sub-014s4615_ses-201304111133030_T1w.nii finished in 5.7812 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4615/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20721\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4615/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4615/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.05\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/014s4615/mri/orig.mgz --rescale /output/014s4615/mri/orig_nu.mgz --aseg /output/014s4615/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/014s4615/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/014s4615/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/014s4615/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 45.52\r\n",
      "- m: 2.2409\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/014s4615/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/014s4615/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/014s4615/stats/aseg+DKT.stats --normfile /output/014s4615/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 014s4615 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/014s4615/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/014s4615/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.29 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:10:48 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-127s4645/ses-201204041407240/anat/sub-127s4645_ses-201204041407240_T1w.nii --asegdkt_segfile /output/127s4645/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/127s4645/mri/orig.mgz --brainmask_name /output/127s4645/mri/mask.mgz --aseg_name /output/127s4645/mri/aseg.auto_noCCseg.mgz --sid 127s4645 --seg_log /output/127s4645/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-127s4645/ses-201204041407240/anat/sub-127s4645_ses-201204041407240_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-127s4645/ses-201204041407240/anat/sub-127s4645_ses-201204041407240_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/127s4645/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4645/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 -0.08] [ 0.04 1.05 0.09] [ 0.08 -0.09 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 12248.1064453125\r\n",
      "rescale:  min: 0.0  max: 7042.6611328125  scale: 0.03620790425538553\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4645/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.34batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-127s4645/ses-201204041407240/anat/sub-127s4645_ses-201204041407240_T1w.nii finished in 5.9087 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.93batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-127s4645/ses-201204041407240/anat/sub-127s4645_ses-201204041407240_T1w.nii finished in 6.2554 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.13batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-127s4645/ses-201204041407240/anat/sub-127s4645_ses-201204041407240_T1w.nii finished in 5.8019 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4645/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19146\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4645/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4645/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.11\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/127s4645/mri/orig.mgz --rescale /output/127s4645/mri/orig_nu.mgz --aseg /output/127s4645/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/127s4645/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/127s4645/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/127s4645/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 49.10\r\n",
      "- m: 2.0774\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/127s4645/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/127s4645/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/127s4645/stats/aseg+DKT.stats --normfile /output/127s4645/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 127s4645 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/127s4645/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/127s4645/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.09 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:13:56 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-137s4672/ses-201204160838030/anat/sub-137s4672_ses-201204160838030_T1w.nii --asegdkt_segfile /output/137s4672/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/137s4672/mri/orig.mgz --brainmask_name /output/137s4672/mri/mask.mgz --aseg_name /output/137s4672/mri/aseg.auto_noCCseg.mgz --sid 137s4672 --seg_log /output/137s4672/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-137s4672/ses-201204160838030/anat/sub-137s4672_ses-201204160838030_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-137s4672/ses-201204160838030/anat/sub-137s4672_ses-201204160838030_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/137s4672/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4672/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 0. ] [-0.04 0.98 0.39] [ 0.01 -0.39 0.98]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 592.8229370117188\r\n",
      "rescale:  min: 0.0  max: 224.6798858642578  scale: 1.1349480574067068\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4672/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-137s4672/ses-201204160838030/anat/sub-137s4672_ses-201204160838030_T1w.nii finished in 5.9634 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.73batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-137s4672/ses-201204160838030/anat/sub-137s4672_ses-201204160838030_T1w.nii finished in 6.2858 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.14batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-137s4672/ses-201204160838030/anat/sub-137s4672_ses-201204160838030_T1w.nii finished in 5.8005 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4672/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20678\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4672/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s4672/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.04\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/137s4672/mri/orig.mgz --rescale /output/137s4672/mri/orig_nu.mgz --aseg /output/137s4672/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/137s4672/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/137s4672/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/137s4672/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 64.24\r\n",
      "- m: 1.5878\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/137s4672/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/137s4672/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/137s4672/stats/aseg+DKT.stats --normfile /output/137s4672/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 137s4672 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/137s4672/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/137s4672/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.60 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:17:04 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-126s4686/ses-201204271102150/anat/sub-126s4686_ses-201204271102150_T1w.nii --asegdkt_segfile /output/126s4686/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/126s4686/mri/orig.mgz --brainmask_name /output/126s4686/mri/mask.mgz --aseg_name /output/126s4686/mri/aseg.auto_noCCseg.mgz --sid 126s4686 --seg_log /output/126s4686/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-126s4686/ses-201204271102150/anat/sub-126s4686_ses-201204271102150_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-126s4686/ses-201204271102150/anat/sub-126s4686_ses-201204271102150_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/126s4686/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4686/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 -0.05] [ 0.02 1.01 0.12] [ 0.06 -0.12 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 18408.5625\r\n",
      "rescale:  min: 0.0  max: 7639.55322265625  scale: 0.033378915306690835\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4686/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.97batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-126s4686/ses-201204271102150/anat/sub-126s4686_ses-201204271102150_T1w.nii finished in 5.9597 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-126s4686/ses-201204271102150/anat/sub-126s4686_ses-201204271102150_T1w.nii finished in 6.2469 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.18batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-126s4686/ses-201204271102150/anat/sub-126s4686_ses-201204271102150_T1w.nii finished in 5.7953 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4686/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19211\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4686/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/126s4686/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.09\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/126s4686/mri/orig.mgz --rescale /output/126s4686/mri/orig_nu.mgz --aseg /output/126s4686/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/126s4686/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/126s4686/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/126s4686/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 48.54\r\n",
      "- m: 2.1014\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/126s4686/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/126s4686/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/126s4686/stats/aseg+DKT.stats --normfile /output/126s4686/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 126s4686 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/126s4686/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/126s4686/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.95 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:20:12 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-137s0668/ses-201208221249470/anat/sub-137s0668_ses-201208221249470_T1w.nii --asegdkt_segfile /output/137s0668/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/137s0668/mri/orig.mgz --brainmask_name /output/137s0668/mri/mask.mgz --aseg_name /output/137s0668/mri/aseg.auto_noCCseg.mgz --sid 137s0668 --seg_log /output/137s0668/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-137s0668/ses-201208221249470/anat/sub-137s0668_ses-201208221249470_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-137s0668/ses-201208221249470/anat/sub-137s0668_ses-201208221249470_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/137s0668/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0668/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 0.03 0.1 ] [-0.06 1. 0.33] [-0.1 -0.33 1. ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 867.7476806640625\r\n",
      "rescale:  min: 0.0  max: 584.8619384765625  scale: 0.43600033311146774\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0668/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.16batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-137s0668/ses-201208221249470/anat/sub-137s0668_ses-201208221249470_T1w.nii finished in 5.9337 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.60batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-137s0668/ses-201208221249470/anat/sub-137s0668_ses-201208221249470_T1w.nii finished in 6.3065 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.13batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-137s0668/ses-201208221249470/anat/sub-137s0668_ses-201208221249470_T1w.nii finished in 5.8022 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0668/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18603\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0668/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0668/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.04\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/137s0668/mri/orig.mgz --rescale /output/137s0668/mri/orig_nu.mgz --aseg /output/137s0668/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/137s0668/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/137s0668/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/137s0668/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 69.39\r\n",
      "- m: 1.4701\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/137s0668/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/137s0668/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/137s0668/stats/aseg+DKT.stats --normfile /output/137s0668/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 137s0668 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/137s0668/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/137s0668/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.77 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:23:21 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-037s0566/ses-201212061253070/anat/sub-037s0566_ses-201212061253070_T1w.nii --asegdkt_segfile /output/037s0566/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/037s0566/mri/orig.mgz --brainmask_name /output/037s0566/mri/mask.mgz --aseg_name /output/037s0566/mri/aseg.auto_noCCseg.mgz --sid 037s0566 --seg_log /output/037s0566/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-037s0566/ses-201212061253070/anat/sub-037s0566_ses-201212061253070_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-037s0566/ses-201212061253070/anat/sub-037s0566_ses-201212061253070_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/037s0566/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s0566/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.02 0.07] [-0.03 1.05 0.14] [-0.07 -0.14 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1220.42431640625\r\n",
      "rescale:  min: 0.0  max: 843.313232421875  scale: 0.30237874872148807\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s0566/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.87batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-037s0566/ses-201212061253070/anat/sub-037s0566_ses-201212061253070_T1w.nii finished in 5.9739 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.34batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-037s0566/ses-201212061253070/anat/sub-037s0566_ses-201212061253070_T1w.nii finished in 6.3465 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.08batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-037s0566/ses-201212061253070/anat/sub-037s0566_ses-201212061253070_T1w.nii finished in 5.8088 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s0566/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16172\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s0566/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s0566/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.96\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/037s0566/mri/orig.mgz --rescale /output/037s0566/mri/orig_nu.mgz --aseg /output/037s0566/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/037s0566/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/037s0566/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/037s0566/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 62.76\r\n",
      "- m: 1.6252\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/037s0566/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/037s0566/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/037s0566/stats/aseg+DKT.stats --normfile /output/037s0566/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 037s0566 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/037s0566/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/037s0566/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.61 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:26:30 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-016s4591/ses-201304011455320/anat/sub-016s4591_ses-201304011455320_T1w.nii --asegdkt_segfile /output/016s4591/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/016s4591/mri/orig.mgz --brainmask_name /output/016s4591/mri/mask.mgz --aseg_name /output/016s4591/mri/aseg.auto_noCCseg.mgz --sid 016s4591 --seg_log /output/016s4591/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-016s4591/ses-201304011455320/anat/sub-016s4591_ses-201304011455320_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-016s4591/ses-201304011455320/anat/sub-016s4591_ses-201304011455320_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/016s4591/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4591/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.18 0.18 -0.05] [-0.2 1.04 0.08] [ 0.08 -0.07 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 8864.8671875\r\n",
      "rescale:  min: 0.0  max: 6400.43408203125  scale: 0.039841047768290255\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4591/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.21batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-016s4591/ses-201304011455320/anat/sub-016s4591_ses-201304011455320_T1w.nii finished in 5.9276 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 41.00batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-016s4591/ses-201304011455320/anat/sub-016s4591_ses-201304011455320_T1w.nii finished in 6.2442 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.32batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-016s4591/ses-201304011455320/anat/sub-016s4591_ses-201304011455320_T1w.nii finished in 5.7770 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4591/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16742\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4591/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/016s4591/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.94\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/016s4591/mri/orig.mgz --rescale /output/016s4591/mri/orig_nu.mgz --aseg /output/016s4591/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/016s4591/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/016s4591/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/016s4591/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 54.49\r\n",
      "- m: 1.8717\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/016s4591/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/016s4591/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/016s4591/stats/aseg+DKT.stats --normfile /output/016s4591/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 016s4591 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/016s4591/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/016s4591/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.53 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:29:36 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-036s4562/ses-201304041541000/anat/sub-036s4562_ses-201304041541000_T1w.nii --asegdkt_segfile /output/036s4562/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/036s4562/mri/orig.mgz --brainmask_name /output/036s4562/mri/mask.mgz --aseg_name /output/036s4562/mri/aseg.auto_noCCseg.mgz --sid 036s4562 --seg_log /output/036s4562/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-036s4562/ses-201304041541000/anat/sub-036s4562_ses-201304041541000_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-036s4562/ses-201304041541000/anat/sub-036s4562_ses-201304041541000_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/036s4562/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/036s4562/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 -0.02] [-0.02 1.03 0.22] [ 0.03 -0.22 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1330.787353515625\r\n",
      "rescale:  min: 0.0  max: 751.8948364257812  scale: 0.33914317221830104\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/036s4562/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.72batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-036s4562/ses-201304041541000/anat/sub-036s4562_ses-201304041541000_T1w.nii finished in 5.9950 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.77batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-036s4562/ses-201304041541000/anat/sub-036s4562_ses-201304041541000_T1w.nii finished in 6.2800 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.08batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-036s4562/ses-201304041541000/anat/sub-036s4562_ses-201304041541000_T1w.nii finished in 5.8079 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/036s4562/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18239\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/036s4562/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/036s4562/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.15\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/036s4562/mri/orig.mgz --rescale /output/036s4562/mri/orig_nu.mgz --aseg /output/036s4562/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/036s4562/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/036s4562/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/036s4562/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 71.16\r\n",
      "- m: 1.4333\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/036s4562/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/036s4562/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/036s4562/stats/aseg+DKT.stats --normfile /output/036s4562/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 036s4562 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/036s4562/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/036s4562/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.03 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:32:45 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-021s4659/ses-201305090858240/anat/sub-021s4659_ses-201305090858240_T1w.nii --asegdkt_segfile /output/021s4659/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/021s4659/mri/orig.mgz --brainmask_name /output/021s4659/mri/mask.mgz --aseg_name /output/021s4659/mri/aseg.auto_noCCseg.mgz --sid 021s4659 --seg_log /output/021s4659/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-021s4659/ses-201305090858240/anat/sub-021s4659_ses-201305090858240_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-021s4659/ses-201305090858240/anat/sub-021s4659_ses-201305090858240_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/021s4659/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4659/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 -0.11 0.04] [ 0.12 0.99 0.2 ] [-0.07 -0.19 1. ]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 20858.08203125\r\n",
      "rescale:  min: 0.0  max: 8447.5234375  scale: 0.030186361942248236\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4659/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.78batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-021s4659/ses-201305090858240/anat/sub-021s4659_ses-201305090858240_T1w.nii finished in 5.9866 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.75batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-021s4659/ses-201305090858240/anat/sub-021s4659_ses-201305090858240_T1w.nii finished in 6.2830 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-021s4659/ses-201305090858240/anat/sub-021s4659_ses-201305090858240_T1w.nii finished in 5.8200 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4659/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21169\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4659/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/021s4659/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.18\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/021s4659/mri/orig.mgz --rescale /output/021s4659/mri/orig_nu.mgz --aseg /output/021s4659/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/021s4659/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/021s4659/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/021s4659/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 20.04\r\n",
      "- m: 5.0901\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/021s4659/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/021s4659/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/021s4659/stats/aseg+DKT.stats --normfile /output/021s4659/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 021s4659 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/021s4659/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/021s4659/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.74 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:35:54 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-014s4668/ses-201305301058270/anat/sub-014s4668_ses-201305301058270_T1w.nii --asegdkt_segfile /output/014s4668/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/014s4668/mri/orig.mgz --brainmask_name /output/014s4668/mri/mask.mgz --aseg_name /output/014s4668/mri/aseg.auto_noCCseg.mgz --sid 014s4668 --seg_log /output/014s4668/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-014s4668/ses-201305301058270/anat/sub-014s4668_ses-201305301058270_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-014s4668/ses-201305301058270/anat/sub-014s4668_ses-201305301058270_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/014s4668/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4668/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.03 -0.03] [ 0.05 0.99 0.35] [ 0.02 -0.35 0.99]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2747.3837890625\r\n",
      "rescale:  min: 0.0  max: 1228.08056640625  scale: 0.20764110024654994\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4668/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-014s4668/ses-201305301058270/anat/sub-014s4668_ses-201305301058270_T1w.nii finished in 5.9631 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.28batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-014s4668/ses-201305301058270/anat/sub-014s4668_ses-201305301058270_T1w.nii finished in 6.3571 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.98batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-014s4668/ses-201305301058270/anat/sub-014s4668_ses-201305301058270_T1w.nii finished in 5.8216 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4668/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17939\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4668/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/014s4668/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.34\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/014s4668/mri/orig.mgz --rescale /output/014s4668/mri/orig_nu.mgz --aseg /output/014s4668/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/014s4668/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/014s4668/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/014s4668/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 51.21\r\n",
      "- m: 1.9917\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/014s4668/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/014s4668/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/014s4668/stats/aseg+DKT.stats --normfile /output/014s4668/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 014s4668 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/014s4668/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/014s4668/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.58 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:39:04 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-068s4431/ses-201202061525130/anat/sub-068s4431_ses-201202061525130_T1w.nii --asegdkt_segfile /output/068s4431/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/068s4431/mri/orig.mgz --brainmask_name /output/068s4431/mri/mask.mgz --aseg_name /output/068s4431/mri/aseg.auto_noCCseg.mgz --sid 068s4431 --seg_log /output/068s4431/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-068s4431/ses-201202061525130/anat/sub-068s4431_ses-201202061525130_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-068s4431/ses-201202061525130/anat/sub-068s4431_ses-201202061525130_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/068s4431/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4431/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 0.01] [ 0.02 1.04 0.17] [-0.02 -0.17 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1858.0517578125\r\n",
      "rescale:  min: 0.0  max: 1049.7991943359375  scale: 0.2429035965885868\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4431/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.41batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-068s4431/ses-201202061525130/anat/sub-068s4431_ses-201202061525130_T1w.nii finished in 5.9001 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.89batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-068s4431/ses-201202061525130/anat/sub-068s4431_ses-201202061525130_T1w.nii finished in 6.2611 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.18batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-068s4431/ses-201202061525130/anat/sub-068s4431_ses-201202061525130_T1w.nii finished in 5.7947 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4431/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20693\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4431/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/068s4431/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.13\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/068s4431/mri/orig.mgz --rescale /output/068s4431/mri/orig_nu.mgz --aseg /output/068s4431/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/068s4431/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/068s4431/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/068s4431/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 64.51\r\n",
      "- m: 1.5811\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/068s4431/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/068s4431/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/068s4431/stats/aseg+DKT.stats --normfile /output/068s4431/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 068s4431 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/068s4431/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/068s4431/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.56 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:42:14 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-127s4604/ses-201203211543180/anat/sub-127s4604_ses-201203211543180_T1w.nii --asegdkt_segfile /output/127s4604/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/127s4604/mri/orig.mgz --brainmask_name /output/127s4604/mri/mask.mgz --aseg_name /output/127s4604/mri/aseg.auto_noCCseg.mgz --sid 127s4604 --seg_log /output/127s4604/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-127s4604/ses-201203211543180/anat/sub-127s4604_ses-201203211543180_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-127s4604/ses-201203211543180/anat/sub-127s4604_ses-201203211543180_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/127s4604/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4604/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.05 -0.04] [ 0.06 1.03 0.23] [ 0.04 -0.23 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 22526.32421875\r\n",
      "rescale:  min: 0.0  max: 6938.10791015625  scale: 0.03675353616606653\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4604/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.15batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-127s4604/ses-201203211543180/anat/sub-127s4604_ses-201203211543180_T1w.nii finished in 5.9355 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.67batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-127s4604/ses-201203211543180/anat/sub-127s4604_ses-201203211543180_T1w.nii finished in 6.2961 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.22batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-127s4604/ses-201203211543180/anat/sub-127s4604_ses-201203211543180_T1w.nii finished in 5.7901 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4604/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  24142\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4604/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/127s4604/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.28\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/127s4604/mri/orig.mgz --rescale /output/127s4604/mri/orig_nu.mgz --aseg /output/127s4604/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/127s4604/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/127s4604/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/127s4604/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 52.35\r\n",
      "- m: 1.9483\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/127s4604/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/127s4604/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/127s4604/stats/aseg+DKT.stats --normfile /output/127s4604/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 127s4604 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/127s4604/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/127s4604/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.81 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:45:24 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-128s4607/ses-201304121516540/anat/sub-128s4607_ses-201304121516540_T1w.nii --asegdkt_segfile /output/128s4607/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/128s4607/mri/orig.mgz --brainmask_name /output/128s4607/mri/mask.mgz --aseg_name /output/128s4607/mri/aseg.auto_noCCseg.mgz --sid 128s4607 --seg_log /output/128s4607/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-128s4607/ses-201304121516540/anat/sub-128s4607_ses-201304121516540_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-128s4607/ses-201304121516540/anat/sub-128s4607_ses-201304121516540_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/128s4607/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4607/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0x1.0\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 0.01] [ 0.01 0.99 0.17] [-0.02 -0.17 0.99]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1843.887451171875\r\n",
      "rescale:  min: 0.0  max: 984.6359252929688  scale: 0.25897897227762356\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4607/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.97batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-128s4607/ses-201304121516540/anat/sub-128s4607_ses-201304121516540_T1w.nii finished in 5.9604 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-128s4607/ses-201304121516540/anat/sub-128s4607_ses-201304121516540_T1w.nii finished in 6.2464 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.26batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-128s4607/ses-201304121516540/anat/sub-128s4607_ses-201304121516540_T1w.nii finished in 5.7849 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4607/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20208\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4607/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4607/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/128s4607/mri/orig.mgz --rescale /output/128s4607/mri/orig_nu.mgz --aseg /output/128s4607/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/128s4607/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/128s4607/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/128s4607/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 56.11\r\n",
      "- m: 1.8179\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/128s4607/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/128s4607/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/128s4607/stats/aseg+DKT.stats --normfile /output/128s4607/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 128s4607 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/128s4607/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/128s4607/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.99 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:48:32 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-009s4543/ses-201303201410350/anat/sub-009s4543_ses-201303201410350_T1w.nii --asegdkt_segfile /output/009s4543/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/009s4543/mri/orig.mgz --brainmask_name /output/009s4543/mri/mask.mgz --aseg_name /output/009s4543/mri/aseg.auto_noCCseg.mgz --sid 009s4543 --seg_log /output/009s4543/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-009s4543/ses-201303201410350/anat/sub-009s4543_ses-201303201410350_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-009s4543/ses-201303201410350/anat/sub-009s4543_ses-201303201410350_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/009s4543/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4543/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.02 -0.07] [-0. 1.02 0.25] [ 0.08 -0.25 1.02]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 5028.22412109375\r\n",
      "rescale:  min: 0.0  max: 1422.9874267578125  scale: 0.17920045898156772\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4543/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-009s4543/ses-201303201410350/anat/sub-009s4543_ses-201303201410350_T1w.nii finished in 5.9856 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.73batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-009s4543/ses-201303201410350/anat/sub-009s4543_ses-201303201410350_T1w.nii finished in 6.2860 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.11batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-009s4543/ses-201303201410350/anat/sub-009s4543_ses-201303201410350_T1w.nii finished in 5.8052 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4543/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18840\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4543/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/009s4543/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/009s4543/mri/orig.mgz --rescale /output/009s4543/mri/orig_nu.mgz --aseg /output/009s4543/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/009s4543/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/009s4543/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/009s4543/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 46.48\r\n",
      "- m: 2.1946\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/009s4543/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/009s4543/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/009s4543/stats/aseg+DKT.stats --normfile /output/009s4543/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 009s4543 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/009s4543/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/009s4543/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.87 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:51:41 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-052s4626/ses-201204031330200/anat/sub-052s4626_ses-201204031330200_T1w.nii --asegdkt_segfile /output/052s4626/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/052s4626/mri/orig.mgz --brainmask_name /output/052s4626/mri/mask.mgz --aseg_name /output/052s4626/mri/aseg.auto_noCCseg.mgz --sid 052s4626 --seg_log /output/052s4626/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-052s4626/ses-201204031330200/anat/sub-052s4626_ses-201204031330200_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-052s4626/ses-201204031330200/anat/sub-052s4626_ses-201204031330200_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/052s4626/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/052s4626/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0547x1.0547\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.03 0. ] [-0.03 1.05 0.04] [-0. -0.04 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 10057.8291015625\r\n",
      "rescale:  min: 0.0  max: 6527.53125  scale: 0.039065305125885075\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/052s4626/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.91batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-052s4626/ses-201204031330200/anat/sub-052s4626_ses-201204031330200_T1w.nii finished in 5.9688 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.50batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-052s4626/ses-201204031330200/anat/sub-052s4626_ses-201204031330200_T1w.nii finished in 6.3216 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.21batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-052s4626/ses-201204031330200/anat/sub-052s4626_ses-201204031330200_T1w.nii finished in 5.7916 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/052s4626/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  16579\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/052s4626/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/052s4626/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.02\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/052s4626/mri/orig.mgz --rescale /output/052s4626/mri/orig_nu.mgz --aseg /output/052s4626/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/052s4626/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/052s4626/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/052s4626/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 53.65\r\n",
      "- m: 1.9014\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/052s4626/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/052s4626/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/052s4626/stats/aseg+DKT.stats --normfile /output/052s4626/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 052s4626 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/052s4626/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/052s4626/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.66 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:54:48 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-128s4603/ses-201204070856120/anat/sub-128s4603_ses-201204070856120_T1w.nii --asegdkt_segfile /output/128s4603/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/128s4603/mri/orig.mgz --brainmask_name /output/128s4603/mri/mask.mgz --aseg_name /output/128s4603/mri/aseg.auto_noCCseg.mgz --sid 128s4603 --seg_log /output/128s4603/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-128s4603/ses-201204070856120/anat/sub-128s4603_ses-201204070856120_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-128s4603/ses-201204070856120/anat/sub-128s4603_ses-201204070856120_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/128s4603/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4603/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.05 -0.04] [-0.04 1.02 0.25] [ 0.05 -0.25 1.02]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1717.2977294921875\r\n",
      "rescale:  min: 0.0  max: 1109.3743896484375  scale: 0.2298592813926504\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4603/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.88batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-128s4603/ses-201204070856120/anat/sub-128s4603_ses-201204070856120_T1w.nii finished in 5.9727 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.65batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-128s4603/ses-201204070856120/anat/sub-128s4603_ses-201204070856120_T1w.nii finished in 6.2988 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.96batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-128s4603/ses-201204070856120/anat/sub-128s4603_ses-201204070856120_T1w.nii finished in 5.8245 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4603/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  20504\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4603/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4603/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.04\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/128s4603/mri/orig.mgz --rescale /output/128s4603/mri/orig_nu.mgz --aseg /output/128s4603/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/128s4603/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/128s4603/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/128s4603/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 56.35\r\n",
      "- m: 1.8101\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/128s4603/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/128s4603/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/128s4603/stats/aseg+DKT.stats --normfile /output/128s4603/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 128s4603 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/128s4603/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/128s4603/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.19 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 06:57:56 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-128s4653/ses-201305311807210/anat/sub-128s4653_ses-201305311807210_T1w.nii --asegdkt_segfile /output/128s4653/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/128s4653/mri/orig.mgz --brainmask_name /output/128s4653/mri/mask.mgz --aseg_name /output/128s4653/mri/aseg.auto_noCCseg.mgz --sid 128s4653 --seg_log /output/128s4653/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-128s4653/ses-201305311807210/anat/sub-128s4653_ses-201305311807210_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-128s4653/ses-201305311807210/anat/sub-128s4653_ses-201305311807210_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/128s4653/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4653/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0x1.0\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.04 0.02] [-0.05 0.97 0.25] [-0.01 -0.25 0.97]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1942.2578125\r\n",
      "rescale:  min: 0.0  max: 992.4937133789062  scale: 0.25692857955932275\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4653/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 42.66batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-128s4653/ses-201305311807210/anat/sub-128s4653_ses-201305311807210_T1w.nii finished in 6.0034 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.46batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-128s4653/ses-201305311807210/anat/sub-128s4653_ses-201305311807210_T1w.nii finished in 6.3279 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.12batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-128s4653/ses-201305311807210/anat/sub-128s4653_ses-201305311807210_T1w.nii finished in 5.8032 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4653/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18968\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4653/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/128s4653/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.05\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/128s4653/mri/orig.mgz --rescale /output/128s4653/mri/orig_nu.mgz --aseg /output/128s4653/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/128s4653/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/128s4653/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/128s4653/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 49.53\r\n",
      "- m: 2.0592\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/128s4653/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/128s4653/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/128s4653/stats/aseg+DKT.stats --normfile /output/128s4653/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 128s4653 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/128s4653/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/128s4653/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.99 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 07:01:04 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-941s1202/ses-201207021018000/anat/sub-941s1202_ses-201207021018000_T1w.nii --asegdkt_segfile /output/941s1202/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/941s1202/mri/orig.mgz --brainmask_name /output/941s1202/mri/mask.mgz --aseg_name /output/941s1202/mri/aseg.auto_noCCseg.mgz --sid 941s1202 --seg_log /output/941s1202/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-941s1202/ses-201207021018000/anat/sub-941s1202_ses-201207021018000_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-941s1202/ses-201207021018000/anat/sub-941s1202_ses-201207021018000_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/941s1202/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1202/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 0.02] [-0. 0.99 0.36] [-0.02 -0.36 0.99]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 2226.611083984375\r\n",
      "rescale:  min: 0.0  max: 1117.7587890625  scale: 0.22813508826343173\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1202/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.89batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-941s1202/ses-201207021018000/anat/sub-941s1202_ses-201207021018000_T1w.nii finished in 5.9713 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.51batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-941s1202/ses-201207021018000/anat/sub-941s1202_ses-201207021018000_T1w.nii finished in 6.3206 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.99batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-941s1202/ses-201207021018000/anat/sub-941s1202_ses-201207021018000_T1w.nii finished in 5.8205 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1202/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  21518\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1202/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1202/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.1\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/941s1202/mri/orig.mgz --rescale /output/941s1202/mri/orig_nu.mgz --aseg /output/941s1202/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/941s1202/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/941s1202/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/941s1202/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 41.90\r\n",
      "- m: 2.4343\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/941s1202/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/941s1202/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/941s1202/stats/aseg+DKT.stats --normfile /output/941s1202/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 941s1202 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/941s1202/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/941s1202/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.10 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 07:04:14 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-137s0722/ses-201208311552330/anat/sub-137s0722_ses-201208311552330_T1w.nii --asegdkt_segfile /output/137s0722/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/137s0722/mri/orig.mgz --brainmask_name /output/137s0722/mri/mask.mgz --aseg_name /output/137s0722/mri/aseg.auto_noCCseg.mgz --sid 137s0722 --seg_log /output/137s0722/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-137s0722/ses-201208311552330/anat/sub-137s0722_ses-201208311552330_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-137s0722/ses-201208311552330/anat/sub-137s0722_ses-201208311552330_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/137s0722/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0722/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.20e+00 3.45e-02 1.51e-02] [-4.29e-02 9.74e-01 4.03e-01] [-9.07e-04 -4.03e-01 9.75e-01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 963.1763305664062\r\n",
      "rescale:  min: 0.0  max: 604.874755859375  scale: 0.42157487567440155\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0722/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-137s0722/ses-201208311552330/anat/sub-137s0722_ses-201208311552330_T1w.nii finished in 5.9848 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.82batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-137s0722/ses-201208311552330/anat/sub-137s0722_ses-201208311552330_T1w.nii finished in 6.2721 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.10batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-137s0722/ses-201208311552330/anat/sub-137s0722_ses-201208311552330_T1w.nii finished in 5.8057 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0722/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  23163\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0722/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0722/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.28\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/137s0722/mri/orig.mgz --rescale /output/137s0722/mri/orig_nu.mgz --aseg /output/137s0722/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/137s0722/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/137s0722/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/137s0722/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 63.82\r\n",
      "- m: 1.5984\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/137s0722/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/137s0722/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/137s0722/stats/aseg+DKT.stats --normfile /output/137s0722/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 137s0722 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/137s0722/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/137s0722/stats/aseg+DKT.stats.\r\n",
      "Calculation took 11.19 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 07:07:24 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-023s2068/ses-201210171218320/anat/sub-023s2068_ses-201210171218320_T1w.nii --asegdkt_segfile /output/023s2068/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/023s2068/mri/orig.mgz --brainmask_name /output/023s2068/mri/mask.mgz --aseg_name /output/023s2068/mri/aseg.auto_noCCseg.mgz --sid 023s2068 --seg_log /output/023s2068/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-023s2068/ses-201210171218320/anat/sub-023s2068_ses-201210171218320_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-023s2068/ses-201210171218320/anat/sub-023s2068_ses-201210171218320_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/023s2068/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s2068/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0. -0.07] [ 0.01 1.03 0.23] [ 0.08 -0.23 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 3399.73291015625\r\n",
      "rescale:  min: 0.0  max: 1342.89453125  scale: 0.18988833007059727\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s2068/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.95batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-023s2068/ses-201210171218320/anat/sub-023s2068_ses-201210171218320_T1w.nii finished in 5.9620 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.70batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-023s2068/ses-201210171218320/anat/sub-023s2068_ses-201210171218320_T1w.nii finished in 6.2901 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.05batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-023s2068/ses-201210171218320/anat/sub-023s2068_ses-201210171218320_T1w.nii finished in 5.8128 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s2068/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  19354\r\n",
      "  Found 2 connected component(s)!\r\n",
      "  Selecting largest component!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s2068/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 3 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/023s2068/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.08\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/023s2068/mri/orig.mgz --rescale /output/023s2068/mri/orig_nu.mgz --aseg /output/023s2068/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/023s2068/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/023s2068/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/023s2068/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 42.30\r\n",
      "- m: 2.4116\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/023s2068/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/023s2068/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/023s2068/stats/aseg+DKT.stats --normfile /output/023s2068/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 023s2068 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/023s2068/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/023s2068/stats/aseg+DKT.stats.\r\n",
      "Calculation took 14.57 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 07:10:38 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-037s1078/ses-201211281227570/anat/sub-037s1078_ses-201211281227570_T1w.nii --asegdkt_segfile /output/037s1078/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/037s1078/mri/orig.mgz --brainmask_name /output/037s1078/mri/mask.mgz --aseg_name /output/037s1078/mri/aseg.auto_noCCseg.mgz --sid 037s1078 --seg_log /output/037s1078/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-037s1078/ses-201211281227570/anat/sub-037s1078_ses-201211281227570_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-037s1078/ses-201211281227570/anat/sub-037s1078_ses-201211281227570_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/037s1078/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s1078/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.05 0.07] [-0.07 1.03 0.21] [-0.07 -0.21 1.03]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1228.8770751953125\r\n",
      "rescale:  min: 0.0  max: 872.5027465820312  scale: 0.29226269028830537\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s1078/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.97batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-037s1078/ses-201211281227570/anat/sub-037s1078_ses-201211281227570_T1w.nii finished in 5.9607 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.38batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-037s1078/ses-201211281227570/anat/sub-037s1078_ses-201211281227570_T1w.nii finished in 6.3402 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.11batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-037s1078/ses-201211281227570/anat/sub-037s1078_ses-201211281227570_T1w.nii finished in 5.8051 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s1078/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18766\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s1078/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/037s1078/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.03\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/037s1078/mri/orig.mgz --rescale /output/037s1078/mri/orig_nu.mgz --aseg /output/037s1078/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/037s1078/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/037s1078/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/037s1078/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 57.77\r\n",
      "- m: 1.7655\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/037s1078/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/037s1078/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/037s1078/stats/aseg+DKT.stats --normfile /output/037s1078/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 037s1078 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/037s1078/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/037s1078/stats/aseg+DKT.stats.\r\n",
      "Calculation took 10.19 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 07:13:47 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-137s0994/ses-201211141033240/anat/sub-137s0994_ses-201211141033240_T1w.nii --asegdkt_segfile /output/137s0994/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/137s0994/mri/orig.mgz --brainmask_name /output/137s0994/mri/mask.mgz --aseg_name /output/137s0994/mri/aseg.auto_noCCseg.mgz --sid 137s0994 --seg_log /output/137s0994/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-137s0994/ses-201211141033240/anat/sub-137s0994_ses-201211141033240_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-137s0994/ses-201211141033240/anat/sub-137s0994_ses-201211141033240_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/137s0994/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0994/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.19 0.03 0.09] [-0.07 0.99 0.36] [-0.09 -0.36 0.99]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 600.8345947265625\r\n",
      "rescale:  min: 0.0  max: 400.7566833496094  scale: 0.6362963129364578\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0994/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-137s0994/ses-201211141033240/anat/sub-137s0994_ses-201211141033240_T1w.nii finished in 5.9856 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.39batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-137s0994/ses-201211141033240/anat/sub-137s0994_ses-201211141033240_T1w.nii finished in 6.3390 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.04batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-137s0994/ses-201211141033240/anat/sub-137s0994_ses-201211141033240_T1w.nii finished in 5.8136 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0994/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17931\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0994/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/137s0994/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.94\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/137s0994/mri/orig.mgz --rescale /output/137s0994/mri/orig_nu.mgz --aseg /output/137s0994/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/137s0994/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/137s0994/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/137s0994/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 81.95\r\n",
      "- m: 1.2446\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/137s0994/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/137s0994/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/137s0994/stats/aseg+DKT.stats --normfile /output/137s0994/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 137s0994 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/137s0994/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/137s0994/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.36 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 07:16:54 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-941s1195/ses-201303071323450/anat/sub-941s1195_ses-201303071323450_T1w.nii --asegdkt_segfile /output/941s1195/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/941s1195/mri/orig.mgz --brainmask_name /output/941s1195/mri/mask.mgz --aseg_name /output/941s1195/mri/aseg.auto_noCCseg.mgz --sid 941s1195 --seg_log /output/941s1195/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-941s1195/ses-201303071323450/anat/sub-941s1195_ses-201303071323450_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-941s1195/ses-201303071323450/anat/sub-941s1195_ses-201303071323450_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/941s1195/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1195/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.02 0.05] [ 0.02 1.05 0.13] [-0.06 -0.13 1.05]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1351.8680419921875\r\n",
      "rescale:  min: 0.0  max: 894.9366455078125  scale: 0.2849363709487008\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1195/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.92batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-941s1195/ses-201303071323450/anat/sub-941s1195_ses-201303071323450_T1w.nii finished in 5.9670 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.41batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-941s1195/ses-201303071323450/anat/sub-941s1195_ses-201303071323450_T1w.nii finished in 6.3359 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 43.96batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-941s1195/ses-201303071323450/anat/sub-941s1195_ses-201303071323450_T1w.nii finished in 5.8240 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1195/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18735\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1195/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/941s1195/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.09\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/941s1195/mri/orig.mgz --rescale /output/941s1195/mri/orig_nu.mgz --aseg /output/941s1195/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/941s1195/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/941s1195/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/941s1195/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 51.18\r\n",
      "- m: 1.9929\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/941s1195/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/941s1195/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/941s1195/stats/aseg+DKT.stats --normfile /output/941s1195/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 941s1195 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/941s1195/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/941s1195/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.83 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 07:20:03 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-073s0746/ses-201303081132280/anat/sub-073s0746_ses-201303081132280_T1w.nii --asegdkt_segfile /output/073s0746/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/073s0746/mri/orig.mgz --brainmask_name /output/073s0746/mri/mask.mgz --aseg_name /output/073s0746/mri/aseg.auto_noCCseg.mgz --sid 073s0746 --seg_log /output/073s0746/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-073s0746/ses-201303081132280/anat/sub-073s0746_ses-201303081132280_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-073s0746/ses-201303081132280/anat/sub-073s0746_ses-201303081132280_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/073s0746/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0746/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (176, 240, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.1999999x1.0546875x1.0546875\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 0.04 -0.03] [-0.04 1.04 0.16] [ 0.05 -0.16 1.04]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 1526.7708740234375\r\n",
      "rescale:  min: 0.0  max: 916.0625  scale: 0.2783652862113666\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0746/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.79batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-073s0746/ses-201303081132280/anat/sub-073s0746_ses-201303081132280_T1w.nii finished in 5.9854 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.36batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-073s0746/ses-201303081132280/anat/sub-073s0746_ses-201303081132280_T1w.nii finished in 6.3436 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.23batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-073s0746/ses-201303081132280/anat/sub-073s0746_ses-201303081132280_T1w.nii finished in 5.7887 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0746/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  18012\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0746/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/073s0746/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 1.05\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/073s0746/mri/orig.mgz --rescale /output/073s0746/mri/orig_nu.mgz --aseg /output/073s0746/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/073s0746/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/073s0746/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/073s0746/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 67.46\r\n",
      "- m: 1.5120\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/073s0746/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/073s0746/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/073s0746/stats/aseg+DKT.stats --normfile /output/073s0746/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 073s0746 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/073s0746/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/073s0746/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.97 seconds using up to 1 threads.\r\n",
      "Version: 2.3.3+bd65b36\r\n",
      "Tue Jan  7 07:23:11 UTC 2025\r\n",
      "\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/run_prediction.py --t1 /data/sub-131s0123/ses-201303211338380/anat/sub-131s0123_ses-201303211338380_T1w.nii --asegdkt_segfile /output/131s0123/mri/aparc.DKTatlas+aseg.deep.mgz --conformed_name /output/131s0123/mri/orig.mgz --brainmask_name /output/131s0123/mri/mask.mgz --aseg_name /output/131s0123/mri/aseg.auto_noCCseg.mgz --sid 131s0123 --seg_log /output/131s0123/scripts/deep-seg.log --vox_size min --batch_size 1 --viewagg_device auto --device auto --sd /output\r\n",
      "[INFO: run_prediction.py:  648]: Checking or downloading default checkpoints ...\r\n",
      "[INFO: common.py:  827]: Single subject with absolute file path for input.\r\n",
      "[INFO: common.py:  885]: Analyzing single subject /data/sub-131s0123/ses-201303211338380/anat/sub-131s0123_ses-201303211338380_T1w.nii\r\n",
      "[INFO: common.py:  986]: Output will be stored in Subjects Directory: /output\r\n",
      "[INFO: common.py:  114]: Using device: cuda\r\n",
      "[INFO: common.py:  114]: Using viewagg_device: cuda\r\n",
      "[INFO: run_prediction.py:  245]: Running view aggregation on cuda\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_coronal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_sagittal_v2.0.0.pkl\r\n",
      "[INFO: inference.py:  200]: Loading checkpoint /fastsurfer/checkpoints/aparc_vinn_axial_v2.0.0.pkl\r\n",
      "[INFO: run_prediction.py:  324]: Successfully loaded image from /data/sub-131s0123/ses-201303211338380/anat/sub-131s0123_ses-201303211338380_T1w.nii.\r\n",
      "[INFO: run_prediction.py:  441]: Output image directory /output/131s0123/mri/orig does not exist. Creating it now...\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/131s0123/mri/orig/001.mgz.\r\n",
      "The input image is not conformed.\r\n",
      "A min-conformed image must satisfy the following criteria:\r\n",
      " - Number of Dimensions 3        : True\r\n",
      " - Dimensions 256x256x256        : BUT image dimensions (196, 256, 256)\r\n",
      " - Voxel Size 1.0x1.0x1.0        : BUT image 1.2x1.0156x1.0156\r\n",
      " - Orientation LIA               : BUT strict: [[ 1.2 -0.01 -0.06] [ 0.01 1.01 0.12] [ 0.06 -0.12 1.01]]\r\n",
      " - Dtype uint8                   : BUT dtype >f4\r\n",
      "[INFO: run_prediction.py:  337]: Conforming image\r\n",
      "(1.0, 1.0, 1.0)\r\n",
      "Input:    min: 0.0  max: 15064.845703125\r\n",
      "rescale:  min: 0.0  max: 5589.0576171875  scale: 0.04562486513215083\r\n",
      "Output:   min: 0.0  max: 255.0\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/131s0123/mri/orig.mgz.\r\n",
      "[INFO: run_prediction.py:  400]: Run coronal prediction\r\n",
      "[INFO: dataset.py:   78]: Loading Coronal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 42.67batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for coronal successful\r\n",
      "[INFO: inference.py:  468]: Coronal inference on /data/sub-131s0123/ses-201303211338380/anat/sub-131s0123_ses-201303211338380_T1w.nii finished in 6.0022 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run sagittal prediction\r\n",
      "[INFO: dataset.py:   69]: Loading Sagittal with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:06<00:00, 40.75batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for sagittal successful\r\n",
      "[INFO: inference.py:  468]: Sagittal inference on /data/sub-131s0123/ses-201303211338380/anat/sub-131s0123_ses-201303211338380_T1w.nii finished in 6.2826 seconds\r\n",
      "[INFO: run_prediction.py:  400]: Run axial prediction\r\n",
      "[INFO: dataset.py:   74]: Loading Axial with input voxelsize (1.0, 1.0)\r\n",
      "100%|| 256/256 [00:05<00:00, 44.26batch/s]\r\n",
      "[INFO: inference.py:  406]: Inference on 256 batches for axial successful\r\n",
      "[INFO: inference.py:  468]: Axial inference on /data/sub-131s0123/ses-201303211338380/anat/sub-131s0123_ses-201303211338380_T1w.nii finished in 5.7853 seconds\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/131s0123/mri/aparc.DKTatlas+aseg.deep.mgz.\r\n",
      "[INFO: run_prediction.py:  725]: Creating brainmask based on segmentation...\r\n",
      "Creating dilated mask ...\r\n",
      "Frontal region special treatment:  17872\r\n",
      "  Found 1 connected component(s)!\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/131s0123/mri/mask.mgz.\r\n",
      "[INFO: run_prediction.py:  744]: Creating aseg based on segmentation...\r\n",
      "Reducing to aseg ...\r\n",
      "FlipWM: rh 0 and lh 0 flipped.\r\n",
      "[INFO: run_prediction.py:  454]: Successfully saved image   as /output/131s0123/mri/aseg.auto_noCCseg.mgz.\r\n",
      "[INFO: run_prediction.py:  763]: Running volume-based QC check on segmentation...\r\n",
      "Checking total volume ...\r\n",
      "Voxel size in mm3: 1.0\r\n",
      "Total segmentation volume in liter: 0.95\r\n",
      "INFO: Running N4 bias-field correction\r\n",
      "python3.10 -s /fastsurfer/recon_surf/N4_bias_correct.py --in /output/131s0123/mri/orig.mgz --rescale /output/131s0123/mri/orig_nu.mgz --aseg /output/131s0123/mri/aparc.DKTatlas+aseg.deep.mgz --threads 1\r\n",
      "\r\n",
      "N4 Bias Correction Parameters:\r\n",
      "\r\n",
      "- verbosity: -1\r\n",
      "- input volume: /output/131s0123/mri/orig.mgz\r\n",
      "- output volume: do not save\r\n",
      "- rescaled volume: /output/131s0123/mri/orig_nu.mgz\r\n",
      "- mask: default (>0)\r\n",
      "- aseg: /output/131s0123/mri/aparc.DKTatlas+aseg.deep.mgz\r\n",
      "- shrink factor: 4\r\n",
      "- number fitting levels: 4\r\n",
      "- number iterations: 50\r\n",
      "- convergence threshold: 0.0\r\n",
      "- threads: 1\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "executing N4 correction ...\r\n",
      "- mask: ones (default)\r\n",
      "normalize WM to 105.0 (find WM from aseg)\r\n",
      "read MGZ (FreeSurfer) image via nibabel...\r\n",
      "- source white matter intensity: 53.16\r\n",
      "- m: 1.9188\r\n",
      "converting rescaled to UCHAR\r\n",
      "writing PosixPath: /output/131s0123/mri/orig_nu.mgz\r\n",
      "write MGZ (FreeSurfer) image via nibabel...\r\n",
      "python3.10 -s /fastsurfer/FastSurferCNN/segstats.py --segfile /output/131s0123/mri/aparc.DKTatlas+aseg.deep.mgz --segstatsfile /output/131s0123/stats/aseg+DKT.stats --normfile /output/131s0123/mri/orig_nu.mgz --threads 1 --empty --excludeid 0 --sd /output --sid 131s0123 --ids 2 4 5 7 8 10 11 12 13 14 15 16 17 18 24 26 28 31 41 43 44 46 47 49 50 51 52 53 54 58 60 63 77 251 252 253 254 255 1002 1003 1005 1006 1007 1008 1009 1010 1011 1012 1013 1014 1015 1016 1017 1018 1019 1020 1021 1022 1023 1024 1025 1026 1027 1028 1029 1030 1031 1034 1035 2002 2003 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025 2026 2027 2028 2029 2030 2031 2034 2035 --lut /fastsurfer/FastSurferCNN/config/FreeSurferColorLUT.txt measures --compute Mask(/output/131s0123/mri/mask.mgz) BrainSeg BrainSegNotVent SupraTentorial SupraTentorialNotVent SubCortGray rhCerebralWhiteMatter lhCerebralWhiteMatter CerebralWhiteMatter\r\n",
      "None of the labels [251, 252, 253, 254, 255] for merged label 10007 exist in the segmentation.\r\n",
      "Partial volume stats for 100 labels written to /output/131s0123/stats/aseg+DKT.stats.\r\n",
      "Calculation took 9.72 seconds using up to 1 threads.\r\n"
     ]
    }
   ],
   "execution_count": 122
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:29.010272Z",
     "start_time": "2025-01-07T16:59:29.004854Z"
    }
   },
   "cell_type": "code",
   "source": [
    "structs_of_interest = ['Left-Cerebral-White-Matter',\n",
    " 'Left-Lateral-Ventricle',\n",
    " 'Left-Inf-Lat-Vent',\n",
    " 'Left-Cerebellum-White-Matter',\n",
    " 'Left-Cerebellum-Cortex',\n",
    " 'Left-Thalamus',\n",
    " 'Left-Caudate',\n",
    " 'Left-Putamen',\n",
    " 'Left-Pallidum',\n",
    " '3rd-Ventricle',\n",
    " '4th-Ventricle',\n",
    " 'Brain-Stem',\n",
    " 'Left-Hippocampus',\n",
    " 'Left-Amygdala',\n",
    " 'CSF',\n",
    " 'Left-Accumbens-area',\n",
    " 'Left-VentralDC',\n",
    " 'Left-choroid-plexus',\n",
    " 'Right-Cerebral-White-Matter',\n",
    " 'Right-Lateral-Ventricle',\n",
    " 'Right-Inf-Lat-Vent',\n",
    " 'Right-Cerebellum-White-Matter',\n",
    " 'Right-Cerebellum-Cortex',\n",
    " 'Right-Thalamus',\n",
    " 'Right-Caudate',\n",
    " 'Right-Putamen',\n",
    " 'Right-Pallidum',\n",
    " 'Right-Hippocampus',\n",
    " 'Right-Amygdala',\n",
    " 'Right-Accumbens-area',\n",
    " 'Right-VentralDC',\n",
    " 'Right-choroid-plexus',\n",
    " 'WM-hypointensities',\n",
    " 'ctx-lh-caudalanteriorcingulate',\n",
    " 'ctx-lh-caudalmiddlefrontal',\n",
    " 'ctx-lh-cuneus',\n",
    " 'ctx-lh-entorhinal',\n",
    " 'ctx-lh-fusiform',\n",
    " 'ctx-lh-inferiorparietal',\n",
    " 'ctx-lh-inferiortemporal',\n",
    " 'ctx-lh-isthmuscingulate',\n",
    " 'ctx-lh-lateraloccipital',\n",
    " 'ctx-lh-lateralorbitofrontal',\n",
    " 'ctx-lh-lingual',\n",
    " 'ctx-lh-medialorbitofrontal',\n",
    " 'ctx-lh-middletemporal',\n",
    " 'ctx-lh-parahippocampal',\n",
    " 'ctx-lh-paracentral',\n",
    " 'ctx-lh-parsopercularis',\n",
    " 'ctx-lh-parsorbitalis',\n",
    " 'ctx-lh-parstriangularis',\n",
    " 'ctx-lh-pericalcarine',\n",
    " 'ctx-lh-postcentral',\n",
    " 'ctx-lh-posteriorcingulate',\n",
    " 'ctx-lh-precentral',\n",
    " 'ctx-lh-precuneus',\n",
    " 'ctx-lh-rostralanteriorcingulate',\n",
    " 'ctx-lh-rostralmiddlefrontal',\n",
    " 'ctx-lh-superiorfrontal',\n",
    " 'ctx-lh-superiorparietal',\n",
    " 'ctx-lh-superiortemporal',\n",
    " 'ctx-lh-supramarginal',\n",
    " 'ctx-lh-transversetemporal',\n",
    " 'ctx-lh-insula',\n",
    " 'ctx-rh-caudalanteriorcingulate',\n",
    " 'ctx-rh-caudalmiddlefrontal',\n",
    " 'ctx-rh-cuneus',\n",
    " 'ctx-rh-entorhinal',\n",
    " 'ctx-rh-fusiform',\n",
    " 'ctx-rh-inferiorparietal',\n",
    " 'ctx-rh-inferiortemporal',\n",
    " 'ctx-rh-isthmuscingulate',\n",
    " 'ctx-rh-lateraloccipital',\n",
    " 'ctx-rh-lateralorbitofrontal',\n",
    " 'ctx-rh-lingual',\n",
    " 'ctx-rh-medialorbitofrontal',\n",
    " 'ctx-rh-middletemporal',\n",
    " 'ctx-rh-parahippocampal',\n",
    " 'ctx-rh-paracentral',\n",
    " 'ctx-rh-parsopercularis',\n",
    " 'ctx-rh-parsorbitalis',\n",
    " 'ctx-rh-parstriangularis',\n",
    " 'ctx-rh-pericalcarine',\n",
    " 'ctx-rh-postcentral',\n",
    " 'ctx-rh-posteriorcingulate',\n",
    " 'ctx-rh-precentral',\n",
    " 'ctx-rh-precuneus',\n",
    " 'ctx-rh-rostralanteriorcingulate',\n",
    " 'ctx-rh-rostralmiddlefrontal',\n",
    " 'ctx-rh-superiorfrontal',\n",
    " 'ctx-rh-superiorparietal',\n",
    " 'ctx-rh-superiortemporal',\n",
    " 'ctx-rh-supramarginal',\n",
    " 'ctx-rh-transversetemporal',\n",
    " 'ctx-rh-insula']"
   ],
   "outputs": [],
   "execution_count": 248
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:33.352950Z",
     "start_time": "2025-01-07T16:59:33.212685Z"
    }
   },
   "cell_type": "code",
   "source": [
    "fs_table = pd.read_csv(f\"{tables_path}/UCSFFSX7_06Jan2025.csv\")\n",
    "fs_table"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "       PHASE        PTID   RID VISCODE VISCODE2  IMAGEUID    EXAMDATE  \\\n",
       "0      ADNI3  002_S_4213  4213    init      m72    888008  2017-08-14   \n",
       "1      ADNI2  116_S_4453  4453     v02    scmri    281425  2012-01-18   \n",
       "2      ADNI2  099_S_4104  4104     v11      m12    322667  2012-07-11   \n",
       "3      ADNI2  073_S_2153  2153     v21      m36    417092  2014-03-03   \n",
       "4      ADNI2  137_S_4303  4303     v04      m03    288921  2012-02-27   \n",
       "...      ...         ...   ...     ...      ...       ...         ...   \n",
       "6326   ADNI2  021_S_5237  5237     v02    scmri    379938  2013-06-27   \n",
       "6327   ADNI2  100_S_4556  4556     v02    scmri    305408  2012-04-26   \n",
       "6328   ADNI3  005_S_6093  6093      sc       sc    924228  2017-10-27   \n",
       "6329   ADNI3  129_S_6852  6852      y2      m24   1547010  2022-02-21   \n",
       "6330  ADNIGO  021_S_2125  2125     m03      m03    222797  2011-01-21   \n",
       "\n",
       "         RUNDATE   STATUS  FSVER  ...        ST147SV        ST148SV  \\\n",
       "0     2022-07-23  partial  7.2.0  ...  217434.885092  217173.802571   \n",
       "1     2022-10-05  partial  7.2.0  ...  224522.552452  221805.186019   \n",
       "2     2022-09-26  partial  7.2.0  ...  219930.656696  224579.475317   \n",
       "3     2022-08-14  partial  7.2.0  ...  200725.779310  201273.901030   \n",
       "4     2022-10-05  partial  7.2.0  ...  198949.362921  200313.596282   \n",
       "...          ...      ...    ...  ...            ...            ...   \n",
       "6326  2022-09-11  partial  7.2.0  ...  212252.638922  208494.814443   \n",
       "6327  2022-08-14  partial  7.2.0  ...  224566.873931  229790.009048   \n",
       "6328  2022-07-23  partial  7.2.0  ...  237455.319821  235727.874558   \n",
       "6329  2023-08-29  partial  7.2.0  ...  202401.632540  203515.760212   \n",
       "6330  2022-09-11  partial  7.2.0  ...  230069.438106  225881.860461   \n",
       "\n",
       "            ST149SV   ST150SV   ST151SV   ST152SV  ST153SV        ST154SV  \\\n",
       "0     434608.687663  175240.0  171111.0  346351.0  47936.0  576920.687663   \n",
       "1     446327.738471  225291.0  221178.0  446469.0  50711.0  599707.738471   \n",
       "2     444510.132013  199702.0  200555.0  400257.0  52008.0  596412.132013   \n",
       "3     401999.680340  174186.0  176432.0  350618.0  49603.0  538118.680340   \n",
       "4     399262.959202  202949.5  196986.5  399936.0  47355.0  543084.959202   \n",
       "...             ...       ...       ...       ...      ...            ...   \n",
       "6326  420747.453366  211465.5  212177.5  423643.0  51856.0  584021.453366   \n",
       "6327  454356.882979  223665.5  227203.5  450869.0  53580.0  603257.882979   \n",
       "6328  473183.194379  214672.5  213593.5  428266.0  50758.0  630561.194379   \n",
       "6329  405917.392752  189276.5  185867.5  375144.0  48543.0  573948.392752   \n",
       "6330  455951.298567  229819.5  226962.5  456782.0  48852.0  620988.298567   \n",
       "\n",
       "        ST155SV           update_stamp  \n",
       "0      874725.0  2024-12-02 08:27:54.0  \n",
       "1      984072.0  2024-12-02 08:27:54.0  \n",
       "2      928837.0  2024-12-02 08:27:54.0  \n",
       "3      846706.0  2024-12-02 08:27:54.0  \n",
       "4      902306.0  2024-12-02 08:27:54.0  \n",
       "...         ...                    ...  \n",
       "6326   937760.0  2024-12-02 08:27:54.0  \n",
       "6327   999649.0  2024-12-02 08:27:54.0  \n",
       "6328   967388.0  2024-12-02 08:27:54.0  \n",
       "6329   885358.0  2024-12-02 08:27:54.0  \n",
       "6330  1022940.0  2024-12-02 08:27:54.0  \n",
       "\n",
       "[6331 rows x 346 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PHASE</th>\n",
       "      <th>PTID</th>\n",
       "      <th>RID</th>\n",
       "      <th>VISCODE</th>\n",
       "      <th>VISCODE2</th>\n",
       "      <th>IMAGEUID</th>\n",
       "      <th>EXAMDATE</th>\n",
       "      <th>RUNDATE</th>\n",
       "      <th>STATUS</th>\n",
       "      <th>FSVER</th>\n",
       "      <th>...</th>\n",
       "      <th>ST147SV</th>\n",
       "      <th>ST148SV</th>\n",
       "      <th>ST149SV</th>\n",
       "      <th>ST150SV</th>\n",
       "      <th>ST151SV</th>\n",
       "      <th>ST152SV</th>\n",
       "      <th>ST153SV</th>\n",
       "      <th>ST154SV</th>\n",
       "      <th>ST155SV</th>\n",
       "      <th>update_stamp</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ADNI3</td>\n",
       "      <td>002_S_4213</td>\n",
       "      <td>4213</td>\n",
       "      <td>init</td>\n",
       "      <td>m72</td>\n",
       "      <td>888008</td>\n",
       "      <td>2017-08-14</td>\n",
       "      <td>2022-07-23</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>217434.885092</td>\n",
       "      <td>217173.802571</td>\n",
       "      <td>434608.687663</td>\n",
       "      <td>175240.0</td>\n",
       "      <td>171111.0</td>\n",
       "      <td>346351.0</td>\n",
       "      <td>47936.0</td>\n",
       "      <td>576920.687663</td>\n",
       "      <td>874725.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ADNI2</td>\n",
       "      <td>116_S_4453</td>\n",
       "      <td>4453</td>\n",
       "      <td>v02</td>\n",
       "      <td>scmri</td>\n",
       "      <td>281425</td>\n",
       "      <td>2012-01-18</td>\n",
       "      <td>2022-10-05</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>224522.552452</td>\n",
       "      <td>221805.186019</td>\n",
       "      <td>446327.738471</td>\n",
       "      <td>225291.0</td>\n",
       "      <td>221178.0</td>\n",
       "      <td>446469.0</td>\n",
       "      <td>50711.0</td>\n",
       "      <td>599707.738471</td>\n",
       "      <td>984072.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ADNI2</td>\n",
       "      <td>099_S_4104</td>\n",
       "      <td>4104</td>\n",
       "      <td>v11</td>\n",
       "      <td>m12</td>\n",
       "      <td>322667</td>\n",
       "      <td>2012-07-11</td>\n",
       "      <td>2022-09-26</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>219930.656696</td>\n",
       "      <td>224579.475317</td>\n",
       "      <td>444510.132013</td>\n",
       "      <td>199702.0</td>\n",
       "      <td>200555.0</td>\n",
       "      <td>400257.0</td>\n",
       "      <td>52008.0</td>\n",
       "      <td>596412.132013</td>\n",
       "      <td>928837.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ADNI2</td>\n",
       "      <td>073_S_2153</td>\n",
       "      <td>2153</td>\n",
       "      <td>v21</td>\n",
       "      <td>m36</td>\n",
       "      <td>417092</td>\n",
       "      <td>2014-03-03</td>\n",
       "      <td>2022-08-14</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>200725.779310</td>\n",
       "      <td>201273.901030</td>\n",
       "      <td>401999.680340</td>\n",
       "      <td>174186.0</td>\n",
       "      <td>176432.0</td>\n",
       "      <td>350618.0</td>\n",
       "      <td>49603.0</td>\n",
       "      <td>538118.680340</td>\n",
       "      <td>846706.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADNI2</td>\n",
       "      <td>137_S_4303</td>\n",
       "      <td>4303</td>\n",
       "      <td>v04</td>\n",
       "      <td>m03</td>\n",
       "      <td>288921</td>\n",
       "      <td>2012-02-27</td>\n",
       "      <td>2022-10-05</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>198949.362921</td>\n",
       "      <td>200313.596282</td>\n",
       "      <td>399262.959202</td>\n",
       "      <td>202949.5</td>\n",
       "      <td>196986.5</td>\n",
       "      <td>399936.0</td>\n",
       "      <td>47355.0</td>\n",
       "      <td>543084.959202</td>\n",
       "      <td>902306.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6326</th>\n",
       "      <td>ADNI2</td>\n",
       "      <td>021_S_5237</td>\n",
       "      <td>5237</td>\n",
       "      <td>v02</td>\n",
       "      <td>scmri</td>\n",
       "      <td>379938</td>\n",
       "      <td>2013-06-27</td>\n",
       "      <td>2022-09-11</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>212252.638922</td>\n",
       "      <td>208494.814443</td>\n",
       "      <td>420747.453366</td>\n",
       "      <td>211465.5</td>\n",
       "      <td>212177.5</td>\n",
       "      <td>423643.0</td>\n",
       "      <td>51856.0</td>\n",
       "      <td>584021.453366</td>\n",
       "      <td>937760.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6327</th>\n",
       "      <td>ADNI2</td>\n",
       "      <td>100_S_4556</td>\n",
       "      <td>4556</td>\n",
       "      <td>v02</td>\n",
       "      <td>scmri</td>\n",
       "      <td>305408</td>\n",
       "      <td>2012-04-26</td>\n",
       "      <td>2022-08-14</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>224566.873931</td>\n",
       "      <td>229790.009048</td>\n",
       "      <td>454356.882979</td>\n",
       "      <td>223665.5</td>\n",
       "      <td>227203.5</td>\n",
       "      <td>450869.0</td>\n",
       "      <td>53580.0</td>\n",
       "      <td>603257.882979</td>\n",
       "      <td>999649.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6328</th>\n",
       "      <td>ADNI3</td>\n",
       "      <td>005_S_6093</td>\n",
       "      <td>6093</td>\n",
       "      <td>sc</td>\n",
       "      <td>sc</td>\n",
       "      <td>924228</td>\n",
       "      <td>2017-10-27</td>\n",
       "      <td>2022-07-23</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>237455.319821</td>\n",
       "      <td>235727.874558</td>\n",
       "      <td>473183.194379</td>\n",
       "      <td>214672.5</td>\n",
       "      <td>213593.5</td>\n",
       "      <td>428266.0</td>\n",
       "      <td>50758.0</td>\n",
       "      <td>630561.194379</td>\n",
       "      <td>967388.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6329</th>\n",
       "      <td>ADNI3</td>\n",
       "      <td>129_S_6852</td>\n",
       "      <td>6852</td>\n",
       "      <td>y2</td>\n",
       "      <td>m24</td>\n",
       "      <td>1547010</td>\n",
       "      <td>2022-02-21</td>\n",
       "      <td>2023-08-29</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>202401.632540</td>\n",
       "      <td>203515.760212</td>\n",
       "      <td>405917.392752</td>\n",
       "      <td>189276.5</td>\n",
       "      <td>185867.5</td>\n",
       "      <td>375144.0</td>\n",
       "      <td>48543.0</td>\n",
       "      <td>573948.392752</td>\n",
       "      <td>885358.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6330</th>\n",
       "      <td>ADNIGO</td>\n",
       "      <td>021_S_2125</td>\n",
       "      <td>2125</td>\n",
       "      <td>m03</td>\n",
       "      <td>m03</td>\n",
       "      <td>222797</td>\n",
       "      <td>2011-01-21</td>\n",
       "      <td>2022-09-11</td>\n",
       "      <td>partial</td>\n",
       "      <td>7.2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>230069.438106</td>\n",
       "      <td>225881.860461</td>\n",
       "      <td>455951.298567</td>\n",
       "      <td>229819.5</td>\n",
       "      <td>226962.5</td>\n",
       "      <td>456782.0</td>\n",
       "      <td>48852.0</td>\n",
       "      <td>620988.298567</td>\n",
       "      <td>1022940.0</td>\n",
       "      <td>2024-12-02 08:27:54.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6331 rows  346 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 249,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 249
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:35.659685Z",
     "start_time": "2025-01-07T16:59:35.656307Z"
    }
   },
   "cell_type": "code",
   "source": [
    "subjects_in_order = []\n",
    "\n",
    "with open(f'{dataset_base_path}/subjects.txt') as subjects_file:\n",
    "    for line in subjects_file:\n",
    "        subjects_in_order.append(line.strip())"
   ],
   "outputs": [],
   "execution_count": 250
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:37.654258Z",
     "start_time": "2025-01-07T16:59:37.509528Z"
    }
   },
   "cell_type": "code",
   "source": [
    "from datetime import datetime\n",
    "bids_files = glob.glob(f\"{mri_bids_path}/**/*.nii\", recursive=True)\n",
    "\n",
    "session_dates = {}\n",
    "\n",
    "for file in bids_files:\n",
    "    subject_id = re.search(r'sub-(\\w+)', file).group(1)\n",
    "    session_id = re.search(r'ses-(\\w+)', file).group(1)\n",
    "\n",
    "    session_dates[subject_id] = datetime.strptime(session_id[:8], '%Y%m%d')"
   ],
   "outputs": [],
   "execution_count": 251
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:40.660485Z",
     "start_time": "2025-01-07T16:59:39.174697Z"
    }
   },
   "cell_type": "code",
   "source": [
    "fs_table['EXAMDATE'] = pd.to_datetime(fs_table['EXAMDATE'], format='%Y-%m-%d')\n",
    "\n",
    "# ignore all rows where the diagnosis is nan\n",
    "fs_table = fs_table[fs_table['ST10CV'].notna()]\n",
    "\n",
    "# ignore all rows where the exam date is nan\n",
    "fs_table = fs_table[fs_table['EXAMDATE'].notna()]\n",
    "\n",
    "etiv_per_subject = {}\n",
    "\n",
    "# for each subject, get the diagnosis where the date is closest to the session date\n",
    "for subject_id in subjects_in_order:\n",
    "    etiv = fs_table[fs_table['PTID'].map(lambda x: clean_id(x)) == subject_id]\n",
    "    if etiv.empty:\n",
    "        print(f\"Subject {subject_id} has no ETIV\")\n",
    "        continue\n",
    "    # remove NA values\n",
    "\n",
    "    closest_date = etiv.iloc[(etiv['EXAMDATE'] - session_dates[subject_id]).abs().argsort()[:1]]\n",
    "    etiv_per_subject[subject_id] = closest_date['ST10CV'].values[0]"
   ],
   "outputs": [],
   "execution_count": 252
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:50.279845Z",
     "start_time": "2025-01-07T16:59:41.446793Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# collect all processed files and generate feature vectors with volume information\n",
    "\n",
    "processed_folders = glob.glob(f\"{mri_fastsurfer_out}/*\")\n",
    "\n",
    "volumes_per_subject = {}\n",
    "for processed_folder in processed_folders:\n",
    "    subject_id = processed_folder.split(\"/\")[-1]\n",
    "    stats_path = os.path.join(processed_folder, \"stats/aseg+DKT.stats\")\n",
    "    if not os.path.exists(stats_path):\n",
    "        print(f\"Stats folder does not exist for {processed_folder}\")\n",
    "        continue\n",
    "\n",
    "    aseg_csv = pd.read_csv(stats_path, sep=r'\\s+', skiprows=65, names=[\"Index\", \"SegId\", \"NVoxels\", \"Volume_mm3\", \"StructName\", \"normMean\", \"normStdDev\", \"normMin\", \"normMax\", \"normRange\"])\n",
    "\n",
    "    if subject_id not in etiv_per_subject:\n",
    "        print(f\"No etiv found for subject {subject_id}\")\n",
    "        continue\n",
    "\n",
    "    etiv_volume = etiv_per_subject[subject_id]\n",
    "    volumes = []\n",
    "    for structure in structs_of_interest:\n",
    "        try:\n",
    "            volume = aseg_csv.loc[aseg_csv[\"StructName\"] == structure, \"Volume_mm3\"].iloc[0]\n",
    "            if volume > etiv_volume:\n",
    "                print(f\"Volume for structure {structure} and subject {subject_id} is larger than etiv {etiv_volume}\")\n",
    "            volumes.append(volume / etiv_volume)\n",
    "        except:\n",
    "            print(f\"Volume not found for structure {structure} in {subject_id}\")\n",
    "    volumes_per_subject[subject_id] = volumes\n"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Stats folder does not exist for /media/jfallmann/T9/University/master_thesis/dataset/mri/processed/024s4169\n",
      "Stats folder does not exist for /media/jfallmann/T9/University/master_thesis/dataset/mri/processed/137s4331\n",
      "No etiv found for subject 023s4035\n",
      "No etiv found for subject 003s4373\n"
     ]
    }
   ],
   "execution_count": 253
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:51.949341Z",
     "start_time": "2025-01-07T16:59:51.941866Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# create a numpy array in order of subjects with all volumes\n",
    "import numpy as np\n",
    "final_volumes = np.zeros((len(subjects_in_order), len(structs_of_interest)))\n",
    "\n",
    "for i, subject in enumerate(subjects_in_order):\n",
    "    if subject not in volumes_per_subject:\n",
    "        print(f\"subject {subject} not found\")\n",
    "        continue\n",
    "    volumes = volumes_per_subject[subject]\n",
    "    final_volumes[i] = np.asarray(volumes)"
   ],
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "subject 024s4169 not found\n",
      "subject 137s4331 not found\n"
     ]
    }
   ],
   "execution_count": 254
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:54.638114Z",
     "start_time": "2025-01-07T16:59:54.632743Z"
    }
   },
   "cell_type": "code",
   "source": [
    "final_volumes"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[0.1527678 , 0.0123619 , 0.00023059, ..., 0.00552149, 0.00054639,\n",
       "        0.00358952],\n",
       "       [0.07733025, 0.00861479, 0.00097355, ..., 0.00307814, 0.00037573,\n",
       "        0.00244975],\n",
       "       [0.1372052 , 0.02000124, 0.00037388, ..., 0.00546184, 0.00047201,\n",
       "        0.00430054],\n",
       "       ...,\n",
       "       [0.1456701 , 0.01478447, 0.00153851, ..., 0.00553334, 0.0007862 ,\n",
       "        0.0036883 ],\n",
       "       [0.16584519, 0.01207906, 0.00039125, ..., 0.00491391, 0.00052085,\n",
       "        0.00389304],\n",
       "       [0.17378866, 0.00530221, 0.00031831, ..., 0.007226  , 0.0005726 ,\n",
       "        0.0040541 ]])"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 255
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T16:59:57.559653Z",
     "start_time": "2025-01-07T16:59:57.555618Z"
    }
   },
   "cell_type": "code",
   "source": [
    "np.save(f\"{mri_base_path}/processed_volumes.npy\", final_volumes)"
   ],
   "outputs": [],
   "execution_count": 256
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
