{
 "cells": [
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-16T17:58:11.878113Z",
     "start_time": "2025-01-16T17:58:11.875827Z"
    }
   },
   "cell_type": "code",
   "source": [
    "dataset_base_path = \"/media/jfallmann/T9/University/master_thesis/dataset\"\n",
    "raw_snp_path = f\"{dataset_base_path}/snp/raw\"\n",
    "snp_path = f\"{dataset_base_path}/snp\"\n",
    "processed_snps = f\"{snp_path}/processed\""
   ],
   "outputs": [],
   "execution_count": 2
  },
  {
   "cell_type": "code",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2025-01-06T18:53:45.194125Z",
     "start_time": "2025-01-06T18:53:44.940619Z"
    }
   },
   "source": [
    "import pandas as pd\n",
    "\n",
    "#snp data taken from https://advp.niagads.org/\n",
    "variants = pd.read_csv(f\"{snp_path}/advp_variants.tsv\", sep='\\t')\n",
    "variants.head()"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "  #dbSNP_hg38_chr  dbSNP_hg38_position      Top SNP   P-value LocusName  \\\n",
       "0            chr1              6434683   rs12074379   0.00726      ESPN   \n",
       "1            chr1              6434683   rs12074379  8.51E-40        NR   \n",
       "2            chr1              8708071  rs112053331    0.0009      RERE   \n",
       "3            chr1              8708071  rs112053331   0.08392       NaN   \n",
       "4            chr1             11487007    rs2379135    0.0156    PTCHD2   \n",
       "\n",
       "  RA 1(Reported Allele 1) nonref_allele nonref_effect  OR_nonref  \\\n",
       "0                       T             T            NR        NaN   \n",
       "1                       T             T            NR        NaN   \n",
       "2                      NR            NR            NR        NaN   \n",
       "3                      NR            NR            NR        NaN   \n",
       "4                      NR            NR            NR        NaN   \n",
       "\n",
       "  nearest_gene_symb  Study type      Study Design  Pubmed PMID Population_map  \\\n",
       "0              ESPN   SNP-based      Disease risk     30636644      Caucasian   \n",
       "1              ESPN   SNP-based              eQTL     30636644      Caucasian   \n",
       "2              RERE   SNP-based  Cross phenotype      30010129      Caucasian   \n",
       "3              RERE  Gene-based  Cross phenotype      30010129      Caucasian   \n",
       "4             DISP3   SNP-based     Endophenotype     22245343      Caucasian   \n",
       "\n",
       "           Cohort_simple3 Sample size         Analysis group  \\\n",
       "0  ADGC, CHS, CHARGE, HRS       10191  Plan 3 (only females)   \n",
       "1  ADGC, CHS, CHARGE, HRS       10191  Plan 3 (only females)   \n",
       "2                    IGAP       54162                    All   \n",
       "3                    IGAP       54162                    All   \n",
       "4                    ADNI         757                    All   \n",
       "\n",
       "                        Phenotype Phenotype-derived most_severe_consequence  \n",
       "0                              AD                AD          intron_variant  \n",
       "1  ESPN (ILMN_1806710) expression        Expression          intron_variant  \n",
       "2                              AD                AD          intron_variant  \n",
       "3                              AD                AD          intron_variant  \n",
       "4                             MRI           Imaging          intron_variant  "
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#dbSNP_hg38_chr</th>\n",
       "      <th>dbSNP_hg38_position</th>\n",
       "      <th>Top SNP</th>\n",
       "      <th>P-value</th>\n",
       "      <th>LocusName</th>\n",
       "      <th>RA 1(Reported Allele 1)</th>\n",
       "      <th>nonref_allele</th>\n",
       "      <th>nonref_effect</th>\n",
       "      <th>OR_nonref</th>\n",
       "      <th>nearest_gene_symb</th>\n",
       "      <th>Study type</th>\n",
       "      <th>Study Design</th>\n",
       "      <th>Pubmed PMID</th>\n",
       "      <th>Population_map</th>\n",
       "      <th>Cohort_simple3</th>\n",
       "      <th>Sample size</th>\n",
       "      <th>Analysis group</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Phenotype-derived</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6434683</td>\n",
       "      <td>rs12074379</td>\n",
       "      <td>0.00726</td>\n",
       "      <td>ESPN</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ESPN</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Disease risk</td>\n",
       "      <td>30636644</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC, CHS, CHARGE, HRS</td>\n",
       "      <td>10191</td>\n",
       "      <td>Plan 3 (only females)</td>\n",
       "      <td>AD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6434683</td>\n",
       "      <td>rs12074379</td>\n",
       "      <td>8.51E-40</td>\n",
       "      <td>NR</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ESPN</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>eQTL</td>\n",
       "      <td>30636644</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC, CHS, CHARGE, HRS</td>\n",
       "      <td>10191</td>\n",
       "      <td>Plan 3 (only females)</td>\n",
       "      <td>ESPN (ILMN_1806710) expression</td>\n",
       "      <td>Expression</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8708071</td>\n",
       "      <td>rs112053331</td>\n",
       "      <td>0.0009</td>\n",
       "      <td>RERE</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RERE</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Cross phenotype</td>\n",
       "      <td>30010129</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>IGAP</td>\n",
       "      <td>54162</td>\n",
       "      <td>All</td>\n",
       "      <td>AD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8708071</td>\n",
       "      <td>rs112053331</td>\n",
       "      <td>0.08392</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>RERE</td>\n",
       "      <td>Gene-based</td>\n",
       "      <td>Cross phenotype</td>\n",
       "      <td>30010129</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>IGAP</td>\n",
       "      <td>54162</td>\n",
       "      <td>All</td>\n",
       "      <td>AD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11487007</td>\n",
       "      <td>rs2379135</td>\n",
       "      <td>0.0156</td>\n",
       "      <td>PTCHD2</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DISP3</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Endophenotype</td>\n",
       "      <td>22245343</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADNI</td>\n",
       "      <td>757</td>\n",
       "      <td>All</td>\n",
       "      <td>MRI</td>\n",
       "      <td>Imaging</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 3
  },
  {
   "cell_type": "code",
   "source": [
    "# order variants by p-value and select the top 150 variants\n",
    "top_variants = variants.sort_values('P-value')\n",
    "top_variants"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-01-06T18:53:45.239420Z",
     "start_time": "2025-01-06T18:53:45.226817Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "     #dbSNP_hg38_chr  dbSNP_hg38_position     Top SNP      P-value LocusName  \\\n",
       "2206           chr14             73809286  rs17182607            0        NR   \n",
       "3374           chr19             44892362   rs2075650            0    TOMM40   \n",
       "1182           chr11             49980436  rs11602981            0        NR   \n",
       "4870            chr4            173173789  rs62341097  0.000000006    GALNT7   \n",
       "1886           chr11            132065380   rs1629316  0.000000008       HNT   \n",
       "...              ...                  ...         ...          ...       ...   \n",
       "4629            chr3            153652877  rs11711889          NaN        NR   \n",
       "5418            chr7             10277135  rs10271466          NaN        NR   \n",
       "5419            chr7             10277135  rs10271466          NaN        NR   \n",
       "6163            chr9              8095638   rs6477258          NaN        NR   \n",
       "6164            chr9              8095638   rs6477258          NaN        NR   \n",
       "\n",
       "     RA 1(Reported Allele 1) nonref_allele nonref_effect  OR_nonref  \\\n",
       "2206                       A             A            NR        NaN   \n",
       "3374                      NR            NR            NR        NaN   \n",
       "1182                       C             C            NR        NaN   \n",
       "4870                       A             A         -1.15   0.316637   \n",
       "1886                     NaN           NaN           NaN        NaN   \n",
       "...                      ...           ...           ...        ...   \n",
       "4629                      NR            NR            NR        NaN   \n",
       "5418                      NR            NR            NR        NaN   \n",
       "5419                      NR            NR            NR        NaN   \n",
       "6163                      NR            NR            NR        NaN   \n",
       "6164                      NR            NR            NR        NaN   \n",
       "\n",
       "     nearest_gene_symb Study type      Study Design  Pubmed PMID  \\\n",
       "2206            MIDEAS  SNP-based              eQTL     31055733   \n",
       "3374            TOMM40  SNP-based  Cross phenotype      28870582   \n",
       "1182            OR4C12  SNP-based              eQTL     31055733   \n",
       "4870            GALNT7  SNP-based     Endophenotype     25188341   \n",
       "1886               NTM  SNP-based     Endophenotype     17564960   \n",
       "...                ...        ...               ...          ...   \n",
       "4629          ARHGEF26  SNP-based      Disease risk     21059989   \n",
       "5418            NDUFA4  SNP-based      Disease risk     21059989   \n",
       "5419            NDUFA4  SNP-based      Disease risk     21059989   \n",
       "6163             DMAC1  SNP-based      Disease risk     21059989   \n",
       "6164             DMAC1  SNP-based      Disease risk     21059989   \n",
       "\n",
       "                                         Population_map  \\\n",
       "2206                                          Caucasian   \n",
       "3374                                          Caucasian   \n",
       "1182                                          Caucasian   \n",
       "4870                                          Caucasian   \n",
       "1886  genetically isolated community in the southwes...   \n",
       "...                                                 ...   \n",
       "4629                                          Caucasian   \n",
       "5418                                          Caucasian   \n",
       "5419                                          Caucasian   \n",
       "6163                                          Caucasian   \n",
       "6164                                          Caucasian   \n",
       "\n",
       "              Cohort_simple3 Sample size            Analysis group  \\\n",
       "2206  ADGC, CHS, CHARGE, HRS        4806               HTN-N group   \n",
       "3374                    IGAP       54162                       All   \n",
       "1182  ADGC, CHS, CHARGE, HRS        4806               HTN-N group   \n",
       "4870                    ADGC        4046                       All   \n",
       "1886                    GRIP         197                       All   \n",
       "...                      ...         ...                       ...   \n",
       "4629                    ADGC        2449  Family-based individuals   \n",
       "5418                    ADGC        2449  Family-based individuals   \n",
       "5419                    ADGC        2449  Family-based individuals   \n",
       "6163                    ADGC        2449  Family-based individuals   \n",
       "6164                    ADGC        2449  Family-based individuals   \n",
       "\n",
       "                                              Phenotype Phenotype-derived  \\\n",
       "2206          LOC100506476 (ENSG00000259065) expression        Expression   \n",
       "3374                                                 AD                AD   \n",
       "1182                 FOLH1 (ENSG00000086205) expression        Expression   \n",
       "4870                                    Neuritic Plaque    Neuropathology   \n",
       "1886  Cognitive function (Memory - Learning (or work...         Cognitive   \n",
       "...                                                 ...               ...   \n",
       "4629                                               LOAD                AD   \n",
       "5418                                               LOAD                AD   \n",
       "5419                                               LOAD                AD   \n",
       "6163                                               LOAD                AD   \n",
       "6164                                               LOAD                AD   \n",
       "\n",
       "      most_severe_consequence  \n",
       "2206        intergenic_region  \n",
       "3374           intron_variant  \n",
       "1182  downstream_gene_variant  \n",
       "4870           intron_variant  \n",
       "1886           intron_variant  \n",
       "...                       ...  \n",
       "4629           intron_variant  \n",
       "5418        intergenic_region  \n",
       "5419        intergenic_region  \n",
       "6163        intergenic_region  \n",
       "6164        intergenic_region  \n",
       "\n",
       "[6346 rows x 20 columns]"
      ],
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#dbSNP_hg38_chr</th>\n",
       "      <th>dbSNP_hg38_position</th>\n",
       "      <th>Top SNP</th>\n",
       "      <th>P-value</th>\n",
       "      <th>LocusName</th>\n",
       "      <th>RA 1(Reported Allele 1)</th>\n",
       "      <th>nonref_allele</th>\n",
       "      <th>nonref_effect</th>\n",
       "      <th>OR_nonref</th>\n",
       "      <th>nearest_gene_symb</th>\n",
       "      <th>Study type</th>\n",
       "      <th>Study Design</th>\n",
       "      <th>Pubmed PMID</th>\n",
       "      <th>Population_map</th>\n",
       "      <th>Cohort_simple3</th>\n",
       "      <th>Sample size</th>\n",
       "      <th>Analysis group</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Phenotype-derived</th>\n",
       "      <th>most_severe_consequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2206</th>\n",
       "      <td>chr14</td>\n",
       "      <td>73809286</td>\n",
       "      <td>rs17182607</td>\n",
       "      <td>0</td>\n",
       "      <td>NR</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MIDEAS</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>eQTL</td>\n",
       "      <td>31055733</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC, CHS, CHARGE, HRS</td>\n",
       "      <td>4806</td>\n",
       "      <td>HTN-N group</td>\n",
       "      <td>LOC100506476 (ENSG00000259065) expression</td>\n",
       "      <td>Expression</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3374</th>\n",
       "      <td>chr19</td>\n",
       "      <td>44892362</td>\n",
       "      <td>rs2075650</td>\n",
       "      <td>0</td>\n",
       "      <td>TOMM40</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TOMM40</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Cross phenotype</td>\n",
       "      <td>28870582</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>IGAP</td>\n",
       "      <td>54162</td>\n",
       "      <td>All</td>\n",
       "      <td>AD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1182</th>\n",
       "      <td>chr11</td>\n",
       "      <td>49980436</td>\n",
       "      <td>rs11602981</td>\n",
       "      <td>0</td>\n",
       "      <td>NR</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>OR4C12</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>eQTL</td>\n",
       "      <td>31055733</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC, CHS, CHARGE, HRS</td>\n",
       "      <td>4806</td>\n",
       "      <td>HTN-N group</td>\n",
       "      <td>FOLH1 (ENSG00000086205) expression</td>\n",
       "      <td>Expression</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4870</th>\n",
       "      <td>chr4</td>\n",
       "      <td>173173789</td>\n",
       "      <td>rs62341097</td>\n",
       "      <td>0.000000006</td>\n",
       "      <td>GALNT7</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>-1.15</td>\n",
       "      <td>0.316637</td>\n",
       "      <td>GALNT7</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Endophenotype</td>\n",
       "      <td>25188341</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC</td>\n",
       "      <td>4046</td>\n",
       "      <td>All</td>\n",
       "      <td>Neuritic Plaque</td>\n",
       "      <td>Neuropathology</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1886</th>\n",
       "      <td>chr11</td>\n",
       "      <td>132065380</td>\n",
       "      <td>rs1629316</td>\n",
       "      <td>0.000000008</td>\n",
       "      <td>HNT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NTM</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Endophenotype</td>\n",
       "      <td>17564960</td>\n",
       "      <td>genetically isolated community in the southwes...</td>\n",
       "      <td>GRIP</td>\n",
       "      <td>197</td>\n",
       "      <td>All</td>\n",
       "      <td>Cognitive function (Memory - Learning (or work...</td>\n",
       "      <td>Cognitive</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4629</th>\n",
       "      <td>chr3</td>\n",
       "      <td>153652877</td>\n",
       "      <td>rs11711889</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ARHGEF26</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Disease risk</td>\n",
       "      <td>21059989</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC</td>\n",
       "      <td>2449</td>\n",
       "      <td>Family-based individuals</td>\n",
       "      <td>LOAD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intron_variant</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5418</th>\n",
       "      <td>chr7</td>\n",
       "      <td>10277135</td>\n",
       "      <td>rs10271466</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NDUFA4</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Disease risk</td>\n",
       "      <td>21059989</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC</td>\n",
       "      <td>2449</td>\n",
       "      <td>Family-based individuals</td>\n",
       "      <td>LOAD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5419</th>\n",
       "      <td>chr7</td>\n",
       "      <td>10277135</td>\n",
       "      <td>rs10271466</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NDUFA4</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Disease risk</td>\n",
       "      <td>21059989</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC</td>\n",
       "      <td>2449</td>\n",
       "      <td>Family-based individuals</td>\n",
       "      <td>LOAD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6163</th>\n",
       "      <td>chr9</td>\n",
       "      <td>8095638</td>\n",
       "      <td>rs6477258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DMAC1</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Disease risk</td>\n",
       "      <td>21059989</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC</td>\n",
       "      <td>2449</td>\n",
       "      <td>Family-based individuals</td>\n",
       "      <td>LOAD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6164</th>\n",
       "      <td>chr9</td>\n",
       "      <td>8095638</td>\n",
       "      <td>rs6477258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>DMAC1</td>\n",
       "      <td>SNP-based</td>\n",
       "      <td>Disease risk</td>\n",
       "      <td>21059989</td>\n",
       "      <td>Caucasian</td>\n",
       "      <td>ADGC</td>\n",
       "      <td>2449</td>\n",
       "      <td>Family-based individuals</td>\n",
       "      <td>LOAD</td>\n",
       "      <td>AD</td>\n",
       "      <td>intergenic_region</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6346 rows × 20 columns</p>\n",
       "</div>"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 4
  },
  {
   "cell_type": "code",
   "source": [
    "top_k_snps = 1024\n",
    "top_snps = top_variants['Top SNP'].unique()[:top_k_snps]\n",
    "top_snps = set([str(snp).lower() for snp in top_snps])\n",
    "print(f\"Selected {len(top_snps)} top snps, including: {list(top_snps)[:5]}\")"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-01-06T19:34:15.981682Z",
     "start_time": "2025-01-06T19:34:15.977948Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Selected 1024 top snps, including: ['rs1736556', 'rs10911736', 'rs676768', 'rs12422883', 'rs1887726']\n"
     ]
    }
   ],
   "execution_count": 22
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T18:54:31.911229Z",
     "start_time": "2025-01-06T18:54:31.908579Z"
    }
   },
   "cell_type": "code",
   "source": [
    "MISSING_CALL = 0\n",
    "REFERENCE = 1\n",
    "HETEROZYGOUS_MUTANT = 2\n",
    "HOMOZYGOUS_MUTANT = 3"
   ],
   "outputs": [],
   "execution_count": 7
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T18:54:33.148337Z",
     "start_time": "2025-01-06T18:54:33.144296Z"
    }
   },
   "cell_type": "code",
   "source": [
    "class PatientGenome:\n",
    "    def __init__(self, snps: list, subject_id: str):\n",
    "        # we store a map with snp-id to type and init all snps with missing call\n",
    "        self.genome = {snp: MISSING_CALL for snp in snps}\n",
    "        self.subject_id = subject_id\n",
    "\n",
    "    def add_genome(self, snp_id, snp_type):\n",
    "        self.genome[snp_id] = snp_type"
   ],
   "outputs": [],
   "execution_count": 8
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T07:49:46.706005Z",
     "start_time": "2025-01-07T07:49:46.703431Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# load relevant subjects\n",
    "subjects = []\n",
    "\n",
    "with open(f'{dataset_base_path}/subjects.txt') as subjects_file:\n",
    "    for line in subjects_file:\n",
    "        subjects.append(line.strip())"
   ],
   "outputs": [],
   "execution_count": 35
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T18:54:36.206322Z",
     "start_time": "2025-01-06T18:54:36.203535Z"
    }
   },
   "cell_type": "code",
   "source": [
    "def clean_id(id: str) -> str:\n",
    "    return id.replace(\"_\", \"\").replace(\".\", \"\").replace(\"-\", \"\").lower()"
   ],
   "outputs": [],
   "execution_count": 10
  },
  {
   "cell_type": "code",
   "source": [
    "import csv\n",
    "import os\n",
    "import re\n",
    "import glob\n",
    "from tqdm.notebook import tqdm\n",
    "\n",
    "genome_data = []\n",
    "n_worker = 8\n",
    "\n",
    "def get_genotype_with_highest_quality(samples):\n",
    "    highest_quality = -1\n",
    "    highest_quality_sample = None\n",
    "    for sample in samples:\n",
    "        sample_data = sample.split(':')\n",
    "        quality = sample_data[1]\n",
    "        quality_numeric = int(quality)\n",
    "        if quality_numeric > highest_quality:\n",
    "            highest_quality = quality_numeric\n",
    "            highest_quality_sample = sample\n",
    "    return highest_quality_sample\n",
    "\n",
    "\n",
    "def get_snps_from_vcf(vcf_file, top_snps):\n",
    "    genome_split_regex = r'(.*)_SNPs'\n",
    "    subject_id = re.search(genome_split_regex, os.path.basename(vcf_file)).group(1)\n",
    "    subject_id = clean_id(subject_id)\n",
    "    if subject_id not in subjects:\n",
    "        return None\n",
    "    genome = PatientGenome(top_snps, subject_id)\n",
    "    # read vcf file\n",
    "    with open(vcf_file, 'r') as f:\n",
    "        reader = csv.reader(f, delimiter='\\t')\n",
    "        for row in reader:\n",
    "            if row[0].startswith(\"#\"):\n",
    "                continue\n",
    "            chrom, pos, id, ref, alt, qual, filter, info, format, *samples = row\n",
    "\n",
    "            if id.lower() not in top_snps:\n",
    "                continue\n",
    "\n",
    "            # select the sample with the highest quality score\n",
    "            sample = get_genotype_with_highest_quality(samples)\n",
    "            sample_data = sample.split(':')\n",
    "            genotype = sample_data[0]\n",
    "            if genotype == '0/0':\n",
    "                genome.add_genome(id, REFERENCE)\n",
    "            elif genotype == '0/1':\n",
    "                genome.add_genome(id, HETEROZYGOUS_MUTANT)\n",
    "            elif genotype == '1/1':\n",
    "                genome.add_genome(id, HOMOZYGOUS_MUTANT)\n",
    "            else:\n",
    "                genome.add_genome(id, MISSING_CALL)\n",
    "    return genome\n",
    "\n",
    "\n",
    "vcf_files = glob.glob(f\"{raw_snp_path}/**/*.vcf\")\n",
    "\n",
    "for vcf_file in tqdm(vcf_files):\n",
    "    genome = get_snps_from_vcf(vcf_file, top_snps)\n",
    "    if genome is not None:\n",
    "        genome_data.append(genome)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-01-06T19:22:35.457131Z",
     "start_time": "2025-01-06T18:54:39.818930Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "  0%|          | 0/809 [00:00<?, ?it/s]"
      ],
      "application/vnd.jupyter.widget-view+json": {
       "version_major": 2,
       "version_minor": 0,
       "model_id": "234b8ff6d8474700bfc1f195dbca8cee"
      }
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "execution_count": 11
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T07:49:49.350530Z",
     "start_time": "2025-01-07T07:49:49.345155Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# remove all genome data where subject is not in subjects\n",
    "genome_data = [genome for genome in genome_data if genome.subject_id in subjects]\n",
    "len(genome_data)"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "409"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 36
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T07:49:52.238319Z",
     "start_time": "2025-01-07T07:49:52.193801Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# remove all keys of all genomes not in top_snps\n",
    "for genome in genome_data:\n",
    "    for key in list(genome.genome.keys()):\n",
    "        if key not in top_snps:\n",
    "            del genome.genome[key]"
   ],
   "outputs": [],
   "execution_count": 37
  },
  {
   "cell_type": "code",
   "source": [
    "# get average number of missing calls\n",
    "missing_calls = []\n",
    "for genome in genome_data:\n",
    "    missing_calls.append(list(genome.genome.values()).count(MISSING_CALL))\n",
    "\n",
    "avg = sum(missing_calls) / len(missing_calls)\n",
    "std = (sum((x - avg) ** 2 for x in missing_calls) / len(missing_calls)) ** 0.5\n",
    "avg, std"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-01-07T07:49:53.348414Z",
     "start_time": "2025-01-07T07:49:53.335726Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(666.8606356968215, 15.590657367503841)"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 38
  },
  {
   "cell_type": "code",
   "source": [
    "# any genome with less than 1024 snps?\n",
    "for genome in genome_data:\n",
    "    if len(genome.genome) < 1024:\n",
    "        print(genome.subject_id)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2025-01-07T07:49:55.339818Z",
     "start_time": "2025-01-07T07:49:55.336208Z"
    }
   },
   "outputs": [],
   "execution_count": 39
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T07:49:56.866338Z",
     "start_time": "2025-01-07T07:49:56.745767Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# create an array ordered by subjects and columns being the snp values in the same order as top_snps\n",
    "import numpy as np\n",
    "\n",
    "genomes = np.zeros((len(genome_data), len(top_snps)), dtype=np.int8)\n",
    "\n",
    "# sort genome in same order as subjects list\n",
    "genome_data.sort(key=lambda x: subjects.index(x.subject_id))\n",
    "\n",
    "for i, genome in enumerate(genome_data):\n",
    "    for j, snp in enumerate(top_snps):\n",
    "        genomes[i, j] = genome.genome[snp]"
   ],
   "outputs": [],
   "execution_count": 40
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-07T07:49:58.528741Z",
     "start_time": "2025-01-07T07:49:58.525182Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# create folder for processed snps\n",
    "if not os.path.exists(processed_snps):\n",
    "    os.makedirs(processed_snps)\n",
    "\n",
    "# store list of snp names to file\n",
    "with open(f\"{processed_snps}/snp_names.txt\", 'w') as f:\n",
    "    for snp in top_snps:\n",
    "        f.write(f\"{snp}\\n\")\n",
    "\n",
    "# store the genomes to file\n",
    "np.save(f\"{processed_snps}/genomes.npy\", genomes)"
   ],
   "outputs": [],
   "execution_count": 41
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-16T18:21:08.860313Z",
     "start_time": "2025-01-16T18:21:08.842007Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# load the genomes\n",
    "import numpy as np\n",
    "genomes = np.load(f\"{processed_snps}/genomes.npy\")\n"
   ],
   "outputs": [],
   "execution_count": 5
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-16T18:21:10.123731Z",
     "start_time": "2025-01-16T18:21:10.119525Z"
    }
   },
   "cell_type": "code",
   "source": [
    "genomes"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[2, 0, 3, ..., 3, 0, 0],\n",
       "       [2, 0, 2, ..., 3, 0, 0],\n",
       "       [3, 3, 3, ..., 3, 0, 0],\n",
       "       ...,\n",
       "       [2, 3, 3, ..., 3, 3, 0],\n",
       "       [2, 0, 3, ..., 3, 0, 0],\n",
       "       [2, 0, 3, ..., 3, 0, 0]], dtype=int8)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 6
  },
  {
   "metadata": {},
   "cell_type": "code",
   "outputs": [],
   "execution_count": null,
   "source": [
    "# the rows of the genomes array are the subjects and the columns are the snps\n",
    "# we want to"
   ]
  },
  {
   "metadata": {
    "ExecuteTime": {
     "end_time": "2025-01-06T22:43:21.319551Z",
     "start_time": "2025-01-06T22:43:20.759576Z"
    }
   },
   "cell_type": "code",
   "source": [
    "# is there a column with only missing calls?\n",
    "missing_calls = []\n",
    "for i in range(genomes.shape[1]):\n",
    "    missing_calls.append(list(genomes[:, i]).count(MISSING_CALL))\n",
    "\n",
    "missing_calls"
   ],
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[79,\n",
       " 366,\n",
       " 9,\n",
       " 407,\n",
       " 128,\n",
       " 387,\n",
       " 63,\n",
       " 211,\n",
       " 74,\n",
       " 371,\n",
       " 214,\n",
       " 409,\n",
       " 246,\n",
       " 117,\n",
       " 322,\n",
       " 333,\n",
       " 312,\n",
       " 309,\n",
       " 347,\n",
       " 10,\n",
       " 312,\n",
       " 404,\n",
       " 384,\n",
       " 181,\n",
       " 23,\n",
       " 195,\n",
       " 365,\n",
       " 64,\n",
       " 49,\n",
       " 355,\n",
       " 326,\n",
       " 34,\n",
       " 314,\n",
       " 212,\n",
       " 102,\n",
       " 344,\n",
       " 35,\n",
       " 49,\n",
       " 287,\n",
       " 132,\n",
       " 29,\n",
       " 305,\n",
       " 311,\n",
       " 366,\n",
       " 136,\n",
       " 389,\n",
       " 194,\n",
       " 400,\n",
       " 374,\n",
       " 306,\n",
       " 219,\n",
       " 356,\n",
       " 389,\n",
       " 353,\n",
       " 241,\n",
       " 329,\n",
       " 254,\n",
       " 182,\n",
       " 394,\n",
       " 301,\n",
       " 390,\n",
       " 282,\n",
       " 215,\n",
       " 353,\n",
       " 330,\n",
       " 348,\n",
       " 377,\n",
       " 402,\n",
       " 146,\n",
       " 157,\n",
       " 216,\n",
       " 208,\n",
       " 272,\n",
       " 365,\n",
       " 397,\n",
       " 315,\n",
       " 40,\n",
       " 104,\n",
       " 162,\n",
       " 410,\n",
       " 406,\n",
       " 393,\n",
       " 67,\n",
       " 355,\n",
       " 62,\n",
       " 62,\n",
       " 398,\n",
       " 400,\n",
       " 178,\n",
       " 154,\n",
       " 407,\n",
       " 69,\n",
       " 322,\n",
       " 388,\n",
       " 388,\n",
       " 394,\n",
       " 119,\n",
       " 120,\n",
       " 346,\n",
       " 152,\n",
       " 404,\n",
       " 194,\n",
       " 82,\n",
       " 76,\n",
       " 84,\n",
       " 370,\n",
       " 330,\n",
       " 80,\n",
       " 333,\n",
       " 387,\n",
       " 343,\n",
       " 283,\n",
       " 156,\n",
       " 74,\n",
       " 376,\n",
       " 154,\n",
       " 275,\n",
       " 359,\n",
       " 110,\n",
       " 6,\n",
       " 356,\n",
       " 136,\n",
       " 409,\n",
       " 273,\n",
       " 273,\n",
       " 349,\n",
       " 402,\n",
       " 102,\n",
       " 178,\n",
       " 327,\n",
       " 382,\n",
       " 282,\n",
       " 335,\n",
       " 409,\n",
       " 171,\n",
       " 410,\n",
       " 381,\n",
       " 358,\n",
       " 237,\n",
       " 309,\n",
       " 400,\n",
       " 382,\n",
       " 382,\n",
       " 393,\n",
       " 365,\n",
       " 409,\n",
       " 110,\n",
       " 365,\n",
       " 341,\n",
       " 236,\n",
       " 283,\n",
       " 147,\n",
       " 89,\n",
       " 398,\n",
       " 377,\n",
       " 410,\n",
       " 78,\n",
       " 346,\n",
       " 243,\n",
       " 271,\n",
       " 346,\n",
       " 175,\n",
       " 180,\n",
       " 330,\n",
       " 7,\n",
       " 405,\n",
       " 389,\n",
       " 318,\n",
       " 156,\n",
       " 409,\n",
       " 371,\n",
       " 109,\n",
       " 219,\n",
       " 313,\n",
       " 229,\n",
       " 71,\n",
       " 390,\n",
       " 395,\n",
       " 288,\n",
       " 324,\n",
       " 351,\n",
       " 285,\n",
       " 126,\n",
       " 259,\n",
       " 326,\n",
       " 312,\n",
       " 223,\n",
       " 335,\n",
       " 404,\n",
       " 320,\n",
       " 391,\n",
       " 127,\n",
       " 410,\n",
       " 353,\n",
       " 150,\n",
       " 382,\n",
       " 330,\n",
       " 93,\n",
       " 4,\n",
       " 361,\n",
       " 94,\n",
       " 409,\n",
       " 59,\n",
       " 334,\n",
       " 48,\n",
       " 370,\n",
       " 277,\n",
       " 112,\n",
       " 396,\n",
       " 328,\n",
       " 132,\n",
       " 192,\n",
       " 227,\n",
       " 160,\n",
       " 352,\n",
       " 101,\n",
       " 16,\n",
       " 410,\n",
       " 404,\n",
       " 10,\n",
       " 134,\n",
       " 375,\n",
       " 410,\n",
       " 407,\n",
       " 101,\n",
       " 410,\n",
       " 408,\n",
       " 45,\n",
       " 410,\n",
       " 403,\n",
       " 398,\n",
       " 131,\n",
       " 372,\n",
       " 185,\n",
       " 128,\n",
       " 99,\n",
       " 132,\n",
       " 100,\n",
       " 253,\n",
       " 408,\n",
       " 400,\n",
       " 83,\n",
       " 88,\n",
       " 302,\n",
       " 355,\n",
       " 383,\n",
       " 343,\n",
       " 398,\n",
       " 257,\n",
       " 319,\n",
       " 378,\n",
       " 402,\n",
       " 400,\n",
       " 93,\n",
       " 326,\n",
       " 80,\n",
       " 370,\n",
       " 401,\n",
       " 361,\n",
       " 309,\n",
       " 256,\n",
       " 370,\n",
       " 1,\n",
       " 343,\n",
       " 375,\n",
       " 358,\n",
       " 369,\n",
       " 19,\n",
       " 268,\n",
       " 88,\n",
       " 372,\n",
       " 410,\n",
       " 211,\n",
       " 129,\n",
       " 409,\n",
       " 97,\n",
       " 36,\n",
       " 264,\n",
       " 9,\n",
       " 41,\n",
       " 265,\n",
       " 285,\n",
       " 356,\n",
       " 410,\n",
       " 136,\n",
       " 260,\n",
       " 402,\n",
       " 410,\n",
       " 348,\n",
       " 387,\n",
       " 329,\n",
       " 289,\n",
       " 131,\n",
       " 4,\n",
       " 389,\n",
       " 298,\n",
       " 42,\n",
       " 371,\n",
       " 258,\n",
       " 409,\n",
       " 365,\n",
       " 285,\n",
       " 135,\n",
       " 339,\n",
       " 11,\n",
       " 50,\n",
       " 405,\n",
       " 372,\n",
       " 298,\n",
       " 406,\n",
       " 195,\n",
       " 395,\n",
       " 264,\n",
       " 395,\n",
       " 79,\n",
       " 258,\n",
       " 147,\n",
       " 393,\n",
       " 396,\n",
       " 131,\n",
       " 409,\n",
       " 394,\n",
       " 335,\n",
       " 310,\n",
       " 26,\n",
       " 150,\n",
       " 299,\n",
       " 294,\n",
       " 96,\n",
       " 403,\n",
       " 392,\n",
       " 71,\n",
       " 294,\n",
       " 328,\n",
       " 238,\n",
       " 230,\n",
       " 410,\n",
       " 403,\n",
       " 388,\n",
       " 267,\n",
       " 410,\n",
       " 275,\n",
       " 410,\n",
       " 312,\n",
       " 372,\n",
       " 142,\n",
       " 297,\n",
       " 249,\n",
       " 373,\n",
       " 390,\n",
       " 401,\n",
       " 133,\n",
       " 277,\n",
       " 322,\n",
       " 354,\n",
       " 102,\n",
       " 329,\n",
       " 387,\n",
       " 78,\n",
       " 359,\n",
       " 303,\n",
       " 299,\n",
       " 377,\n",
       " 410,\n",
       " 404,\n",
       " 235,\n",
       " 173,\n",
       " 390,\n",
       " 409,\n",
       " 366,\n",
       " 361,\n",
       " 410,\n",
       " 406,\n",
       " 284,\n",
       " 145,\n",
       " 305,\n",
       " 250,\n",
       " 223,\n",
       " 401,\n",
       " 408,\n",
       " 172,\n",
       " 336,\n",
       " 283,\n",
       " 63,\n",
       " 301,\n",
       " 315,\n",
       " 409,\n",
       " 304,\n",
       " 118,\n",
       " 89,\n",
       " 238,\n",
       " 405,\n",
       " 408,\n",
       " 355,\n",
       " 15,\n",
       " 104,\n",
       " 68,\n",
       " 154,\n",
       " 409,\n",
       " 299,\n",
       " 406,\n",
       " 132,\n",
       " 219,\n",
       " 269,\n",
       " 130,\n",
       " 410,\n",
       " 12,\n",
       " 335,\n",
       " 325,\n",
       " 106,\n",
       " 406,\n",
       " 389,\n",
       " 319,\n",
       " 377,\n",
       " 392,\n",
       " 50,\n",
       " 57,\n",
       " 100,\n",
       " 161,\n",
       " 195,\n",
       " 268,\n",
       " 396,\n",
       " 165,\n",
       " 315,\n",
       " 371,\n",
       " 361,\n",
       " 212,\n",
       " 345,\n",
       " 406,\n",
       " 259,\n",
       " 410,\n",
       " 410,\n",
       " 410,\n",
       " 294,\n",
       " 272,\n",
       " 276,\n",
       " 196,\n",
       " 311,\n",
       " 386,\n",
       " 102,\n",
       " 13,\n",
       " 117,\n",
       " 381,\n",
       " 331,\n",
       " 215,\n",
       " 69,\n",
       " 291,\n",
       " 39,\n",
       " 410,\n",
       " 309,\n",
       " 77,\n",
       " 181,\n",
       " 26,\n",
       " 70,\n",
       " 350,\n",
       " 269,\n",
       " 345,\n",
       " 410,\n",
       " 410,\n",
       " 136,\n",
       " 192,\n",
       " 141,\n",
       " 407,\n",
       " 135,\n",
       " 198,\n",
       " 18,\n",
       " 169,\n",
       " 408,\n",
       " 193,\n",
       " 56,\n",
       " 409,\n",
       " 94,\n",
       " 346,\n",
       " 255,\n",
       " 224,\n",
       " 209,\n",
       " 307,\n",
       " 410,\n",
       " 363,\n",
       " 393,\n",
       " 397,\n",
       " 350,\n",
       " 355,\n",
       " 95,\n",
       " 142,\n",
       " 102,\n",
       " 323,\n",
       " 268,\n",
       " 305,\n",
       " 410,\n",
       " 393,\n",
       " 410,\n",
       " 285,\n",
       " 191,\n",
       " 299,\n",
       " 306,\n",
       " 112,\n",
       " 300,\n",
       " 201,\n",
       " 46,\n",
       " 153,\n",
       " 316,\n",
       " 189,\n",
       " 349,\n",
       " 402,\n",
       " 410,\n",
       " 191,\n",
       " 46,\n",
       " 281,\n",
       " 360,\n",
       " 408,\n",
       " 61,\n",
       " 160,\n",
       " 388,\n",
       " 325,\n",
       " 233,\n",
       " 10,\n",
       " 50,\n",
       " 131,\n",
       " 12,\n",
       " 148,\n",
       " 322,\n",
       " 83,\n",
       " 125,\n",
       " 403,\n",
       " 221,\n",
       " 381,\n",
       " 409,\n",
       " 230,\n",
       " 303,\n",
       " 390,\n",
       " 10,\n",
       " 153,\n",
       " 286,\n",
       " 330,\n",
       " 213,\n",
       " 328,\n",
       " 294,\n",
       " 131,\n",
       " 400,\n",
       " 349,\n",
       " 258,\n",
       " 215,\n",
       " 10,\n",
       " 347,\n",
       " 42,\n",
       " 345,\n",
       " 400,\n",
       " 235,\n",
       " 400,\n",
       " 410,\n",
       " 65,\n",
       " 406,\n",
       " 197,\n",
       " 265,\n",
       " 375,\n",
       " 16,\n",
       " 115,\n",
       " 183,\n",
       " 0,\n",
       " 295,\n",
       " 409,\n",
       " 385,\n",
       " 350,\n",
       " 331,\n",
       " 282,\n",
       " 408,\n",
       " 325,\n",
       " 406,\n",
       " 410,\n",
       " 235,\n",
       " 410,\n",
       " 97,\n",
       " 388,\n",
       " 401,\n",
       " 21,\n",
       " 372,\n",
       " 37,\n",
       " 144,\n",
       " 97,\n",
       " 368,\n",
       " 406,\n",
       " 409,\n",
       " 345,\n",
       " 403,\n",
       " 347,\n",
       " 402,\n",
       " 272,\n",
       " 187,\n",
       " 139,\n",
       " 349,\n",
       " 22,\n",
       " 386,\n",
       " 410,\n",
       " 410,\n",
       " 180,\n",
       " 359,\n",
       " 326,\n",
       " 341,\n",
       " 82,\n",
       " 224,\n",
       " 377,\n",
       " 62,\n",
       " 169,\n",
       " 129,\n",
       " 228,\n",
       " 332,\n",
       " 406,\n",
       " 359,\n",
       " 36,\n",
       " 92,\n",
       " 246,\n",
       " 287,\n",
       " 375,\n",
       " 282,\n",
       " 75,\n",
       " 408,\n",
       " 162,\n",
       " 410,\n",
       " 360,\n",
       " 404,\n",
       " 275,\n",
       " 269,\n",
       " 372,\n",
       " 220,\n",
       " 14,\n",
       " 404,\n",
       " 410,\n",
       " 213,\n",
       " 379,\n",
       " 410,\n",
       " 374,\n",
       " 322,\n",
       " 128,\n",
       " 301,\n",
       " 74,\n",
       " 347,\n",
       " 84,\n",
       " 397,\n",
       " 356,\n",
       " 25,\n",
       " 353,\n",
       " 310,\n",
       " 164,\n",
       " 136,\n",
       " 369,\n",
       " 10,\n",
       " 348,\n",
       " 282,\n",
       " 211,\n",
       " 399,\n",
       " 146,\n",
       " 396,\n",
       " 322,\n",
       " 410,\n",
       " 171,\n",
       " 262,\n",
       " 218,\n",
       " 395,\n",
       " 410,\n",
       " 218,\n",
       " 409,\n",
       " 406,\n",
       " 398,\n",
       " 352,\n",
       " 407,\n",
       " 303,\n",
       " 343,\n",
       " 393,\n",
       " 406,\n",
       " 323,\n",
       " 372,\n",
       " 410,\n",
       " 400,\n",
       " 225,\n",
       " 212,\n",
       " 95,\n",
       " 201,\n",
       " 351,\n",
       " 262,\n",
       " 193,\n",
       " 117,\n",
       " 205,\n",
       " 150,\n",
       " 37,\n",
       " 289,\n",
       " 301,\n",
       " 4,\n",
       " 80,\n",
       " 367,\n",
       " 135,\n",
       " 405,\n",
       " 368,\n",
       " 410,\n",
       " 324,\n",
       " 375,\n",
       " 169,\n",
       " 247,\n",
       " 389,\n",
       " 219,\n",
       " 353,\n",
       " 405,\n",
       " 408,\n",
       " 135,\n",
       " 271,\n",
       " 337,\n",
       " 194,\n",
       " 178,\n",
       " 26,\n",
       " 10,\n",
       " 311,\n",
       " 278,\n",
       " 315,\n",
       " 124,\n",
       " 351,\n",
       " 157,\n",
       " 394,\n",
       " 140,\n",
       " 364,\n",
       " 123,\n",
       " 317,\n",
       " 167,\n",
       " 169,\n",
       " 390,\n",
       " 265,\n",
       " 341,\n",
       " 169,\n",
       " 304,\n",
       " 73,\n",
       " 35,\n",
       " 410,\n",
       " 408,\n",
       " 410,\n",
       " 389,\n",
       " 347,\n",
       " 216,\n",
       " 137,\n",
       " 220,\n",
       " 367,\n",
       " 140,\n",
       " 314,\n",
       " 274,\n",
       " 224,\n",
       " 67,\n",
       " 129,\n",
       " 407,\n",
       " 21,\n",
       " 11,\n",
       " 236,\n",
       " 356,\n",
       " 291,\n",
       " 63,\n",
       " 191,\n",
       " 370,\n",
       " 265,\n",
       " 344,\n",
       " 78,\n",
       " 155,\n",
       " 369,\n",
       " 112,\n",
       " 160,\n",
       " 42,\n",
       " 115,\n",
       " 159,\n",
       " 410,\n",
       " 410,\n",
       " 409,\n",
       " 279,\n",
       " 406,\n",
       " 374,\n",
       " 409,\n",
       " 398,\n",
       " 405,\n",
       " 323,\n",
       " 63,\n",
       " 258,\n",
       " 403,\n",
       " 202,\n",
       " 36,\n",
       " 402,\n",
       " 404,\n",
       " 280,\n",
       " 409,\n",
       " 331,\n",
       " 38,\n",
       " 25,\n",
       " 23,\n",
       " 10,\n",
       " 302,\n",
       " 357,\n",
       " 306,\n",
       " 410,\n",
       " 407,\n",
       " 279,\n",
       " 384,\n",
       " 276,\n",
       " 409,\n",
       " 94,\n",
       " 410,\n",
       " 13,\n",
       " 388,\n",
       " 408,\n",
       " 410,\n",
       " 390,\n",
       " 312,\n",
       " 144,\n",
       " 209,\n",
       " 148,\n",
       " 102,\n",
       " 317,\n",
       " 335,\n",
       " 263,\n",
       " 378,\n",
       " 91,\n",
       " 375,\n",
       " 403,\n",
       " 335,\n",
       " 405,\n",
       " 391,\n",
       " 404,\n",
       " 339,\n",
       " 30,\n",
       " 333,\n",
       " 52,\n",
       " 407,\n",
       " 76,\n",
       " 307,\n",
       " 192,\n",
       " 410,\n",
       " 283,\n",
       " 156,\n",
       " 96,\n",
       " 383,\n",
       " 343,\n",
       " 405,\n",
       " 320,\n",
       " 144,\n",
       " 30,\n",
       " 410,\n",
       " 180,\n",
       " 405,\n",
       " 273,\n",
       " 351,\n",
       " 94,\n",
       " 393,\n",
       " 199,\n",
       " 403,\n",
       " 364,\n",
       " 127,\n",
       " 99,\n",
       " 127,\n",
       " 313,\n",
       " 102,\n",
       " 333,\n",
       " 398,\n",
       " 118,\n",
       " 400,\n",
       " 311,\n",
       " 268,\n",
       " 381,\n",
       " 390,\n",
       " 157,\n",
       " 253,\n",
       " 19,\n",
       " 153,\n",
       " 161,\n",
       " 409,\n",
       " 201,\n",
       " 339,\n",
       " 382,\n",
       " 180,\n",
       " 242,\n",
       " 293,\n",
       " 407,\n",
       " 406,\n",
       " 190,\n",
       " 397,\n",
       " 407,\n",
       " 54,\n",
       " 400,\n",
       " 309,\n",
       " 410,\n",
       " 10,\n",
       " 404,\n",
       " 407,\n",
       " 248,\n",
       " 52,\n",
       " 309,\n",
       " 271,\n",
       " 259,\n",
       " 408,\n",
       " 173,\n",
       " 272,\n",
       " 182,\n",
       " 241,\n",
       " 43,\n",
       " 305,\n",
       " 107,\n",
       " 46,\n",
       " 357,\n",
       " 14,\n",
       " 406,\n",
       " 409,\n",
       " 394,\n",
       " 405,\n",
       " 332,\n",
       " 350,\n",
       " 410,\n",
       " 410,\n",
       " 387,\n",
       " 328,\n",
       " 119,\n",
       " 379,\n",
       " 384,\n",
       " 322,\n",
       " 366,\n",
       " 67,\n",
       " 410,\n",
       " 163,\n",
       " 409,\n",
       " 283,\n",
       " 186,\n",
       " 407,\n",
       " 338,\n",
       " 214,\n",
       " 410,\n",
       " 348,\n",
       " 376,\n",
       " 8,\n",
       " 132,\n",
       " 314,\n",
       " 82,\n",
       " 409,\n",
       " 395,\n",
       " 225,\n",
       " 153,\n",
       " 30,\n",
       " 212,\n",
       " 137,\n",
       " 409,\n",
       " 409,\n",
       " 376,\n",
       " 410,\n",
       " 26,\n",
       " 408,\n",
       " 70,\n",
       " 361,\n",
       " 15,\n",
       " 360,\n",
       " 398,\n",
       " 409,\n",
       " 410,\n",
       " 189,\n",
       " 327,\n",
       " 274,\n",
       " 54,\n",
       " 397,\n",
       " 409,\n",
       " 410,\n",
       " 142,\n",
       " 101,\n",
       " 93,\n",
       " 346,\n",
       " 276,\n",
       " 410,\n",
       " 381,\n",
       " 49,\n",
       " 394,\n",
       " 354,\n",
       " 375,\n",
       " 174,\n",
       " 372,\n",
       " 399,\n",
       " 189,\n",
       " 404,\n",
       " 401,\n",
       " 272,\n",
       " 330,\n",
       " 158,\n",
       " 401,\n",
       " 409,\n",
       " 408,\n",
       " 388,\n",
       " 10,\n",
       " 410,\n",
       " 204,\n",
       " 292,\n",
       " 410,\n",
       " 391,\n",
       " 124,\n",
       " 410,\n",
       " 387,\n",
       " 403,\n",
       " 401,\n",
       " 356,\n",
       " 336,\n",
       " 63,\n",
       " 134,\n",
       " 246,\n",
       " 266,\n",
       " ...]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "execution_count": 34
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
